<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>dDAGtermSim @ dnet 1.0.2</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 1.0.2</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="rdata.html"><i class="icon-file icon-white"></i> RData</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to calculate pair-wise semantic similarity between input terms based on a direct acyclic graph (DAG) with annotated data</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>dDAGtermSim</code> is supposed to calculate pair-wise
semantic similarity between input terms based on a direct
acyclic graph (DAG) with annotated data. Parallel computing
is also supported for Linux or Mac operating systems.</p>
  
    <h3>Usage</h3>
    <pre>dDAGtermSim(g, terms = NULL, method = c("Resnik", "Lin", "Schlicker", "Jiang", "Pesquita"), 
  fast = T, parallel = TRUE, multicores = NULL, verbose = T)</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>g</dt>
      <dd>an object of class "igraph" or "graphNEL"</dd>
      <dt>terms</dt>
      <dd>the terms/nodes between which pair-wise
semantic similarity is calculated. If NULL, all terms in
the input DAG will be used for calcluation, which is very
prohibitively expensive!</dd>
      <dt>method</dt>
      <dd>the method used to measure semantic
similarity between input terms. It can be "Resnik" for
information content (IC) of most informative information
ancestor (MICA) (see
<a href = 'http://arxiv.org/pdf/cmp-lg/9511007.pdf'>http://arxiv.org/pdf/cmp-lg/9511007.pdf</a>), "Lin" for
2*IC at MICA divided by the sum of IC at pairs of terms
(see
<a href = 'http://webdocs.cs.ualberta.ca/~lindek/papers/sim.pdf'>http://webdocs.cs.ualberta.ca/~lindek/papers/sim.pdf</a>),
"Schlicker" for weighted version of 'Lin' by the
1-prob(MICA) (see
<a href = 'http://www.ncbi.nlm.nih.gov/pubmed/16776819'>http://www.ncbi.nlm.nih.gov/pubmed/16776819</a>),
"Jiang" for 1 - difference between the sum of IC at pairs
of terms and 2*IC at MICA (see
<a href = 'http://arxiv.org/pdf/cmp-lg/9709008.pdf'>http://arxiv.org/pdf/cmp-lg/9709008.pdf</a>),
"Pesquita" for graph information content similarity
related to Tanimoto-Jacard index (ie. summed information
content of common ancestors divided by summed information
content of all ancestors of term1 and term2 (see
<a href = 'http://www.ncbi.nlm.nih.gov/pubmed/18460186'>http://www.ncbi.nlm.nih.gov/pubmed/18460186</a>)). By
default, it uses "Schlicker" method</dd>
      <dt>fast</dt>
      <dd>logical to indicate whether a vectorised fast
computation is used. By default, it sets to true. It is
always advisable to use this vectorised fast computation;
since the conventional computation is just used for
understanding scripts</dd>
      <dt>parallel</dt>
      <dd>logical to indicate whether parallel
computation with multicores is used. By default, it sets
to true, but not necessarily does so. Partly because
parallel backends available will be system-specific (now
only Linux or Mac OS). Also, it will depend on whether
these two packages "foreach" and "doMC" have been
installed. It can be installed via:
<code>source("http://bioconductor.org/biocLite.R");
biocLite(c("foreach","doMC"))</code>. If not yet installed,
this option will be disabled</dd>
      <dt>multicores</dt>
      <dd>an integer to specify how many cores
will be registered as the multicore parallel backend to
the 'foreach' package. If NULL, it will use a half of
cores available in a user's computer. This option only
works when parallel computation is enabled</dd>
      <dt>verbose</dt>
      <dd>logical to indicate whether the messages
will be displayed in the screen. By default, it sets to
true for display</dd>
    </dl>
    <!-- 
    -->
    
    <div class="Value">
      <h3>Value</h3>
      
      <p>It returns a sparse matrix containing pair-wise semantic
similarity between input terms</p>
    </div>
    <div class="Note">
      <h3>Note</h3>
      
      <p>none</p>
    </div>
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'># 1) load HPPA as igraph object
data(ig.HPPA)
g <- ig.HPPA

# 2) load human genes annotated by HPPA
data(org.Hs.egHPPA)

# 3) prepare for ontology and its annotation information
dag <- dDAGannotate(g, annotations=org.Hs.egHPPA,
path.mode="all_paths", verbose=TRUE)
</div>
<strong class='message'>	At level 14, there are 5 nodes, and 12 incoming neighbors.</strong>
<strong class='message'>	At level 13, there are 14 nodes, and 21 incoming neighbors.</strong>
<strong class='message'>	At level 12, there are 30 nodes, and 38 incoming neighbors.</strong>
<strong class='message'>	At level 11, there are 80 nodes, and 79 incoming neighbors.</strong>
<strong class='message'>	At level 10, there are 162 nodes, and 144 incoming neighbors.</strong>
<strong class='message'>	At level 9, there are 296 nodes, and 222 incoming neighbors.</strong>
<strong class='message'>	At level 8, there are 475 nodes, and 325 incoming neighbors.</strong>
<strong class='message'>	At level 7, there are 560 nodes, and 329 incoming neighbors.</strong>
<strong class='message'>	At level 6, there are 617 nodes, and 320 incoming neighbors.</strong>
<strong class='message'>	At level 5, there are 522 nodes, and 205 incoming neighbors.</strong>
<strong class='message'>	At level 4, there are 280 nodes, and 86 incoming neighbors.</strong>
<strong class='message'>	At level 3, there are 94 nodes, and 20 incoming neighbors.</strong>
<strong class='message'>	At level 2, there are 20 nodes, and 1 incoming neighbors.</strong>
<strong class='message'>	At level 1, there are 1 nodes, and 0 incoming neighbors.</strong>
<div class='input'>
# 4) calculate pair-wise semantic similarity between 5 randomly chosen terms
terms <- sample(V(dag)$name, 5)
sim <- dDAGtermSim(g=dag, terms=terms, method="Schlicker",
parallel=FALSE)
</div>
<strong class='message'>Start at 2014-05-17 18:18:05</strong>
<strong class='message'></strong>
<strong class='message'>Calculate semantic similarity between 5 terms using Schlicker method (2014-05-17 18:18:05)...</strong>
<strong class='message'>Build a sparse matrix of children x ancestors (with 5 rows and 3156 columns (2014-05-17 18:18:05)...</strong>
<strong class='message'>	1 out of 5 (2014-05-17 18:18:05)</strong>
<strong class='message'>	2 out of 5 (2014-05-17 18:18:05)</strong>
<strong class='message'>	3 out of 5 (2014-05-17 18:18:05)</strong>
<strong class='message'>	4 out of 5 (2014-05-17 18:18:05)</strong>
<strong class='message'>	5 out of 5 (2014-05-17 18:18:06)</strong>
<strong class='message'></strong>
<strong class='message'>Finish at 2014-05-17 18:18:06</strong>
<strong class='message'>Runtime in total is: 1 secs
</strong>
<div class='input'>sim
</div>
<div class='output'>5 x 5 sparse Matrix of class "dgCMatrix"
           HP:0011922 HP:0009465 HP:0100273 HP:0008002 HP:0000481
HP:0011922   1.991753   .          .         .          .        
HP:0009465   .          1.983505   .         .          .        
HP:0100273   .          .          1.983505  .          .        
HP:0008002   .          .          .         1.9924399  0.1560963
HP:0000481   .          .          .         0.1560963  1.7422680
</div></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>dDAGtermSim</li>
    </ul>
    <ul>
      
    </ul> 
    -->
      
    <h2>See also</h2>
    
<code><a href='dDAGinduce.html'>dDAGinduce</a></code>, <code><a href='dDAGancestor.html'>dDAGancestor</a></code>,
<code><a href='dDAGgeneSim.html'>dDAGgeneSim</a></code>

    
    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) dnet: an open-source R package for integrative analysis of digitised data in terms of network, ontology and evolution. <a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></p>

    
    <!--
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>