<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>dDAGtermSim @ dnet 1.0.7</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 1.0.7</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="rdata.html"><i class="icon-file icon-white"></i> RData</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="faqs.html"><i class="icon-question-sign icon-white"></i> FAQs</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dnet R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:djnz9oat1hu';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to calculate pair-wise semantic similarity between input terms based on a direct acyclic graph (DAG) with annotated data</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>dDAGtermSim</code> is supposed to calculate pair-wise semantic
similarity between input terms based on a direct acyclic graph (DAG)
with annotated data. Parallel computing is also supported for Linux or
Mac operating systems.</p>
  
    <h3>Usage</h3>
    <pre>dDAGtermSim(g, terms = NULL, method = c("Resnik", "Lin", "Schlicker", "Jiang", "Pesquita"), 
  fast = T, parallel = TRUE, multicores = NULL, verbose = T)</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>g</dt>
      <dd>an object of class "igraph" or "graphNEL". It must contain a
vertex attribute called 'annotations' for storing annotation data (see
example for howto)</dd>
      <dt>terms</dt>
      <dd>the terms/nodes between which pair-wise semantic
similarity is calculated. If NULL, all terms in the input DAG will be
used for calcluation, which is very prohibitively expensive!</dd>
      <dt>method</dt>
      <dd>the method used to measure semantic similarity between
input terms. It can be "Resnik" for information content (IC) of most
informative common ancestor (MICA) (see
<a href = 'http://arxiv.org/pdf/cmp-lg/9511007.pdf'>http://arxiv.org/pdf/cmp-lg/9511007.pdf</a>), "Lin" for 2*IC at MICA
divided by the sum of IC at pairs of terms (see
<a href = 'http://webdocs.cs.ualberta.ca/~lindek/papers/sim.pdf'>http://webdocs.cs.ualberta.ca/~lindek/papers/sim.pdf</a>),
"Schlicker" for weighted version of 'Lin' by the 1-prob(MICA) (see
<a href = 'http://www.ncbi.nlm.nih.gov/pubmed/16776819'>http://www.ncbi.nlm.nih.gov/pubmed/16776819</a>), "Jiang" for 1 -
difference between the sum of IC at pairs of terms and 2*IC at MICA
(see <a href = 'http://arxiv.org/pdf/cmp-lg/9709008.pdf'>http://arxiv.org/pdf/cmp-lg/9709008.pdf</a>), "Pesquita" for
graph information content similarity related to Tanimoto-Jacard index
(ie. summed information content of common ancestors divided by summed
information content of all ancestors of term1 and term2 (see
<a href = 'http://www.ncbi.nlm.nih.gov/pubmed/18460186'>http://www.ncbi.nlm.nih.gov/pubmed/18460186</a>)). By default, it
uses "Schlicker" method</dd>
      <dt>fast</dt>
      <dd>logical to indicate whether a vectorised fast computation
is used. By default, it sets to true. It is always advisable to use
this vectorised fast computation; since the conventional computation is
just used for understanding scripts</dd>
      <dt>parallel</dt>
      <dd>logical to indicate whether parallel computation with
multicores is used. By default, it sets to true, but not necessarily
does so. Partly because parallel backends available will be
system-specific (now only Linux or Mac OS). Also, it will depend on
whether these two packages "foreach" and "doMC" have been installed. It
can be installed via:
<code>source("http://bioconductor.org/biocLite.R");
biocLite(c("foreach","doMC"))</code>. If not yet installed, this option will
be disabled</dd>
      <dt>multicores</dt>
      <dd>an integer to specify how many cores will be
registered as the multicore parallel backend to the 'foreach' package.
If NULL, it will use a half of cores available in a user's computer.
This option only works when parallel computation is enabled</dd>
      <dt>verbose</dt>
      <dd>logical to indicate whether the messages will be
displayed in the screen. By default, it sets to true for display</dd>
    </dl>
    <!-- 
    -->
    
    <!-- 
    <div class="Value">
    -->
      <h3>Value</h3>
      
      <p>It returns a sparse matrix containing pair-wise semantic similarity
between input terms. This sparse matrix can be converted to the full
matrix via the function <code>as.matrix</code></p>
    <!-- 
    </div>
    -->
    <!-- 
    <div class="Note">
    -->
      <h3>Note</h3>
      
      <p>none</p>
    <!-- 
    </div>
    -->
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'># 1) load HPPA as igraph object
ig.HPPA <-dRDataLoader(RData='ig.HPPA')
</div>
<strong class='message'>'ig.HPPA' (from package 'dnet' version 1.0.7) has been loaded into the working environment</strong>
<div class='input'>g <- ig.HPPA

# 2) load human genes annotated by HPPA
org.Hs.egHPPA <- dRDataLoader(RData='org.Hs.egHPPA')
</div>
<strong class='message'>'org.Hs.egHPPA' (from package 'dnet' version 1.0.7) has been loaded into the working environment</strong>
<div class='input'>
# 3) prepare for ontology and its annotation information
dag <- dDAGannotate(g, annotations=org.Hs.egHPPA,
path.mode="all_paths", verbose=TRUE)
</div>
<strong class='message'>	At level 13, there are 5 nodes, and 12 incoming neighbors.</strong>
<strong class='message'>	At level 12, there are 17 nodes, and 27 incoming neighbors.</strong>
<strong class='message'>	At level 11, there are 47 nodes, and 63 incoming neighbors.</strong>
<strong class='message'>	At level 10, there are 141 nodes, and 142 incoming neighbors.</strong>
<strong class='message'>	At level 9, there are 322 nodes, and 284 incoming neighbors.</strong>
<strong class='message'>	At level 8, there are 515 nodes, and 361 incoming neighbors.</strong>
<strong class='message'>	At level 7, there are 616 nodes, and 384 incoming neighbors.</strong>
<strong class='message'>	At level 6, there are 704 nodes, and 383 incoming neighbors.</strong>
<strong class='message'>	At level 5, there are 580 nodes, and 228 incoming neighbors.</strong>
<strong class='message'>	At level 4, there are 294 nodes, and 89 incoming neighbors.</strong>
<strong class='message'>	At level 3, there are 102 nodes, and 23 incoming neighbors.</strong>
<strong class='message'>	At level 2, there are 23 nodes, and 1 incoming neighbors.</strong>
<strong class='message'>	At level 1, there are 1 nodes, and 0 incoming neighbors.</strong>
<div class='input'>
# 4) calculate pair-wise semantic similarity between 5 randomly chosen terms
terms <- sample(V(dag)$name, 5)
sim <- dDAGtermSim(g=dag, terms=terms, method="Schlicker",
parallel=FALSE)
</div>
<strong class='message'>Start at 2015-07-21 17:55:32</strong>
<strong class='message'></strong>
<strong class='message'>Calculate semantic similarity between 5 terms using Schlicker method (2015-07-21 17:55:32)...</strong>
<strong class='message'>Build a sparse matrix of children x ancestors (with 5 rows and 3367 columns (2015-07-21 17:55:32)...</strong>
<strong class='message'>	1 out of 5 (2015-07-21 17:55:32)</strong>
<strong class='message'>	2 out of 5 (2015-07-21 17:55:32)</strong>
<strong class='message'>	3 out of 5 (2015-07-21 17:55:32)</strong>
<strong class='message'>	4 out of 5 (2015-07-21 17:55:32)</strong>
<strong class='message'>	5 out of 5 (2015-07-21 17:55:32)</strong>
<strong class='message'></strong>
<strong class='message'>Finish at 2015-07-21 17:55:32</strong>
<strong class='message'>Runtime in total is: 0 secs
</strong>
<div class='input'>sim
</div>
<div class='output'>5 x 5 sparse Matrix of class "dgCMatrix"
           HP:0030061 HP:0100730 HP:0100650 HP:0012093 HP:0000921
HP:0030061  1.9763578   .         0.2780308   .          .       
HP:0100730  .           1.996166  .           .          .       
HP:0100650  0.2780308   .         1.9968051   .          .       
HP:0012093  .           .         .           1.991054   .       
HP:0000921  .           .         .           .          1.990415
</div></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>dDAGtermSim</li>
    </ul>
    <ul>
      
    </ul> 
    -->
    
    <h2>Source code</h2>
    <a href="javascript:newWin('R/dDAGtermSim.r', 'R/dDAGtermSim.r', '1200', '1200')" title="Click to view"><B><code>dDAGtermSim.r</code></B></a>
    
    <h2>Source man</h2>
    <a href="javascript:newWin('man/dDAGtermSim.txt', 'man/dDAGtermSim.txt', '1200', '1200')" title="Click to view"><B><code>dDAGtermSim.Rd</code></B></a>
    
    <a href="javascript:newWin('man/dDAGtermSim.pdf', 'man/dDAGtermSim.pdf', '1200', '1200')" title="Click to view"><B><code>dDAGtermSim.pdf</code></B></a>
    
    <h2>See also</h2>
    
<code><a href='dDAGinduce.html'>dDAGinduce</a></code>, <code><a href='dDAGancestor.html'>dDAGancestor</a></code>,
<code><a href='dDAGgeneSim.html'>dDAGgeneSim</a></code>, <code><a href='dCheckParallel.html'>dCheckParallel</a></code>

    <!--

    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) <B>The &#39;dnet&#39; approach promotes emerging research on cancer patient survival</B>. <I>Genome Medicine</I>, 6(8):64. <a href="http://dx.doi.org/10.1186/s13073-014-0064-8">doi:10.1186/s13073-014-0064-8</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/25246945">25246945</a></p>

<p>You may also cite the package as:</p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dnet&rdquo; [<a href="http://cran.r-project.org/package=dnet">http://cran.r-project.org/package=dnet</a>]</p>

    
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dnet">http://supfam.org/dnet</a></li>
      <li><a href=" http://dnet.r-forge.r-project.org"> http://dnet.r-forge.r-project.org</a></li>
      <li><a href=" http://cran.r-
project.org/package=dnet"> http://cran.r-
project.org/package=dnet</a></li>
      <li><a href=" https://github.com/hfang-bristol/dnet"> https://github.com/hfang-bristol/dnet</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@well.ox.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>