<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>dDAGgeneSim @ dnet 1.0.2</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 1.0.2</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="rdata.html"><i class="icon-file icon-white"></i> RData</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to calculate pair-wise semantic similarity between genes based on a direct acyclic graph (DAG) with annotated data</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>dDAGgeneSim</code> is supposed to calculate pair-wise
semantic similarity between genes based on a direct acyclic
graph (DAG) with annotated data.</p>
  
    <h3>Usage</h3>
    <pre>dDAGgeneSim(g, genes = NULL, method.gene = c("BM.average", "BM.max", "BM.complete", 
  "average", "max"), method.term = c("Resnik", "Lin", "Schlicker", "Jiang", "Pesquita"), 
      force = TRUE, fast = TRUE, verbose = TRUE)</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>g</dt>
      <dd>an object of class "igraph" or "graphNEL"</dd>
      <dt>genes</dt>
      <dd>the genes between which pair-wise semantic
similarity is calculated. If NULL, all genes annotatable
in the input dag will be used for calcluation, which is
very prohibitively expensive!</dd>
      <dt>method.gene</dt>
      <dd>the method used for how to derive
semantic similarity between genes from semantic
similarity between terms. It can be "average" for average
similarity between any two terms (one from gene 1, the
other from gene 2), "max" for the maximum similarity
between any two terms, "BM.average" for best-matching
(BM) based average similarity (i.e. for each term of
either gene, first calculate maximum similarity to any
term in the other gene, then take average of maximum
similarity; the final BM-based average similiary is the
pre-calculated average between two genes in pair),
"BM.max" for BM based maximum similarity (i.e. the same
as "BM.average", but the final BM-based maximum similiary
is the maximum of the pre-calculated average between two
genes in pair), "BM.complete" for BM-based
complete-linkage similarity (inspired by complete-linkage
concept: the least of any maximum similarity between a
term of one gene and a term of the other gene). When
comparing BM-based similarity between genes, "BM.average"
and "BM.max" are sensitive to the number of terms
invovled; instead, "BM.complete" is much robust in this
aspect. By default, it uses "BM.average".</dd>
      <dt>method.term</dt>
      <dd>the method used to measure semantic
similarity between terms. It can be "Resnik" for
information content (IC) of most informative information
ancestor (MICA) (see
<a href = 'http://arxiv.org/pdf/cmp-lg/9511007.pdf'>http://arxiv.org/pdf/cmp-lg/9511007.pdf</a>), "Lin" for
2*IC at MICA divided by the sum of IC at pairs of terms
(see
<a href = 'http://webdocs.cs.ualberta.ca/~lindek/papers/sim.pdf'>http://webdocs.cs.ualberta.ca/~lindek/papers/sim.pdf</a>),
"Schlicker" for weighted version of 'Lin' by the
1-prob(MICA) (see
<a href = 'http://www.ncbi.nlm.nih.gov/pubmed/16776819'>http://www.ncbi.nlm.nih.gov/pubmed/16776819</a>),
"Jiang" for 1 - difference between the sum of IC at pairs
of terms and 2*IC at MICA (see
<a href = 'http://arxiv.org/pdf/cmp-lg/9709008.pdf'>http://arxiv.org/pdf/cmp-lg/9709008.pdf</a>),
"Pesquita" for graph information content similarity
related to Tanimoto-Jacard index (ie. summed information
content of common ancestors divided by summed information
content of all ancestors of term1 and term2 (see
<a href = 'http://www.ncbi.nlm.nih.gov/pubmed/18460186'>http://www.ncbi.nlm.nih.gov/pubmed/18460186</a>))</dd>
      <dt>force</dt>
      <dd>logical to indicate whether the only most
specific terms (for each gene) will be used. By default,
it sets to true. It is always advisable to use this since
it is computationally fast but without compromising
accuracy (considering the fact that true-path-rule has
been applied when running <code><a href='dDAGannotate.html'>dDAGannotate</a></code>)</dd>
      <dt>fast</dt>
      <dd>logical to indicate whether a vectorised fast
computation is used. By default, it sets to true. It is
always advisable to use this vectorised fast computation;
since the conventional computation is just used for
understanding scripts</dd>
      <dt>verbose</dt>
      <dd>logical to indicate whether the messages
will be displayed in the screen. By default, it sets to
true for display</dd>
    </dl>
    <!-- 
    -->
    
    <div class="Value">
      <h3>Value</h3>
      
      <p>It returns a sparse matrix containing pair-wise semantic
similarity between input terms</p>
    </div>
    <div class="Note">
      <h3>Note</h3>
      
      <p>For the mode "shortest_paths", the induced subgraph is the
most concise, and thus informative for visualisation when
there are many nodes in query, while the mode "all_paths"
results in the complete subgraph.</p>
    </div>
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'># 1) load HPPA as igraph object
data(ig.HPPA)
g <- ig.HPPA

# 2) load human genes annotated by HPPA
data(org.Hs.egHPPA)

# 3) prepare for ontology and its annotation information
dag <- dDAGannotate(g, annotations=org.Hs.egHPPA,
path.mode="all_paths", verbose=TRUE)
</div>
<strong class='message'>	At level 14, there are 5 nodes, and 12 incoming neighbors.</strong>
<strong class='message'>	At level 13, there are 14 nodes, and 21 incoming neighbors.</strong>
<strong class='message'>	At level 12, there are 30 nodes, and 38 incoming neighbors.</strong>
<strong class='message'>	At level 11, there are 80 nodes, and 79 incoming neighbors.</strong>
<strong class='message'>	At level 10, there are 162 nodes, and 144 incoming neighbors.</strong>
<strong class='message'>	At level 9, there are 296 nodes, and 222 incoming neighbors.</strong>
<strong class='message'>	At level 8, there are 475 nodes, and 325 incoming neighbors.</strong>
<strong class='message'>	At level 7, there are 560 nodes, and 329 incoming neighbors.</strong>
<strong class='message'>	At level 6, there are 617 nodes, and 320 incoming neighbors.</strong>
<strong class='message'>	At level 5, there are 522 nodes, and 205 incoming neighbors.</strong>
<strong class='message'>	At level 4, there are 280 nodes, and 86 incoming neighbors.</strong>
<strong class='message'>	At level 3, there are 94 nodes, and 20 incoming neighbors.</strong>
<strong class='message'>	At level 2, there are 20 nodes, and 1 incoming neighbors.</strong>
<strong class='message'>	At level 1, there are 1 nodes, and 0 incoming neighbors.</strong>
<div class='input'>
# 4) calculate pair-wise semantic similarity between 5 randomly chosen genes
allgenes <- unique(unlist(V(dag)$annotations))
genes <- sample(allgenes,5)
sim <- dDAGgeneSim(g=dag, genes=genes, method.gene="BM.average",
method.term="Resnik", verbose=TRUE)
</div>
<strong class='message'>Start at 2014-05-13 14:40:44</strong>
<strong class='message'></strong>
<strong class='message'>First, extract all annotatable genes (2014-05-13 14:40:44)...</strong>
<strong class='message'>	there are 5 input genes amongst 2910 annotatable genes</strong>
<strong class='message'>Second, pre-compute semantic similarity between 91 terms (forced to be the most specific for each gene) using Resnik method (2014-05-13 14:40:44)...</strong>
<strong class='message'>Last, calculate pair-wise semantic similarity between 5 genes using BM.average method (2014-05-13 14:40:45)...</strong>
<strong class='message'>	1 out of 5 (2014-05-13 14:40:45)</strong>
<strong class='message'>	2 out of 5 (2014-05-13 14:40:45)</strong>
<strong class='message'>	3 out of 5 (2014-05-13 14:40:45)</strong>
<strong class='message'>	4 out of 5 (2014-05-13 14:40:45)</strong>
<strong class='message'></strong>
<strong class='message'>Finish at 2014-05-13 14:40:45</strong>
<strong class='message'>Runtime in total is: 1 secs
</strong>
<div class='input'>sim
</div>
<div class='output'>5 x 5 sparse Matrix of class "dgCMatrix"
             7299      7399       4750      7153     115908
7299   .          0.6052493 0.09829316 0.4255154 0.09952181
7399   0.60524933 .         0.24407421 0.1103988 0.14851424
4750   0.09829316 0.2440742 .          0.1018560 0.27722327
7153   0.42551537 0.1103988 0.10185604 .         0.46050377
115908 0.09952181 0.1485142 0.27722327 0.4605038 .         
</div></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>dDAGgeneSim</li>
    </ul>
    <ul>
      
    </ul> 
    -->
      
    <h2>See also</h2>
    
<code><a href='dDAGtermSim.html'>dDAGtermSim</a></code>, <code><a href='dDAGinduce.html'>dDAGinduce</a></code>,
<code><a href='dDAGtip.html'>dDAGtip</a></code>

    
    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) dnet: an open-source R package for integrative analysis of digitised data in terms of network, ontology and evolution. <a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></p>

    
    <!--
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>