<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>dDAGannotate @ dnet 1.0.0</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 1.0.0</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="rdata.html"><i class="icon-file icon-white"></i> RData</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to generate a subgraph of a direct acyclic graph (DAG) induced by the input annotation data</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>dDAGannotate</code> is supposed to produce a subgraph
induced by the input annotation data, given a direct
acyclic graph (DAG; an ontology). The input is a graph of
"igraph" or "graphNET" object, a list of the vertices
containing annotation data, and the mode defining the paths
to the root of DAG. The induced subgraph contains vertices
(with annotation data) and their ancestors along with the
defined paths to the root of DAG. The annotations at these
vertices (including their ancestors) are also updated
according to the true-path rule: a gene annotated to a term
should also be annotated by its all ancestor terms.</p>
  
    <h3>Usage</h3>
    <pre>dDAGannotate(g, annotations, path.mode = c("all_paths", "shortest_paths", "all_shortest_paths"), 
  verbose = TRUE)</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>g</dt>
      <dd>an object of class "igraph" or "graphNEL"</dd>
      <dt>annotations</dt>
      <dd>the vertices/nodes for which
annotation data are provided</dd>
      <dt>path.mode</dt>
      <dd>the mode of paths induced by
vertices/nodes with input annotation data. It can be
"all_paths" for all possible paths to the root,
"shortest_paths" for only one path to the root (for each
node in query), "all_shortest_paths" for all shortest
paths to the root (i.e. for each node, find all shortest
paths with the equal lengths)</dd>
      <dt>verbose</dt>
      <dd>logical to indicate whether the messages
will be displayed in the screen. By default, it sets to
true for display</dd>
    </dl>
    <!-- 
    -->
    
    <div class="Value">
      <h3>Value</h3>
      
      <p><ul>
<li><code>subg</code>: an induced subgraph, an object
of class "igraph". In addition to the original attributes
to nodes and edges, the return subgraph is also appended a
new node attribute called "annotations", which contains a
list of genes either as original annotations or inherited
annotations </li>
</ul></p>
      <p></p>
    </div>
    <div class="Note">
      <h3>Note</h3>
      
      <p>For the mode "shortest_paths", the induced subgraph is the
most concise, and thus informative for visualisation when
there are many nodes in query, while the mode "all_paths"
results in the complete subgraph.</p>
    </div>
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'># 1) load GO Molelular Function as igraph object
load(url("http://dnet.r-forge.r-project.org/data/Obo/ig.GOMF.RData"))
g <- ig.GOMF

# 2) load human genes annotated by GO Molelular Function terms
load(url("http://dnet.r-forge.r-project.org/data/Hs/org.Hs.egGOMF.RData"))
GS <- org.Hs.egGOMF # as 'GS' object

# 3) prepare for annotation data
# randomly select vertices with annotation data
annotations <- GS$gs[sample(1:length(GS$gs),5)]

# 4) obtain the induced subgraph
# 4a) based on all possible paths (i.e. the complete subgraph induced)
dDAGannotate(g, annotations, path.mode="all_paths", verbose=TRUE)
</div>
<strong class='message'>	At level 7, there are 3 nodes, and 3 incoming neighbors.</strong>
<strong class='message'>	At level 6, there are 3 nodes, and 3 incoming neighbors.</strong>
<strong class='message'>	At level 5, there are 4 nodes, and 4 incoming neighbors.</strong>
<strong class='message'>	At level 4, there are 5 nodes, and 3 incoming neighbors.</strong>
<strong class='message'>	At level 3, there are 3 nodes, and 3 incoming neighbors.</strong>
<strong class='message'>	At level 2, there are 3 nodes, and 1 incoming neighbors.</strong>
<strong class='message'>	At level 1, there are 1 nodes, and 0 incoming neighbors.</strong>
<div class='output'>IGRAPH DN-- 22 21 -- 
+ attr: name (v/c), term_id (v/c), term_name (v/c), term_distance
  (v/n), annotations (v/x), relation (e/c)
</div>
<div class='input'># 4b) based on shortest paths (i.e. the most concise subgraph induced)
dag <- dDAGannotate(g, annotations, path.mode="shortest_paths",
verbose=TRUE)
</div>
<strong class='message'>	At level 7, there are 3 nodes, and 3 incoming neighbors.</strong>
<strong class='message'>	At level 6, there are 3 nodes, and 3 incoming neighbors.</strong>
<strong class='message'>	At level 5, there are 4 nodes, and 4 incoming neighbors.</strong>
<strong class='message'>	At level 4, there are 5 nodes, and 3 incoming neighbors.</strong>
<strong class='message'>	At level 3, there are 3 nodes, and 3 incoming neighbors.</strong>
<strong class='message'>	At level 2, there are 3 nodes, and 1 incoming neighbors.</strong>
<strong class='message'>	At level 1, there are 1 nodes, and 0 incoming neighbors.</strong>
<div class='input'>
# 5) color-code nodes/terms according to the number of annotations
data <- sapply(V(dag)$annotations, length)
names(data) <- V(dag)$name
visDAG(g=dag, data=data, node.info="both")
</div>
<strong class='message'>
Attaching package: ‘graph’
</strong>
<strong class='message'>The following objects are masked _by_ ‘package:igraph’:

    degree, edges
</strong>
<a href="javascript:newWin('dDAGannotate-21.png', 'dDAGannotate-21.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='dDAGannotate-21.png', width="400" /></a></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>dDAGannotate</li>
    </ul>
    <ul>
      
    </ul> 
    -->
      
    <h2>See also</h2>
    
<code><a href='dDAGinduce.html'>dDAGinduce</a></code>, <code><a href='dDAGlevel.html'>dDAGlevel</a></code>

    
    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) DNET: dynamic networks via integrative analysis of digitised data in terms of network, ontology and evolution. R package version 1.0.0. <a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></p>

    
    <!--
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>