<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>Demo for ALL dataset @ dnet 0.99.0</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="Hai Fang and Julian Gough">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 0.99.0</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <!--
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
        -->
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Demo for ALL dataset</h2>

Notes:
<li> All results are based on dnet (version 0.99.0). </li>
<li> Images displayed below may be distorted, and they should be normal in your screen. </li>
<li> Functions contained in dnet 0.99.0 are hyperlinked in-place and also listed on the right side. </li>

<div class="row">
    <div class="span8">
      <pre>
      <div class='input'># This is a demo for human embryo dataset from Fang et al
# 
# This human embryo expression dataset (available from <a href="http://www.ncbi.nlm.nih.gov/pubmed/20643359" target="20643359">http://www.ncbi.nlm.nih.gov/pubmed/20643359</a>) involves six successive developmental stages (S9-S14) with three replicates (R1-R3) for each stage, including:
## Fang: an expression matrix of 5,441 genes X 18 samples;
## Fang.geneinfo: a matrix of 5,441 X 3 containing gene information;
## Fang.sampleinfo: a matrix of 18 X 3 containing sample information.
###############################################################################
library(BioNet)
</div>
<strong class='message'>Loading required package: Biobase</strong>
<strong class='message'>Loading required package: BiocGenerics</strong>
<strong class='message'>Loading required package: parallel</strong>
<strong class='message'>
Attaching package: ‘BiocGenerics’
</strong>
<strong class='message'>The following objects are masked from ‘package:parallel’:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB
</strong>
<strong class='message'>The following object is masked from ‘package:stats’:

    xtabs
</strong>
<strong class='message'>The following objects are masked from ‘package:base’:

    anyDuplicated, append, as.data.frame, as.vector, cbind, colnames,
    duplicated, eval, evalq, Filter, Find, get, intersect, is.unsorted,
    lapply, Map, mapply, match, mget, order, paste, pmax, pmax.int,
    pmin, pmin.int, Position, rank, rbind, Reduce, rep.int, rownames,
    sapply, setdiff, sort, table, tapply, union, unique, unlist
</strong>
<strong class='message'>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.
</strong>
<strong class='message'>Loading required package: graph</strong>
<strong class='message'>
Attaching package: ‘graph’
</strong>
<strong class='message'>The following objects are masked from ‘package:igraph’:

    degree, edges
</strong>
<strong class='message'>Loading required package: RBGL</strong>
<strong class='message'>
Attaching package: ‘RBGL’
</strong>
<strong class='message'>The following object is masked from ‘package:igraph’:

    transitivity
</strong>
<div class='input'>library(ALL)
data(ALL)
library(DLBCL)
data(interactome)

# Since several probesets represent one gene, we have to select one or concatenate them into one gene. One possibility is to use the probeset with the highest variance for each gene. It also maps the Affymetrix IDs to the identifiers of the network using the chip annotations and network geneIDs, which are unique, and returns the network names in the expression matrix. This reduces the expression matrix to the genes which are present in the network.
mapped.eset <- mapByVar(ALL, network=interactome, attr="geneID")
</div>
<strong class='message'>Loading required package: hgu95av2.db</strong>
<strong class='message'>Loading required package: AnnotationDbi</strong>
<strong class='message'>Loading required package: org.Hs.eg.db</strong>
<strong class='message'>Loading required package: DBI</strong>
<strong class='message'></strong>
<strong class='message'></strong>
<div class='input'>mapped.eset[1:5,1:5]
</div>
<div class='output'>                 01005     01010    03002    04006    04007
MAPK3(5595)   7.597323  7.479445 7.567593 7.384684 7.905312
TIE1(7075)    5.046194  4.932537 4.799294 4.922627 4.844565
CYP2C19(1557) 3.900466  4.208155 3.886169 4.206798 3.416923
BLR1(643)     5.903856  6.169024 5.860459 6.116890 5.687997
DUSP1(1843)   8.570990 10.428299 9.616713 9.937155 9.983809
</div>
<div class='input'>
# extract network that only contains genes in mapped.eset
network <- <a href='dNetInduce.html' target=dNetInduce>dNetInduce</a>(g=interactome, nodes_query=rownames(mapped.eset), knn=0, remove.loops=T, largest.comp=T)
g <- igraph.from.graphNEL(network)

# 1) preparation of node p-values
library(limma)
</div>
<strong class='message'>
Attaching package: ‘limma’
</strong>
<strong class='message'>The following object is masked from ‘package:BiocGenerics’:

    plotMA
</strong>
<div class='input'>design <- model.matrix(~ -1 + factor(c(substr(unlist(ALL$BT), 0, 1))))
colnames(design)<- c("B", "T")
contrast.matrix <- makeContrasts(B-T, levels=design)
contrast.matrix 
</div>
<div class='output'>      Contrasts
Levels B - T
     B     1
     T    -1
</div>
<div class='input'>fit <- lmFit(mapped.eset, design)
fit2 <- contrasts.fit(fit, contrast.matrix)
fit2 <- eBayes(fit2)
# get the corresponding p-values and and calculate the scores thereupon. 
pval <- fit2$p.value[,1]

# 2) identification of module
fdr_target <- 1e-14
module <- <a href='dNetPipeline.html' target=dNetPipeline>dNetPipeline</a>(g=g, pval=pval, fdr=fdr_target)
</div>
<strong class='message'>Start at 2014-03-01 20:37:02</strong>
<strong class='message'></strong>
<strong class='message'>First, fit the p-value distribution under beta-uniform mixture model...</strong>
<strong class='message'>	A total of p-values: 6190</strong>
<strong class='message'>	Maximum Log-Likelihood: 12068.4</strong>
<strong class='message'>	Mixture parameter (lambda): 0.453</strong>
<strong class='message'>	Shape parameter (a): 0.144</strong>
<strong class='message'>Second, determine FDR threshold...</strong>
<strong class='message'>	FDR threshold: 1.00e-14</strong>
<strong class='message'>Third, calculate the scores according to the fitted BUM and FDR threshold...</strong>
<strong class='message'>Finally, find the module with the maximum score...</strong>
<strong class='message'>	Size of the module: 30 nodes and 38 edges</strong>
<strong class='message'></strong>
<strong class='message'>Finish at 2014-03-01 20:37:04</strong>
<strong class='message'>Runtime in total is: 2 secs
</strong>
<div class='input'>
g <- module
glayout <- layout.fruchterman.reingold(g)

# 3) color nodes according to communities identified via a spin-glass model and simulated annealing
#com <- spinglass.community(g, spins=3)
com <- walktrap.community(g, modularity=T)
com$csize <- sapply(1:length(com),function(x) sum(com$membership==x))
vgroups <- com$membership
colormap <- "yellow-darkorange"
palette.name <- visColormap(colormap=colormap)
mcolors <- palette.name(length(com))
vcolors <- mcolors[vgroups]

com$significance <- sapply(1:length(com), function(x) {
    community.significance.test <- function(g, vids, ...) {
        subg <- induced.subgraph(g, vids)
        within.degrees <- igraph::degree(subg)
        cross.degrees <- igraph::degree(g, vids) - within.degrees
        wilcox.test(within.degrees, cross.degrees, ...)
    }
    tmp <- suppressWarnings(community.significance.test(g, vids=V(g)$name[com$membership==x]))
    signif(tmp$p.value, digits=3)
})

# 4) size nodes according to degrees
vdegrees <- igraph::degree(g)

# 5) sort nodes: first by communities and then degrees
tmp <- data.frame(ind=1:vcount(g), vgroups, vdegrees)
ordering <- tmp[order(vgroups,vdegrees),]$ind

# 6) visualise graph using 1-dimensional arc diagram
<a href='visNetArc.html' target=visNetArc>visNetArc</a>(g, ordering=ordering, labels=V(g)$geneSymbol, vertex.label.color=vcolors, vertex.color=vcolors, vertex.frame.color=vcolors, vertex.size=log(vdegrees)+0.1, vertex.label.cex=0.6)
</div>
<a href="javascript:newWin('ALL-47.png', 'ALL-47.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='ALL-47.png', width="400" /></a>
<div class='input'>
# 7) visualise graph using circle diagram
# 7a) drawn into a single circle 
<a href='visNetCircle.html' target=visNetCircle>visNetCircle</a>(g=g, com=com, ordering=ordering, colormap=colormap, vertex.label=V(g)$geneSymbol, vertex.size=igraph::degree(g)+5, vertex.label.color="black", vertex.label.cex=0.6, vertex.label.dist=0.75, vertex.shape="sphere", edge.color.within="grey", edge.color.crossing="black", edge.width=1, edge.lty=1, mark.shape=1, mark.expand=10)
</div>
<a href="javascript:newWin('ALL-49.png', 'ALL-49.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='ALL-49.png', width="400" /></a>
<div class='input'># 7b) drawn into multiple circles
<a href='visNetCircle.html' target=visNetCircle>visNetCircle</a>(g=g, com=com, circles="multiple", ordering=ordering, colormap=colormap, vertex.label=V(g)$geneSymbol, vertex.size=igraph::degree(g)+5, vertex.label.color="black", vertex.label.cex=0.6, vertex.label.dist=0.25, vertex.shape="sphere", edge.color.within="grey", edge.color.crossing="black", edge.width=1, edge.lty=1, mark.shape=1, mark.expand=10)
</div>
<a href="javascript:newWin('ALL-51.png', 'ALL-51.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='ALL-51.png', width="400" /></a>
<div class='input'>
# 8) as comparison, also visualise graph on 2-dimensional layout 
mark.groups <- communities(com)
mark.col <- visColoralpha(mcolors, alpha=0.2)
mark.border <- visColoralpha(mcolors, alpha=0.2)
edge.color <- c("grey", "black")[crossing(com,g)+1]
<a href='visNet.html' target=visNet>visNet</a>(g, glayout=glayout, vertex.label=V(g)$geneSymbol, vertex.color=vcolors, vertex.frame.color=vcolors, vertex.shape="sphere", mark.groups=mark.groups, mark.col=mark.col, mark.border=mark.border, mark.shape=1, mark.expand=10, edge.color=edge.color)
</div>
<a href="javascript:newWin('ALL-53.png', 'ALL-53.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='ALL-53.png', width="400" /></a>
<div class='input'>
legend_name <- paste("C",1:length(mcolors)," (n=",com$csize,", pval=",signif(com$significance,digits=2),")",sep='')
legend("bottomleft", legend=legend_name, fill=mcolors, bty="n", cex=0.7)
</div>
<a href="javascript:newWin('ALL-55.png', 'ALL-55.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='ALL-55.png', width="400" /></a>
<div class='input'>
# 9) color by score and FC
logFC <- fit2$coefficients[V(g)$name,1]
par(mfrow=c(1,2), mar=c(0.5,0.5,0.5,0.5))
<a href='visNet.html' target=visNet>visNet</a>(g, glayout=glayout, pattern=V(module)$score, newpage=F, colorbar=F, mark.groups=mark.groups, mark.col=mark.col, mark.border=mark.border, mark.shape=1, mark.expand=10, edge.color=edge.color)
<a href='visNet.html' target=visNet>visNet</a>(g, glayout=glayout, pattern=logFC, newpage=F, colorbar=F, mark.groups=mark.groups, mark.col=mark.col, mark.border=mark.border, mark.shape=1, mark.expand=10, edge.color=edge.color)
</div>
<a href="javascript:newWin('ALL-57.png', 'ALL-57.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='ALL-57.png', width="400" /></a>
<div class='input'>
# 10) color by additional data
tmp <- substr(unlist(ALL$BT), 0, 2)
ind <- union(which(tmp=="B4"), which(tmp=="T3"))
data <- mapped.eset[V(g)$name,ind]
colnames(data) <- tmp[ind]
<a href='visNetMul.html' target=visNetMul>visNetMul</a>(g=g, data=data, height=ceiling(sqrt(ncol(data)))*2, newpage=T,glayout=glayout,colormap="darkgreen-lightgreen-lightpink-darkred",vertex.label=NA,vertex.shape="sphere", vertex.size=18,mtext.cex=0.8,border.color="888888", mark.groups=mark.groups, mark.col=mark.col, mark.border=mark.border, mark.shape=1, mark.expand=10, edge.color=edge.color)
</div>
<a href="javascript:newWin('ALL-59.png', 'ALL-59.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='ALL-59.png', width="400" /></a>
<div class='input'>
# 11) color by additional data (be reordered)
sReorder <- <a href='dNetReorder.html' target=dNetReorder>dNetReorder</a>(g, data, feature="edge", node.normalise="degree", amplifier=2, metric="none")
</div>
<strong class='message'>Start at 2014-03-01 20:38:07</strong>
<strong class='message'></strong>
<strong class='message'>First, define topology of a map grid...</strong>
<strong class='message'>Second, initialise the codebook matrix given a topology and input data...</strong>
<strong class='message'>Third, get training at the rough stage...</strong>
<strong class='message'>Fourth, get training at the finetune stage...</strong>
<strong class='message'>Next, identify the best-matching hexagon/rectangle for the input data...</strong>
<strong class='message'>Finally, append the response data (hits and mqe) into the sMap object...</strong>
<strong class='message'></strong>
<strong class='message'>Below are the summaries of the training results:</strong>
<strong class='message'>   dimension of input data: 22x38
   xy-dimension of map grid: xdim=7, ydim=7
   grid lattice: rect
   grid shape: sheet
   dimension of grid coord: 49x2
   initialisation method: linear
   dimension of codebook matrix: 49x38
   mean quantization error: 3.3268215319217
</strong>
<strong class='message'>Below are the details of trainology:</strong>
<strong class='message'>   training algorithm: sequential
   alpha type: invert
   training neighborhood kernel: gaussian
   trainlength (x input data length): 23 at rough stage; 90 at finetune stage
   radius (at rough stage): from 1 to 1
   radius (at finetune stage): from 1 to 1
</strong>
<strong class='message'>End at 2014-03-01 20:38:09</strong>
<strong class='message'>Runtime in total is: 2 secs
</strong>
<div class='input'><a href='visNetReorder.html' target=visNetReorder>visNetReorder</a>(g=g, data=data, sReorder=sReorder, height=ceiling(sqrt(ncol(data)))*2, newpage=T, glayout=glayout, colormap="darkgreen-lightgreen-lightpink-darkred", vertex.label=NA,vertex.shape="sphere", vertex.size=18,mtext.cex=0.8,border.color="888888", mark.groups=mark.groups, mark.col=mark.col, mark.border=NA, mark.shape=1, mark.expand=10, edge.color=edge.color)</div>
<a href="javascript:newWin('ALL-77.png', 'ALL-77.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='ALL-77.png', width="400" /></a>
      </pre>
    </div>

  <div class="span4">
    
    <h2>Functions used in this demo</h2>
    <ul>
      <li><a href="dNetInduce.html">dNetInduce</a></li>
      <li><a href="dNetPipeline.html">dNetPipeline</a></li>
      <li><a href="visNetArc.html">visNetArc</a></li>
      <li><a href="visNetCircle.html">visNetCircle</a></li>
      <li><a href="visNet.html">visNet</a></li>
      <li><a href="visNetMul.html">visNetMul</a></li>
      <li><a href="dNetReorder.html">dNetReorder</a></li>
      <li><a href="visNetReorder.html">visNetReorder</a></li>
    </ul>
    
    
    <!--
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></li>
    </ul>
    
    <h2>Author</h2>
    Hai Fang and Julian Gough
    
    -->
    
  </div>

</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>