<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>Demo for multilayer omics dataset from Hiratani et al @ dnet 0.99.0</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="Hai Fang and Julian Gough">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 0.99.0</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <!--
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
        -->
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Demo for multilayer omics dataset from Hiratani et al</h2>

Notes:
<li> All results are based on dnet (version 0.99.0). </li>
<li> Images displayed below may be distorted, and they should be normal in your screen. </li>
<li> Functions contained in dnet 0.99.0 are hyperlinked in-place and also listed on the right side. </li>

<div class="row">
    <div class="span8">
      <pre>
      <div class='input'># This is a demo for multilayer omics dataset from Hiratani et al
# 
# This multilay omics dataset (available from <a href="http://www.ncbi.nlm.nih.gov/pubmed/19952138" target="19952138">http://www.ncbi.nlm.nih.gov/pubmed/19952138</a>) involves genome-wide replication-timing profiles of 22 cell lines from early mouse embryogenesis. These cell lines can be categorised into: 1) pluripotent cells, including ESCs (ESC_46C, ESC_D3 and ESC_TT2) and iPSCs (iPSC, iPSC_1D4 and iPSC_2D4); 2) partially-reprogrammed iPSCs (piPSC_1A2, piPSC_1B3 and piPSC_V3); 3) early epiblast (EPL and EMB3_D3); 4) late epiblast (EpiSC5 and EpiSC7); 5) Ectoderm (EBM6_D3, EBM9_D3, NPC_46C and NPC_TT2); 6) Mesoderm and Endoderm; and 7) late Mesoderm (Myoblast, MEF_female and MEF_male).
#
# The dataset is extracted for RefSeq gene TSS locations, including:
## RT: a replication timing matrix of 17,292 genes X 22 samples;
## CpG: a matrix of 17,292 genes X 1 containing gene additional information on promoter CpG classification (see <a href="http://www.ncbi.nlm.nih.gov/pubmed/17603471" target="17603471">http://www.ncbi.nlm.nih.gov/pubmed/17603471</a>), with '1' for HCP (high CpG density promoters), '-1' for LCP (low CpG density promoters), '0' for ICP (intermediate CpG density promoters), and 'NA' for unclassified;
## EX: an expression matrix of 17,292 genes X 8 samples, and samples include pluripotent cells (ESC_D3); early epiblast (EMB3_D3); late epiblast (EpiSC7); Ectoderm (EBM6_D3 and EBM9_D3); Mesoderm and Endoderm.
###############################################################################
library(dnet)

# Load or install packages specifically used in this demo
if(!require(affy)){
    install.packages("affy",repos="http://www.stats.bris.ac.uk/R")
    library(affy)
}
if(!require(limma)){
    install.packages("limma",repos="http://www.stats.bris.ac.uk/R")
    library(limma)
}

# This multilayer omics dataset involves the information on DNA replication timing, promoter CpG classification and gene expression. It consists of digitised replication timing, promoter CpG status and expression levels of 17,292 genes in a variety of samples.
load(url("http://dnet.r-forge.r-project.org/data/Hiratani_TableS1.RData"))
ls() # you should see three variables: 'RT', 'CpG' and 'EX'
</div>
<div class='output'>[1] "CpG" "EX"  "RT" 
</div>
<div class='input'>
# Entrez Gene information for the mouse
load(url("http://dnet.r-forge.r-project.org/data/org.Mm.eg.RData"))
gene_info <- org.Mm.eg$gene_info
gene_info[1:2,]
</div>
<div class='output'>      GeneID Symbol                        description chromosome map_location
11287  11287    Pzp             pregnancy zone protein          6   6 F1-G3|6 
11298  11298  Aanat arylalkylamine N-acetyltransferase         11   11 E2|11 8
                    Synonyms
11287   A1m|A2m|AI893533|MAM
11298 AA-NAT|Nat-2|Nat4|Snat
                                                             dbXrefs
11287   MGI:87854|Ensembl:ENSMUSG00000030359|Vega:OTTMUSG00000022212
11298 MGI:1328365|Ensembl:ENSMUSG00000020804|Vega:OTTMUSG00000003949
</div>
<div class='input'>
# create the 'eset' object for RT
group <- c(rep("ESC",3), rep("iPSC",3), rep("eEpiblast",2), rep("lEpiblast",2), rep("Ectoderm",4), rep("Mesoderm",1), rep("Endoderm",1), rep("piPSC",3), rep("Myoblast",3))
pdata <- data.frame(group=group, row.names=colnames(RT))
esetGene <- new("ExpressionSet", exprs=as.matrix(RT), phenoData=as(pdata,"AnnotatedDataFrame"))

# An igraph object that contains a functional protein association network in mouse. The network is extracted from the STRING database (version 9.0.5). Only those associations with medium confidence (score>=0.4) are retained.
load(url("http://dnet.r-forge.r-project.org/data/org.Mm.string.RData"))
org.Mm.string
</div>
<div class='output'>IGRAPH UN-- 18296 391653 -- 
+ attr: name (v/c), seqid (v/c), geneid (v/n), symbol (v/c),
  description (v/c), neighborhood_score (e/n), fusion_score (e/n),
  cooccurence_score (e/n), coexpression_score (e/n), experimental_score
  (e/n), database_score (e/n), textmining_score (e/n), combined_score
  (e/n)
</div>
<div class='input'>
# extract network that only contains genes in esetGene
## for extracted expression
ind <- match(V(org.Mm.string)$symbol, rownames(esetGene))
esetGeneSub <- esetGene[ind[!is.na(ind)],]
esetGeneSub
</div>
<div class='output'>ExpressionSet (storageMode: lockedEnvironment)
assayData: 9502 features, 22 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: ESC_46C ESC_D3 ... Myoblast (22 total)
  varLabels: group
  varMetadata: labelDescription
featureData: none
experimentData: use 'experimentData(object)'
Annotation:  
</div>
<div class='input'>## for extracted graph
ind <- match(rownames(esetGene), V(org.Mm.string)$symbol)
nodes_mapped <- V(org.Mm.string)$name[ind[!is.na(ind)]]
network <- <a href='dNetInduce.html' target=dNetInduce>dNetInduce</a>(g=org.Mm.string, nodes_query=nodes_mapped, knn=0, remove.loops=T, largest.comp=T)
V(network)$name <- V(network)$symbol
network
</div>
<div class='output'>IGRAPH UN-- 9041 155746 -- 
+ attr: name (v/c), seqid (v/c), geneid (v/n), symbol (v/c),
  description (v/c)
</div>
<div class='input'>
# 1) preparation of node p-values
## define the design matrix in a order manner
all <- as.vector(pData(esetGeneSub)$group)
level <- levels(factor(all))
index_level <- sapply(level, function(x) which(all==x)[1])
level_sorted <- all[sort(index_level, decreasing=F)]
design <- sapply(level_sorted, function(x) as.numeric(all==x)) # Convert a factor column to multiple boolean columns
## a linear model is fitted for every gene by the function lmFit
fit <- lmFit(exprs(esetGeneSub), design)
## define a contrast matrix
contrasts <- <a href='dContrast.html' target=dContrast>dContrast</a>(level_sorted, contrast.type="pairwise")
contrast.matrix <- makeContrasts(contrasts=contrasts$each, levels=design)
colnames(contrast.matrix) <- contrasts$name
## computes moderated t-statistics and log-odds of differential expression by empirical Bayes shrinkage of the standard errors towards a common value
fit2 <- contrasts.fit(fit, contrast.matrix)
fit2 <- eBayes(fit2)
# for p-value
pvals <- as.matrix(fit2$p.value)
# for adjusted p-value
adjpvals <- sapply(1:ncol(pvals),function(x) {
    p.adjust(pvals[,x], method="BH")
})
colnames(adjpvals) <- colnames(pvals)
# num of differentially expressed genes
apply(adjpvals<1e-2, 2, sum)
</div>
<div class='output'>           iPSC_ESC       eEpiblast_ESC       lEpiblast_ESC        Ectoderm_ESC 
                  0                  63                 698                1186 
       Mesoderm_ESC        Endoderm_ESC           piPSC_ESC        Myoblast_ESC 
                618                 549                1477                2341 
     eEpiblast_iPSC      lEpiblast_iPSC       Ectoderm_iPSC       Mesoderm_iPSC 
                 85                 705                1491                 679 
      Endoderm_iPSC          piPSC_iPSC       Myoblast_iPSC lEpiblast_eEpiblast 
                696                1402                2682                  59 
 Ectoderm_eEpiblast  Mesoderm_eEpiblast  Endoderm_eEpiblast     piPSC_eEpiblast 
                308                  51                  75                 756 
 Myoblast_eEpiblast  Ectoderm_lEpiblast  Mesoderm_lEpiblast  Endoderm_lEpiblast 
               1596                  41                 122                 159 
    piPSC_lEpiblast  Myoblast_lEpiblast   Mesoderm_Ectoderm   Endoderm_Ectoderm 
                692                1495                  76                 211 
     piPSC_Ectoderm   Myoblast_Ectoderm   Endoderm_Mesoderm      piPSC_Mesoderm 
               1354                1635                   3                 584 
  Myoblast_Mesoderm      piPSC_Endoderm   Myoblast_Endoderm      Myoblast_piPSC 
                735                 657                1005                1679 
</div>
<div class='input'>
# only for the comparisons of piPSC against iPSC
# visualise expression patterns for differentially expressed genes
my_contrast <- "piPSC_iPSC"
data <- exprs(esetGeneSub)[adjpvals[,my_contrast]<1e-2,]
heatmap(as.matrix(data),col=visColormap("bwr")(64),zlim=c(min(data),max(data)), scale="none", cexRow=0.2+0.5/log10(nrow(data)), cexCol=0.2+0.5/log10(ncol(data)), Rowv=NULL,Colv=NA)
</div>
<a href="javascript:newWin('Hiratani-14.png', 'Hiratani-14.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='Hiratani-14.png', width="400" /></a>
<div class='input'>
# get the p-values and calculate the scores thereupon
pval <- pvals[,my_contrast]
#pval <- <a href='dPvalAggregate.html' target=dPvalAggregate>dPvalAggregate</a>(pvals, method="orderStatistic", order=ncol(pvals))
#pval <- <a href='dPvalAggregate.html' target=dPvalAggregate>dPvalAggregate</a>(pvals, method="fishers", order=ncol(pvals))

# 2) identification of module
g <- <a href='dNetPipeline.html' target=dNetPipeline>dNetPipeline</a>(g=network, pval=pval, nsize=40)
</div>
<strong class='message'>Start at 2014-03-12 19:42:02</strong>
<strong class='message'></strong>
<strong class='message'>First, fit the p-value distribution under beta-uniform mixture model...</strong>
<strong class='message'>	A total of p-values: 9502</strong>
<strong class='message'>	Maximum Log-Likelihood: 10059.1</strong>
<strong class='message'>	Mixture parameter (lambda): 0.405</strong>
<strong class='message'>	Shape parameter (a): 0.217</strong>
<strong class='message'>Second, determine FDR threshold...</strong>
<strong class='message'>	Via constraint on the network size (40 nodes)</strong>
<strong class='message'>	Scanning FDR at rough stage...</strong>
<strong class='message'>		FDR: 1.00e-14, corresponding to the network size (0 nodes)</strong>
<strong class='message'>		FDR: 1.00e-13, corresponding to the network size (0 nodes)</strong>
<strong class='message'>		FDR: 1.00e-12, corresponding to the network size (0 nodes)</strong>
<strong class='message'>		FDR: 1.00e-11, corresponding to the network size (0 nodes)</strong>
<strong class='message'>		FDR: 1.00e-10, corresponding to the network size (1 nodes)</strong>
<strong class='message'>		FDR: 1.00e-09, corresponding to the network size (1 nodes)</strong>
<strong class='message'>		FDR: 1.00e-08, corresponding to the network size (2 nodes)</strong>
<strong class='message'>		FDR: 1.00e-07, corresponding to the network size (13 nodes)</strong>
<strong class='message'>		FDR: 1.00e-06, corresponding to the network size (34 nodes)</strong>
<strong class='message'>		FDR: 1.00e-05, corresponding to the network size (116 nodes)</strong>
<strong class='message'>	Scanning FDR at finetune stage...</strong>
<strong class='message'>		FDR: 1.50e-06, corresponding to the network size (39 nodes)</strong>
<strong class='message'>		FDR: 2.00e-06, corresponding to the network size (54 nodes)</strong>
<strong class='message'>	FDR threshold: 2.00e-06</strong>
<strong class='message'>Third, calculate the scores according to the fitted BUM and FDR threshold (if any)...</strong>
<strong class='message'>	Amongst 9502 scores, there are 179 positives.</strong>
<strong class='message'>Finally, find the module with the maximum score...</strong>
<strong class='message'>	Size of the module: 54 nodes and 78 edges</strong>
<strong class='message'></strong>
<strong class='message'>Finish at 2014-03-12 19:45:53</strong>
<strong class='message'>Runtime in total is: 231 secs
</strong>
<div class='input'>glayout <- layout.fruchterman.reingold(g)

# 3) color nodes according to communities identified via a spin-glass model and simulated annealing
#com <- spinglass.community(g, spins=3)
com <- walktrap.community(g, modularity=T)
com$csize <- sapply(1:length(com),function(x) sum(com$membership==x))
vgroups <- com$membership
colormap <- "yellow-darkorange"
palette.name <- visColormap(colormap=colormap)
mcolors <- palette.name(length(com))
vcolors <- mcolors[vgroups]

com$significance <- sapply(1:length(com), function(x) {
    community.significance.test <- function(g, vids, ...) {
        subg <- induced.subgraph(g, vids)
        within.degrees <- igraph::degree(subg)
        cross.degrees <- igraph::degree(g, vids) - within.degrees
        wilcox.test(within.degrees, cross.degrees, ...)
    }
    tmp <- suppressWarnings(community.significance.test(g, vids=V(g)$name[com$membership==x]))
    signif(tmp$p.value, digits=3)
})

# 4) size nodes according to degrees
vdegrees <- igraph::degree(g)

# 5) sort nodes: first by communities and then degrees
tmp <- data.frame(ind=1:vcount(g), vgroups, vdegrees)
ordering <- tmp[order(vgroups,vdegrees),]$ind

# 6) visualise graph using 1-dimensional arc diagram
<a href='visNetArc.html' target=visNetArc>visNetArc</a>(g, ordering=ordering, labels=V(g)$geneSymbol, vertex.label.color=vcolors, vertex.color=vcolors, vertex.frame.color=vcolors, vertex.size=log(vdegrees)+0.1, vertex.label.cex=0.4)
</div>
<a href="javascript:newWin('Hiratani-48.png', 'Hiratani-48.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='Hiratani-48.png', width="400" /></a>
<div class='input'>
# 7) visualise graph using circle diagram
# 7a) drawn into a single circle 
<a href='visNetCircle.html' target=visNetCircle>visNetCircle</a>(g=g, com=com, ordering=ordering, colormap=colormap, vertex.label=V(g)$symbol, vertex.size=igraph::degree(g)+5, vertex.label.color="black", vertex.label.cex=0.6, vertex.label.dist=0.75, vertex.shape="sphere", edge.color.within="grey", edge.color.crossing="black", edge.width=1, edge.lty=1, mark.shape=1, mark.expand=10)
</div>
<a href="javascript:newWin('Hiratani-50.png', 'Hiratani-50.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='Hiratani-50.png', width="400" /></a>
<div class='input'># 7b) drawn into multiple circles
<a href='visNetCircle.html' target=visNetCircle>visNetCircle</a>(g=g, com=com, circles="multiple", ordering=ordering, colormap=colormap, vertex.label=V(g)$symbol, vertex.size=igraph::degree(g)+5, vertex.label.color="black", vertex.label.cex=0.6, vertex.label.dist=0.25, vertex.shape="sphere", edge.color.within="grey", edge.color.crossing="black", edge.width=1, edge.lty=1, mark.shape=1, mark.expand=10)
</div>
<a href="javascript:newWin('Hiratani-52.png', 'Hiratani-52.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='Hiratani-52.png', width="400" /></a>
<div class='input'>
# 8) as comparison, also visualise graph on 2-dimensional layout 
mark.groups <- communities(com)
mark.col <- visColoralpha(mcolors, alpha=0.2)
mark.border <- visColoralpha(mcolors, alpha=0.2)
edge.color <- c("grey", "black")[crossing(com,g)+1]
<a href='visNet.html' target=visNet>visNet</a>(g, glayout=glayout, vertex.label=V(g)$geneSymbol, vertex.color=vcolors, vertex.frame.color=vcolors, vertex.shape="sphere", mark.groups=mark.groups, mark.col=mark.col, mark.border=mark.border, mark.shape=1, mark.expand=10, edge.color=edge.color)
</div>
<a href="javascript:newWin('Hiratani-54.png', 'Hiratani-54.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='Hiratani-54.png', width="400" /></a>
<div class='input'>
legend_name <- paste("C",1:length(mcolors)," (n=",com$csize,", pval=",signif(com$significance,digits=2),")",sep='')
legend("bottomleft", legend=legend_name, fill=mcolors, bty="n", cex=0.6)
</div>
<a href="javascript:newWin('Hiratani-56.png', 'Hiratani-56.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='Hiratani-56.png', width="400" /></a>
<div class='input'>
# 9) color by score and FC
# colored by score
<a href='visNet.html' target=visNet>visNet</a>(g, glayout=glayout, pattern=V(g)$score, zlim=c(-1*ceiling(max(abs(V(g)$score))),ceiling(max(abs(V(g)$score)))), vertex.shape="circle", mark.groups=mark.groups, mark.col=mark.col, mark.border=mark.border, mark.shape=1, mark.expand=10, edge.color=edge.color)
</div>
<a href="javascript:newWin('Hiratani-58.png', 'Hiratani-58.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='Hiratani-58.png', width="400" /></a>
<div class='input'># colored by FC
logFC <- fit2$coefficients[V(g)$name,my_contrast]
<a href='visNet.html' target=visNet>visNet</a>(g, glayout=glayout, pattern=logFC, zlim=c(-1,1), mark.groups=mark.groups, mark.col=mark.col, mark.border=mark.border, mark.shape=1, mark.expand=10, edge.color=edge.color)
</div>
<a href="javascript:newWin('Hiratani-60.png', 'Hiratani-60.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='Hiratani-60.png', width="400" /></a>
<div class='input'>
# 10) color by additional data
data <- exprs(esetGeneSub)[V(g)$name,]
<a href='visNetMul.html' target=visNetMul>visNetMul</a>(g=g, data=data, height=ceiling(sqrt(ncol(data)))*2, newpage=T,glayout=glayout,colormap="darkgreen-lightgreen-lightpink-darkred",vertex.label=NA,vertex.shape="sphere", vertex.size=18,mtext.cex=0.7,border.color="888888", mark.groups=mark.groups, mark.col=mark.col, mark.border=mark.border, mark.shape=1, mark.expand=10, edge.color=edge.color)
</div>
<a href="javascript:newWin('Hiratani-62.png', 'Hiratani-62.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='Hiratani-62.png', width="400" /></a>
<div class='input'>
# 11) color by additional data (be reordered)
sReorder <- <a href='dNetReorder.html' target=dNetReorder>dNetReorder</a>(g, data, feature="edge", node.normalise="degree", amplifier=3, metric="none")
</div>
<strong class='message'>Start at 2014-03-12 19:47:48</strong>
<strong class='message'></strong>
<strong class='message'>First, define topology of a map grid...</strong>
<strong class='message'>Second, initialise the codebook matrix given a topology and input data...</strong>
<strong class='message'>Third, get training at the rough stage...</strong>
<strong class='message'>Fourth, get training at the finetune stage...</strong>
<strong class='message'>Next, identify the best-matching hexagon/rectangle for the input data...</strong>
<strong class='message'>Finally, append the response data (hits and mqe) into the sMap object...</strong>
<strong class='message'></strong>
<strong class='message'>Below are the summaries of the training results:</strong>
<strong class='message'>   dimension of input data: 22x78
   xy-dimension of map grid: xdim=9, ydim=9
   grid lattice: rect
   grid shape: sheet
   dimension of grid coord: 81x2
   initialisation method: linear
   dimension of codebook matrix: 81x78
   mean quantization error: 0.66157179452147
</strong>
<strong class='message'>Below are the details of trainology:</strong>
<strong class='message'>   training algorithm: sequential
   alpha type: invert
   training neighborhood kernel: gaussian
   trainlength (x input data length): 37 at rough stage; 148 at finetune stage
   radius (at rough stage): from 2 to 1
   radius (at finetune stage): from 1 to 1
</strong>
<strong class='message'>End at 2014-03-12 19:47:55</strong>
<strong class='message'>Runtime in total is: 7 secs
</strong>
<div class='input'><a href='visNetReorder.html' target=visNetReorder>visNetReorder</a>(g=g, data=data, sReorder=sReorder, height=ceiling(sqrt(ncol(data)))*2, newpage=T, glayout=glayout, colormap="darkgreen-lightgreen-lightpink-darkred", vertex.label=NA,vertex.shape="sphere", vertex.size=18,mtext.cex=0.5,border.color="888888", mark.groups=mark.groups, mark.col=mark.col, mark.border=NA, mark.shape=1, mark.expand=10, edge.color=edge.color)</div>
<a href="javascript:newWin('Hiratani-80.png', 'Hiratani-80.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='Hiratani-80.png', width="400" /></a>
      </pre>
    </div>

  <div class="span4">
    
    <h2>Functions used in this demo</h2>
    <ul>
      <li><a href="dNetInduce.html">dNetInduce</a></li>
      <li><a href="dContrast.html">dContrast</a></li>
      <li><a href="dPvalAggregate.html">dPvalAggregate</a></li>
      <li><a href="dNetPipeline.html">dNetPipeline</a></li>
      <li><a href="visNetArc.html">visNetArc</a></li>
      <li><a href="visNetCircle.html">visNetCircle</a></li>
      <li><a href="visNet.html">visNet</a></li>
      <li><a href="visNetMul.html">visNetMul</a></li>
      <li><a href="dNetReorder.html">dNetReorder</a></li>
      <li><a href="visNetReorder.html">visNetReorder</a></li>
    </ul>
    
    
    <!--
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></li>
    </ul>
    
    <h2>Author</h2>
    Hai Fang and Julian Gough
    
    -->
    
  </div>

</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>