<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>dEnricher @ dnet 1.0.0</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 1.0.0</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="rdata.html"><i class="icon-file icon-white"></i> RData</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to conduct enrichment analysis given the input data and the ontology in query</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>dEnricher</code> is supposed to conduct enrichment analysis
given the input data and the ontology in query. It returns
an object of class "eTerm". Enrichment analysis is based on
either Fisher's exact test or Hypergeometric test. The test
can respect the hierarchy of the ontology.</p>
  
    <h3>Usage</h3>
    <pre>dEnricher(data, identity = c("symbol", "entrez"), check.symbol.identity = FALSE, 
  genome = c("Hs", "Mm", "Rn", "Gg", "Ce", "Dm", "Da", "At"), ontology = c("GOBP", 
          "GOMF", "GOCC", "PS", "DO", "HPPA", "HPMI", "HPON", "MP", "MsigdbC1", "MsigdbC2CGP", 
          "MsigdbC2CP", "MsigdbC2KEGG", "MsigdbC2REACTOME", "MsigdbC2BIOCARTA", "MsigdbC3TFT", 
          "MsigdbC3MIR", "MsigdbC4CGN", "MsigdbC4CM", "MsigdbC5BP", "MsigdbC5MF", "MsigdbC5CC", 
          "MsigdbC6", "MsigdbC7"), sizeRange = c(10, 1000), which_distance = NULL, 
      test = c("FisherTest", "HypergeoTest", "BinomialTest"), p.adjust.method = c("BH", 
          "BY", "bonferroni", "holm", "hochberg", "hommel"), ontology.algorithm = c("none", 
          "pc", "elim", "lea"), elim.pvalue = 0.01, lea.depth = 2, verbose = T, RData.location = "http://dnet.r-forge.r-project.org/data")</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>data</dt>
      <dd>an input vector. It contains either Entrez
Gene ID or Symbol</dd>
      <dt>identity</dt>
      <dd>the type of gene identity (i.e. row names
of input data), either "symbol" for gene symbols (by
default) or "entrez" for Entrez Gene ID. The option
"symbol" is preferred as it is relatively stable from one
update to another; also it is possible to search against
synonyms (see the next parameter)</dd>
      <dt>check.symbol.identity</dt>
      <dd>logical to indicate whether
synonyms will be searched against when gene symbols
cannot be matched. By default, it sets to FALSE since it
may take a while to do such check using all possible
synoyms</dd>
      <dt>genome</dt>
      <dd>the genome identity. It can be one of "Hs"
for human, "Mm" for mouse, "Rn" for rat, "Gg" for
chicken, "Ce" for c.elegans, "Dm" for fruitfly, "Da" for
zebrafish, and "At" for arabidopsis</dd>
      <dt>ontology</dt>
      <dd>the ontology supported currently. It can
be "GOBP" for Gene Ontology Biological Process, "GOMF"
for Gene Ontology Molecular Function, "GOCC" for Gene
Ontology Cellular Component, "PS" for phylostratific age
information, "DO" for Disease Ontology, "HPPA" for Human
Phenotype Phenotypic Abnormality, "HPMI" for Human
Phenotype Mode of Inheritance, "HPON" for Human Phenotype
ONset and clinical course, "MP" for Mammalian Phenotype,
and the molecular signatures database (Msigdb) in human
(including "MsigdbC1", "MsigdbC2CGP", "MsigdbC2CP",
"MsigdbC2KEGG", "MsigdbC2REACTOME", "MsigdbC2BIOCARTA",
"MsigdbC3TFT", "MsigdbC3MIR", "MsigdbC4CGN",
"MsigdbC4CM", "MsigdbC5BP", "MsigdbC5MF", "MsigdbC5CC",
"MsigdbC6", "MsigdbC7"). Note: These four ("GOBP",
"GOMF", "GOCC" and "PS") are availble for all
genomes/species; for "Hs" and "Mm", these five ("DO",
"HPPA", "HPMI", "HPON" and "MP") are also supported; all
"Msigdb" are only supported in "Hs". For details on the
eligibility for pairs of input genome and ontology,
please refer to the online Documentations at
<a href = 'http://dnet.r-forge.r-project.org/docs.html'>http://dnet.r-forge.r-project.org/docs.html</a></dd>
      <dt>sizeRange</dt>
      <dd>the minimum and maximum size of members
of each gene set in consideration. By default, it sets to
a minimum of 10 but no more than 1000</dd>
      <dt>which_distance</dt>
      <dd>which distance of terms in the
ontology is used to restrict terms in consideration. By
default, it sets to 'NULL' to consider all distances</dd>
      <dt>test</dt>
      <dd>the statistic test used. It can be
"FisherTest" for using fisher's exact test,
"HypergeoTest" for using hypergeometric test, or
"BinomialTest" for using binomial test. Fisher's exact
test is to test the independence between gene group
(genes belonging to a group or not) and gene annotation
(genes annotated by a term or not), and thus compare
sampling to the left part of background (after sampling
without replacement). Hypergeometric test is to sample at
random (without replacement) from the background
containing annotated and non-annotated genes, and thus
compare sampling to background. Unlike hypergeometric
test, binomial test is to sample at random (with
replacement) from the background with the constant
probability. In terms of the ease of finding the
significance, they are in order: hypergeometric test >
binomial test > fisher's exact test. In other words, in
terms of the calculated p-value, hypergeometric test <
binomial test < fisher's exact test</dd>
      <dt>p.adjust.method</dt>
      <dd>the method used to adjust
p-values. It can be one of "BH", "BY", "bonferroni",
"holm", "hochberg" and "hommel". The first two methods
"BH" (widely used) and "BY" control the false discovery
rate (FDR: the expected proportion of false discoveries
amongst the rejected hypotheses); the last four methods
"bonferroni", "holm", "hochberg" and "hommel" are
designed to give strong control of the family-wise error
rate (FWER). Notes: FDR is a less stringent condition
than FWER</dd>
      <dt>ontology.algorithm</dt>
      <dd>the algorithm used to account
for the hierarchy of the ontology. It can be one of
"none", "pc", "elim" and "lea". For details, please see
'Note'</dd>
      <dt>elim.pvalue</dt>
      <dd>the parameter only used when
"ontology.algorithm" is "elim". It is used to control how
to declare a signficantly enriched term (and subsequently
all genes in this term are eliminated from all its
ancestors)</dd>
      <dt>lea.depth</dt>
      <dd>the parameter only used when
"ontology.algorithm" is "lea". It is used to control how
many maximum depth is uded to consider the children of a
term (and subsequently all genes in these children term
are eliminated from the use for the recalculation of the
signifance at this term)</dd>
      <dt>verbose</dt>
      <dd>logical to indicate whether the messages
will be displayed in the screen. By default, it sets to
false for no display</dd>
      <dt>RData.location</dt>
      <dd>the characters to tell the location
of built-in RData files. By default, it remotely locates
at <a href = '"http://dnet.r-forge.r-project.org/data"'>"http://dnet.r-forge.r-project.org/data"</a>. For
the user equipped with fast internet connection, this
option can be just left as default. But it is always
advisable to download these files locally. Especially
when the user needs to run this function many times,
there is no need to ask the function to remotely download
every time (also it will unnecessarily increase the
runtime). For examples, these files (as a whole or part
of them) can be first downloaded into your current
working directory, and then set this option as:
<code class = 'eq'>RData.location="."</code>. Surely, the location can be
anywhere as long as the user provides the correct path
pointing to (otherwise, the script will have to remote
download each time). Here is the UNIX command for
downloading all RData files (preserving the directory
structure): <code class = 'eq'>wget -r -l2 -A "*.RData" -np -nH
--cut-dirs=0 "http://dnet.r-forge.r-project.org/data"</code></dd>
    </dl>
    <!-- 
    -->
    
    <div class="Value">
      <h3>Value</h3>
      
      <p>an object of class "eTerm", a list with following
components: <ul>
<li><code>set_info</code>: a matrix of
nSet X 4 containing gene set information, where nSet is the
number of gene set in consideration, and the 4 columns are
"setID" (i.e. "Term ID"), "name" (i.e. "Term Name"),
"namespace" and "distance" </li>
<li><code>gs</code>: a list of gene
sets, each storing gene members. Always, gene sets are
identified by "setID" and gene members identified by
"Entrez ID" </li>
<li><code>data</code>: a vector containing input
data in consideration. It is not always the same as the
input data as only those mappable are retained
</li>
<li><code>pvalue</code>: a vector containing p-values
</li>
<li><code>adjp</code>: a vector containing adjusted p-values.
It is the p value but after being adjusted for multiple
comparisons </li>
<li><code>call</code>: the call that produced this
result </li>
</ul></p>
      <p></p>
    </div>
    <div class="Note">
      <h3>Note</h3>
      
      <p>The interpretation of the algorithms used to account for
the hierarchy of the ontology: <ul>
<li>"none": does
not consider the ontology hierarchy at all; </li>
<li>"lea":
computers the significance of a term in terms of the
significance of its children at the maximum depth (e.g. 2).
Precisely, once genes are already annotated to any children
terms with a more signficance than itself, then all these
genes are eliminated from the use for the recalculation of
the signifance at that term. The final p-values takes the
maximum of the original p-value and the recalculated
p-value </li>
<li>"elim": computers the significance of a term
in terms of the significance of its all children.
Precisely, once genes are already annotated to a
signficantly enriched term under the cutoff of e.g.
pvalue<1e-2, all these genes are eliminated from the
ancestors of that term); </li>
<li>"pc": requires the
significance of a term not only using the whole genes as
background but also using genes annotated to all its direct
parents/ancestors as background. The final p-value takes
the maximum of both p-values in these two calculations;
</li>
<li>"Notes": the order of the number of significant terms
is: "none" > "lea" > "elim" > "pc". </li>
</ul></p>
      <p></p>
    </div>
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'>#\dontrun{
load(url("http://dnet.r-forge.r-project.org/data/Datasets/Hiratani_TableS1.RData"))
data <- rownames(RT)[1:1000]
eTerm <- dEnricher(data, identity="symbol", genome="Mm", ontology="MP",
RData.location="./RData_Rd")
</div>
<strong class='message'>Start at 2014-04-05 20:18:20</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology MP and its gene associations in the genome Mm (2014-04-05 20:18:20) ...</strong>
<strong class='message'>	Load Enterz Gene information from http://dnet.r-forge.r-project.org/data/Mm/org.Mm.eg.RData</strong>
<strong class='message'>	Load annotation information from http://dnet.r-forge.r-project.org/data/Mm/org.Mm.egMP.RData</strong>
<strong class='message'>Then, do mapping based on symbol (2014-04-05 20:18:22) ...</strong>
<strong class='message'>	Among 1000 symbols of input data, there are 853 mappable via official gene symbols but 147 left unmappable</strong>
<strong class='message'>Third, perform enrichment analysis using FisherTest (2014-04-05 20:18:22) ...</strong>
<strong class='message'>	There are 4519 terms being used, each restricted within [10,1000] annotations</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-04-05 20:18:25) ...</strong>
<strong class='message'>
End at 2014-04-05 20:18:25</strong>
<strong class='message'>Runtime in total is: 5 secs
</strong>
<div class='input'>eTerm <- dEnricher(data, identity="symbol", genome="Mm", ontology="MP",
ontology.algorithm="pc", RData.location="./RData_Rd")
</div>
<strong class='message'>Start at 2014-04-05 20:18:25</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology MP and its gene associations in the genome Mm (2014-04-05 20:18:25) ...</strong>
<strong class='message'>	Load Enterz Gene information from http://dnet.r-forge.r-project.org/data/Mm/org.Mm.eg.RData</strong>
<strong class='message'>	Load annotation information from http://dnet.r-forge.r-project.org/data/Mm/org.Mm.egMP.RData</strong>
<strong class='message'>Then, do mapping based on symbol (2014-04-05 20:18:27) ...</strong>
<strong class='message'>	Among 1000 symbols of input data, there are 853 mappable via official gene symbols but 147 left unmappable</strong>
<strong class='message'>Third, perform enrichment analysis using FisherTest based on pc algorithm to respect ontology structure (2014-04-05 20:18:27) ...</strong>
<strong class='message'>	Load ontology information from http://dnet.r-forge.r-project.org/data/Obo/ig.MP.RData</strong>
<strong class='message'>	There are 4609 terms being used</strong>
<strong class='message'>	At level 15, there are 4 nodes/terms</strong>
<strong class='message'>	At level 14, there are 19 nodes/terms</strong>
<strong class='message'>	At level 13, there are 43 nodes/terms</strong>
<strong class='message'>	At level 12, there are 93 nodes/terms</strong>
<strong class='message'>	At level 11, there are 143 nodes/terms</strong>
<strong class='message'>	At level 10, there are 227 nodes/terms</strong>
<strong class='message'>	At level 9, there are 373 nodes/terms</strong>
<strong class='message'>	At level 8, there are 640 nodes/terms</strong>
<strong class='message'>	At level 7, there are 865 nodes/terms</strong>
<strong class='message'>	At level 6, there are 918 nodes/terms</strong>
<strong class='message'>	At level 5, there are 820 nodes/terms</strong>
<strong class='message'>	At level 4, there are 364 nodes/terms</strong>
<strong class='message'>	At level 3, there are 70 nodes/terms</strong>
<strong class='message'>	At level 2, there are 29 nodes/terms</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-04-05 20:18:57) ...</strong>
<strong class='message'>
End at 2014-04-05 20:18:57</strong>
<strong class='message'>Runtime in total is: 32 secs
</strong>
<div class='input'>eTerm <- dEnricher(data, identity="symbol", genome="Mm", ontology="MP",
ontology.algorithm="elim", RData.location="./RData_Rd")
</div>
<strong class='message'>Start at 2014-04-05 20:18:57</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology MP and its gene associations in the genome Mm (2014-04-05 20:18:57) ...</strong>
<strong class='message'>	Load Enterz Gene information from http://dnet.r-forge.r-project.org/data/Mm/org.Mm.eg.RData</strong>
<strong class='message'>	Load annotation information from http://dnet.r-forge.r-project.org/data/Mm/org.Mm.egMP.RData</strong>
<strong class='message'>Then, do mapping based on symbol (2014-04-05 20:19:02) ...</strong>
<strong class='message'>	Among 1000 symbols of input data, there are 853 mappable via official gene symbols but 147 left unmappable</strong>
<strong class='message'>Third, perform enrichment analysis using FisherTest based on elim algorithm to respect ontology structure (2014-04-05 20:19:03) ...</strong>
<strong class='message'>	Load ontology information from http://dnet.r-forge.r-project.org/data/Obo/ig.MP.RData</strong>
<strong class='message'>	There are 4609 terms being used</strong>
<strong class='message'>	At level 15, there are 4 nodes/terms: so far, a total of 0 nodes called significant, and 0 ancestral nodes changed (0 genes eliminated)</strong>
<strong class='message'>	At level 14, there are 19 nodes/terms: so far, a total of 0 nodes called significant, and 0 ancestral nodes changed (0 genes eliminated)</strong>
<strong class='message'>	At level 13, there are 43 nodes/terms: so far, a total of 2 nodes called significant, and 44 ancestral nodes changed (243 genes eliminated)</strong>
<strong class='message'>	At level 12, there are 93 nodes/terms: so far, a total of 3 nodes called significant, and 50 ancestral nodes changed (284 genes eliminated)</strong>
<strong class='message'>	At level 11, there are 143 nodes/terms: so far, a total of 6 nodes called significant, and 58 ancestral nodes changed (380 genes eliminated)</strong>
<strong class='message'>	At level 10, there are 227 nodes/terms: so far, a total of 9 nodes called significant, and 78 ancestral nodes changed (505 genes eliminated)</strong>
<strong class='message'>	At level 9, there are 373 nodes/terms: so far, a total of 19 nodes called significant, and 123 ancestral nodes changed (822 genes eliminated)</strong>
<strong class='message'>	At level 8, there are 640 nodes/terms: so far, a total of 26 nodes called significant, and 153 ancestral nodes changed (1020 genes eliminated)</strong>
<strong class='message'>	At level 7, there are 865 nodes/terms: so far, a total of 39 nodes called significant, and 197 ancestral nodes changed (1516 genes eliminated)</strong>
<strong class='message'>	At level 6, there are 918 nodes/terms: so far, a total of 47 nodes called significant, and 211 ancestral nodes changed (2071 genes eliminated)</strong>
<strong class='message'>	At level 5, there are 820 nodes/terms: so far, a total of 59 nodes called significant, and 219 ancestral nodes changed (2683 genes eliminated)</strong>
<strong class='message'>	At level 4, there are 364 nodes/terms: so far, a total of 60 nodes called significant, and 220 ancestral nodes changed (2731 genes eliminated)</strong>
<strong class='message'>	At level 3, there are 70 nodes/terms: so far, a total of 62 nodes called significant, and 221 ancestral nodes changed (5196 genes eliminated)</strong>
<strong class='message'>	At level 2, there are 29 nodes/terms: so far, a total of 62 nodes called significant, and 221 ancestral nodes changed (5196 genes eliminated)</strong>
<strong class='message'>	At level 1, there are 1 nodes/terms: so far, a total of 62 nodes called significant, and 221 ancestral nodes changed (5196 genes eliminated)</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-04-05 20:19:23) ...</strong>
<strong class='message'>
End at 2014-04-05 20:19:23</strong>
<strong class='message'>Runtime in total is: 26 secs
</strong>
<div class='input'>eTerm <- dEnricher(data, identity="symbol", genome="Mm", ontology="MP",
ontology.algorithm="lea", RData.location="./RData_Rd")
</div>
<strong class='message'>Start at 2014-04-05 20:19:23</strong>
<strong class='message'></strong>
<strong class='message'>First, load the ontology MP and its gene associations in the genome Mm (2014-04-05 20:19:23) ...</strong>
<strong class='message'>	Load Enterz Gene information from http://dnet.r-forge.r-project.org/data/Mm/org.Mm.eg.RData</strong>
<strong class='message'>	Load annotation information from http://dnet.r-forge.r-project.org/data/Mm/org.Mm.egMP.RData</strong>
<strong class='message'>Then, do mapping based on symbol (2014-04-05 20:19:24) ...</strong>
<strong class='message'>	Among 1000 symbols of input data, there are 853 mappable via official gene symbols but 147 left unmappable</strong>
<strong class='message'>Third, perform enrichment analysis using FisherTest based on lea algorithm to respect ontology structure (2014-04-05 20:19:25) ...</strong>
<strong class='message'>	Load ontology information from http://dnet.r-forge.r-project.org/data/Obo/ig.MP.RData</strong>
<strong class='message'>	There are 4609 terms being used</strong>
<strong class='message'>	At level 15, there are 4 nodes/terms and 0 being recalculated</strong>
<strong class='message'>	At level 14, there are 19 nodes/terms and 2 being recalculated</strong>
<strong class='message'>	At level 13, there are 43 nodes/terms and 5 being recalculated</strong>
<strong class='message'>	At level 12, there are 93 nodes/terms and 19 being recalculated</strong>
<strong class='message'>	At level 11, there are 143 nodes/terms and 42 being recalculated</strong>
<strong class='message'>	At level 10, there are 227 nodes/terms and 51 being recalculated</strong>
<strong class='message'>	At level 9, there are 373 nodes/terms and 85 being recalculated</strong>
<strong class='message'>	At level 8, there are 640 nodes/terms and 128 being recalculated</strong>
<strong class='message'>	At level 7, there are 865 nodes/terms and 224 being recalculated</strong>
<strong class='message'>	At level 6, there are 918 nodes/terms and 284 being recalculated</strong>
<strong class='message'>	At level 5, there are 820 nodes/terms and 309 being recalculated</strong>
<strong class='message'>	At level 4, there are 364 nodes/terms and 215 being recalculated</strong>
<strong class='message'>	At level 3, there are 70 nodes/terms and 57 being recalculated</strong>
<strong class='message'>	At level 2, there are 29 nodes/terms and 23 being recalculated</strong>
<strong class='message'>	At level 1, there are 1 nodes/terms and 1 being recalculated</strong>
<strong class='message'>Last, adjust the p-values using the BH method (2014-04-05 20:19:52) ...</strong>
<strong class='message'>
End at 2014-04-05 20:19:52</strong>
<strong class='message'>Runtime in total is: 29 secs
</strong>
<div class='input'>cbind(eTerm$set_info[which(eTerm$pvalue < 1e-3), c(1,2)],
eTerm$pvalue[which(eTerm$pvalue < 1e-3)])
</div>
<div class='output'>                setID                                                      name
MP:0000520 MP:0000520                                             absent kidney
MP:0000743 MP:0000743                                              muscle spasm
MP:0001327 MP:0001327               decreased retinal photoreceptor cell number
MP:0001970 MP:0001970                                   abnormal pain threshold
MP:0002268 MP:0002268                   abnormal terminal bronchiole morphology
MP:0002625 MP:0002625                          heart left ventricle hypertrophy
MP:0002896 MP:0002896                              abnormal bone mineralization
MP:0002966 MP:0002966          decreased circulating alkaline phosphatase level
MP:0003015 MP:0003015                    abnormal circulating bicarbonate level
MP:0003073 MP:0003073                       abnormal metacarpal bone morphology
MP:0003290 MP:0003290                                intestinal hypoperistalsis
MP:0008084 MP:0008084                            absent single-positive T cells
MP:0008101 MP:0008101                                     lymph node hypoplasia
MP:0008586 MP:0008586                  disorganized photoreceptor outer segment
MP:0011059 MP:0011059                abnormal ependyma motile cilium morphology
MP:0011117 MP:0011117                    abnormal susceptibility to weight gain
MP:0011241 MP:0011241 abnormal fetal derived definitive erythrocyte cell number
MP:0011368 MP:0011368                                increased kidney apoptosis
           eTerm$pvalue[which(eTerm$pvalue < 0.001)]
MP:0000520                              3.708892e-05
MP:0000743                              1.131583e-04
MP:0001327                              3.248387e-05
MP:0001970                              9.439912e-06
MP:0002268                              2.822217e-04
MP:0002625                              9.706980e-04
MP:0002896                              4.101757e-04
MP:0002966                              6.874354e-06
MP:0003015                              1.430607e-04
MP:0003073                              6.741427e-06
MP:0003290                              7.911911e-05
MP:0008084                              2.195097e-04
MP:0008101                              9.097551e-04
MP:0008586                              3.380702e-04
MP:0011059                              7.727206e-04
MP:0011117                              3.931322e-04
MP:0011241                              1.670333e-05
MP:0011368                              5.674676e-04
</div>
<div class='input'>
# highlight the top significant terms and also color-code all terms according to the adjust p-values
load(url("http://dnet.r-forge.r-project.org/data/Obo/ig.MP.RData"))
g <- ig.MP
nodes_query <- names(sort(eTerm$adjp)[1:5])
nodes.highlight <- rep("red", length(nodes_query))
names(nodes.highlight) <- nodes_query
subg <- dDAGinduce(g, nodes_query)
visDAG(g=subg, data=-1*log10(eTerm$adjp[V(subg)$name]),
node.info="both", zlim=c(0,2), node.attrs=list(color=nodes.highlight))
</div>
<a href="javascript:newWin('dEnricher-103.png', 'dEnricher-103.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='dEnricher-103.png', width="400" /></a>
<div class='input'>#}
</div></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>dEnricher</li>
    </ul>
    <ul>
      
    </ul> 
    -->
      
    <h2>See also</h2>
    
<code><a href='dEnricher.html'>dEnricher</a></code>

    
    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) DNET: dynamic networks via integrative analysis of network, expression, evolution and ontology data. R package version 1.0.0. <a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></p>

    
    <!--
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>