<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>dFDRscore @ dnet 0.99.0</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 0.99.0</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to transform fdr into scores according to log-likelihood ratio between the true positives and the false positivies and/or after controlling false discovery rate</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>dFDRscore</code> is supposed to take as input a vector of
  fdr, which are transformed into scores according to
  according to log-likelihood ratio between the true
  positives and the false positivies. Also if the FDR
  threshold is given, it is used to make sure that fdr
  below threshold are considered significant and thus
  scored positively. Instead, those fdr above the given
  threshold are considered insigificant and thus scored
  negatively.</p>
  
    <h3>Usage</h3>
    <pre>dFDRscore(fdr, fdr.threshold = NULL, scatter = F)</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>fdr</dt>
      <dd>a vector containing a list of input fdr</dd>
      <dt>fdr.threshold</dt>
      <dd>the given FDR threshold. By default,
  it is set to NULL, meaning there is no constraint. If
  given, those fdr with the FDR below threshold are
  considered significant and thus scored positively.
  Instead, those fdr with the FDR above given threshold are
  considered insigificant and thus scored negatively</dd>
      <dt>scatter</dt>
      <dd>logical to indicate whether the scatter
  graph of scores against p-values should be drawn. Also
  indicated is the score corresponding to the given FDR
  threshold (if any)</dd>
    </dl>
    <!-- 
    -->
    
    <div class="Value">
      <h3>Value</h3>
      
      <p><ul>
<li><code>scores</code>: a vector of scores </li>
</ul></p>
      <p></p>
    </div>
    <div class="Note">
      <h3>Note</h3>
      
      <p>none</p>
    </div>
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'># 1) generate data with three different distributions, each with an iid normal random matrix of 1000 x 3
data <- cbind(matrix(rnorm(1000*3,mean=0,sd=1), nrow=1000, ncol=3),
matrix(rnorm(1000*3,mean=0.5,sd=1), nrow=1000, ncol=3),
matrix(rnorm(1000*3,mean=-0.5,sd=1), nrow=1000, ncol=3))

# 2) calculate the significance according to SVD
# using "fdr" significance
fdr <- dSVDSignif(data, signif="fdr", num.permutation=100)
</div>
<strong class='message'>Start at 2014-03-18 18:16:14</strong>
<strong class='message'></strong>
<strong class='message'>First, singular value decomposition of the input matrix (with 1000 rows and 9 columns)...</strong>
<strong class='message'>Second, determinate the eigens...</strong>
<strong class='message'>	via automatically deciding on the number of dominant eigens under the cutoff of 1.00e-02 pvalue</strong>
<strong class='message'>	number of the eigens in consideration: 1</strong>
<strong class='message'>Third, construct the gene-specific projection vector,and calculate distance statistics...</strong>
<strong class='message'>Finally, obtain gene significance (fdr) based on 100 permutations...</strong>
<strong class='message'>	doing row-wise permutations...</strong>
<strong class='message'>	estimating fdr...</strong>
<strong class='message'>1 (out of 1000) at 2014-03-18 18:16:16</strong>
<strong class='message'>100 (out of 1000) at 2014-03-18 18:16:16</strong>
<strong class='message'>200 (out of 1000) at 2014-03-18 18:16:16</strong>
<strong class='message'>300 (out of 1000) at 2014-03-18 18:16:17</strong>
<strong class='message'>400 (out of 1000) at 2014-03-18 18:16:17</strong>
<strong class='message'>500 (out of 1000) at 2014-03-18 18:16:17</strong>
<strong class='message'>600 (out of 1000) at 2014-03-18 18:16:17</strong>
<strong class='message'>700 (out of 1000) at 2014-03-18 18:16:17</strong>
<strong class='message'>800 (out of 1000) at 2014-03-18 18:16:18</strong>
<strong class='message'>900 (out of 1000) at 2014-03-18 18:16:18</strong>
<strong class='message'>1000 (out of 1000) at 2014-03-18 18:16:18</strong>
<strong class='message'>	using stepup procedure...</strong>
<strong class='message'></strong>
<strong class='message'>Finish at 2014-03-18 18:16:18</strong>
<strong class='message'>Runtime in total is: 4 secs
</strong>
<div class='input'>
# 3) calculate the scores according to the fitted BUM and fdr=0.01
# no fdr threshold
scores <- dFDRscore(fdr)
# using fdr threshold of 0.01
scores <- dFDRscore(fdr, fdr.threshold=0.1, scatter=TRUE)
</div>
<a href="javascript:newWin('dFDRscore-28.png', 'dFDRscore-28.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:3px solid #FFFFFF;" src='dFDRscore-28.png', width="400" /></a></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>dFDRscore</li>
    </ul>
    <ul>
      
    </ul> 
    -->
      
    <h2>See also</h2>
    
  <code><a href='dSVDSignif.html'>dSVDSignif</a></code>, <code><a href='dNetPipeline.html'>dNetPipeline</a></code>

    
    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) dnet: Expression-active dynamic networks for sample stratifications and visualisations. R package version 0.99.0. <a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></p>

    
    <!--
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>