<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>Docs @ dnet 1.0.1</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="Hai Fang and Julian Gough">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 1.0.1</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="rdata.html"><i class="icon-file icon-white"></i> RData</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <div class="row">
  <div class="span8">

    <h1>Documentations</h1><p>

    <h3>Network analysis and visualisation functions</h3>
    <p>These functions are used for network analysis and visualisation, including: identification of gene-active dynamic networks; network-based sample classifications and visualisations on 2D sample landscape; and random walk with restart (RWR) for network affinity calculation.</p>

    
    <ul class="index">
        
      <li>
        <code><a href="dNetPipeline.html">dNetPipeline</a></code><br />Function to setup the pipeline for finding maximum-scoring subgraph from an input graph and the signficance imposed on its nodes</li>
            
      <li>
        <code><a href="dNetReorder.html">dNetReorder</a></code><br />Function to reorder the multiple graph colorings within a sheet-shape rectangle grid</li>
            
      <li>
        <code><a href="dNetConfidence.html">dNetConfidence</a></code><br />Function to append the confidence information from the source graphs into the target graph</li>
            
      <li>
        <code><a href="dPvalAggregate.html">dPvalAggregate</a></code><br />Function to aggregate p values</li>
            
      <li>
        <code><a href="dNetInduce.html">dNetInduce</a></code><br />Function to generate a subgraph induced by given vertices and their k nearest neighbors</li>
            
      <li>
        <code><a href="dBUMfit.html">dBUMfit</a></code><br />Function to fit a p-value distribution under beta-uniform mixture model</li>
            
      <li>
        <code><a href="dBUMscore.html">dBUMscore</a></code><br />Function to transform p-values into scores according to the fitted beta-uniform mixture model and/or after controlling false discovery rate</li>
            
      <li>
        <code><a href="dSVDsignif.html">dSVDsignif</a></code><br />Function to obtain SVD-based gene significance from the input gene-sample matrix</li>
            
      <li>
        <code><a href="dFDRscore.html">dFDRscore</a></code><br />Function to transform fdr into scores according to log-likelihood ratio between the true positives and the false positivies and/or after controlling false discovery rate</li>
            
      <li>
        <code><a href="dNetFind.html">dNetFind</a></code><br />Function to find heuristically maximum scoring subgraph</li>
            
      <li>
        <code><a href="dCommSignif.html">dCommSignif</a></code><br />Function to test the significance of communities within a graph</li>
            
      <li>
        <code><a href="dContrast.html">dContrast</a></code><br />Function to help build the contrast matrix</li>
            
      <li>
        <code><a href="dRWR.html">dRWR</a></code><br />Function to implement Random Walk with Restart (RWR) on the input graph</li>
            
      <li>
        <code><a href="visNet.html">visNet</a></code><br />Function to visualise a graph object of class "igraph" or "graphNEL"</li>
            
      <li>
        <code><a href="visNetMul.html">visNetMul</a></code><br />Function to visualise the same graph but with multiple graph node colorings according to input data matrix</li>
            
      <li>
        <code><a href="visNetReorder.html">visNetReorder</a></code><br />Function to visualise the multiple graph colorings reorded within a sheet-shape rectangle grid</li>
            
      <li>
        <code><a href="visNetArc.html">visNetArc</a></code><br />Function to visualise an igraph object via arc diagram</li>
            
      <li>
        <code><a href="visNetCircle.html">visNetCircle</a></code><br />Function to visualise an igraph object via circle diagram</li>
            
      <li>
        <code><a href="visBoxplotAdv.html">visBoxplotAdv</a></code><br />Function to visualise a data frame using advanced boxplot</li>
    
    </ul>
    <h3>Enrichment analysis and visualisation functions</h3>
    <p>These functions are used for enrichment analysis and visualisation, including: enrichment analysis (Fisher&#39;s exact test, Hypergeometric test or Binomial test) that can account for the hierarchy of the ontology; and gene set enrichment analysis (GSEA).</p>

    
    <ul class="index">
        
      <li>
        <code><a href="dEnricher.html">dEnricher</a></code><br />Function to conduct enrichment analysis given the input data and the ontology in query</li>
            
      <li>
        <code><a href="dGSEA.html">dGSEA</a></code><br />Function to conduct gene set enrichment analysis given the input data and the ontology in query</li>
            
      <li>
        <code><a href="dGSEAview.html">dGSEAview</a></code><br />Function to view enrichment results in a sample-specific manner</li>
            
      <li>
        <code><a href="dGSEAwrite.html">dGSEAwrite</a></code><br />Function to write out enrichment results</li>
            
      <li>
        <code><a href="visGSEA.html">visGSEA</a></code><br />Function to visualise running enrichment score for a given sample and a gene set</li>
    
    </ul>
    <h3>Built-in ontologies, and analysis and visualisation functions</h3>
    <p>These ontologies each are represented as a direct acyclic graph (DAG). DAG is stored as an object of the class &#39;igraph&#39;, which can be processed by various analysis and visualisation functions.</p>

    
    <ul class="index">
        
      <li>
        <code><a href="ig.GOMF.html">ig.GOMF</a></code><br />Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="ig.GOBP.html">ig.GOBP</a></code><br />Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="ig.GOCC.html">ig.GOCC</a></code><br />Gene Ontology Cellular Component (GOCC).</li>
            
      <li>
        <code><a href="ig.HPPA.html">ig.HPPA</a></code><br />Human Phenotype Phenotypic Abnormality (HPPA).</li>
            
      <li>
        <code><a href="ig.HPON.html">ig.HPON</a></code><br />Human Phenotype ONset and clinical course (HPON).</li>
            
      <li>
        <code><a href="ig.HPMI.html">ig.HPMI</a></code><br />Human Phenotype Mode of Inheritance (HPMI).</li>
            
      <li>
        <code><a href="ig.DO.html">ig.DO</a></code><br />Disease Ontology (DO).</li>
            
      <li>
        <code><a href="ig.MP.html">ig.MP</a></code><br />Mammalian Phenotype (MP).</li>
            
      <li>
        <code><a href="dDAGinduce.html">dDAGinduce</a></code><br />Function to generate a subgraph of a direct acyclic graph (DAG) induced by given vertices</li>
            
      <li>
        <code><a href="dDAGreverse.html">dDAGreverse</a></code><br />Function to reverse the edge direction of a direct acyclic graph (DAG)</li>
            
      <li>
        <code><a href="dDAGroot.html">dDAGroot</a></code><br />Function to find the root node of a direct acyclic graph (DAG)</li>
            
      <li>
        <code><a href="dDAGtip.html">dDAGtip</a></code><br />Function to find the tip node(s) of a direct acyclic graph (DAG)</li>
            
      <li>
        <code><a href="dDAGlevel.html">dDAGlevel</a></code><br />Function to define/calculate the level of nodes in a direct acyclic graph (DAG)</li>
            
      <li>
        <code><a href="dDAGannotate.html">dDAGannotate</a></code><br />Function to generate a subgraph of a direct acyclic graph (DAG) induced by the input annotation data</li>
            
      <li>
        <code><a href="visDAG.html">visDAG</a></code><br />Function to visualise a direct acyclic graph (DAG) with node colorings according to a named input data vector (if provided)</li>
    
    </ul>
    <h3>Built-in databases in human</h3>
    <p>These databases are used for analysis (i.e. enrichment, evolution and network analysis) in human (Hs; tax_id=10090).</p>

    
    <ul class="index">
        
      <li>
        <code><a href="org.Hs.eg.html">org.Hs.eg</a></code><br />Human Entrez Genes (EG).</li>
            
      <li>
        <code><a href="org.Hs.egDO.html">org.Hs.egDO</a></code><br />Annotations of Human Entrez Genes (EG) by Disease Ontology (DO).</li>
            
      <li>
        <code><a href="org.Hs.egGOBP.html">org.Hs.egGOBP</a></code><br />Annotations of Human Entrez Genes (EG) by Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="org.Hs.egGOCC.html">org.Hs.egGOCC</a></code><br />Annotations of Human Entrez Genes (EG) by Gene Ontology Cellular Component (GOCC).</li>
            
      <li>
        <code><a href="org.Hs.egGOMF.html">org.Hs.egGOMF</a></code><br />Annotations of Human Entrez Genes (EG) by Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="org.Hs.egHPMI.html">org.Hs.egHPMI</a></code><br />Annotations of Human Entrez Genes (EG) by Human Phenotype Mode of Inheritance (HPMI).</li>
            
      <li>
        <code><a href="org.Hs.egHPON.html">org.Hs.egHPON</a></code><br />Annotations of Human Entrez Genes (EG) by Human Phenotype ONset and clinical course (HPON).</li>
            
      <li>
        <code><a href="org.Hs.egHPPA.html">org.Hs.egHPPA</a></code><br />Annotations of Human Entrez Genes (EG) by Human Phenotype Phenotypic Abnormality (HPPA).</li>
            
      <li>
        <code><a href="org.Hs.egMP.html">org.Hs.egMP</a></code><br />Annotations of Human Entrez Genes (EG) by Mammalian Phenotype (MP).</li>
            
      <li>
        <code><a href="org.Hs.egPS.html">org.Hs.egPS</a></code><br />Annotations of Human Entrez Genes (EG) by phylostratific age (PS).</li>
            
      <li>
        <code><a href="org.Hs.egSF.html">org.Hs.egSF</a></code><br />Annotations of Human Entrez Genes (EG) by domain superfamilies (SF).</li>
            
      <li>
        <code><a href="org.Hs.egDGIdb.html">org.Hs.egDGIdb</a></code><br />Annotations of Human Entrez Genes (EG) by DGIdb categories.</li>
            
      <li>
        <code><a href="org.Hs.string.html">org.Hs.string</a></code><br />Human functional protein association network from STRING (version 9.1).</li>
            
      <li>
        <code><a href="org.Hs.string900.html">org.Hs.string900</a></code><br />Human functional protein association network from STRING with highest confidence (no less than 900).</li>
    
    </ul>
    <h3>Built-in genesets in human</h3>
    <p>These genesets are derived from the molecular signatures database (Msigdb) in human (Hs).</p>

    
    <ul class="index">
        
      <li>
        <code><a href="org.Hs.egMsigdbC1.html">org.Hs.egMsigdbC1</a></code><br />Annotations of Human Entrez Genes (EG) by C1 collections.</li>
            
      <li>
        <code><a href="org.Hs.egMsigdbC2CGP.html">org.Hs.egMsigdbC2CGP</a></code><br />Annotations of Human Entrez Genes (EG) by C2:CGP collections.</li>
            
      <li>
        <code><a href="org.Hs.egMsigdbC2CP.html">org.Hs.egMsigdbC2CP</a></code><br />Annotations of Human Entrez Genes (EG) by C2:CP collections.</li>
            
      <li>
        <code><a href="org.Hs.egMsigdbC2KEGG.html">org.Hs.egMsigdbC2KEGG</a></code><br />Annotations of Human Entrez Genes (EG) by C2:KEGG collections.</li>
            
      <li>
        <code><a href="org.Hs.egMsigdbC2REACTOME.html">org.Hs.egMsigdbC2REACTOME</a></code><br />Annotations of Human Entrez Genes (EG) by C2:REACTOME collections.</li>
            
      <li>
        <code><a href="org.Hs.egMsigdbC2BIOCARTA.html">org.Hs.egMsigdbC2BIOCARTA</a></code><br />Annotations of Human Entrez Genes (EG) by C2:BIOCARTA collections.</li>
            
      <li>
        <code><a href="org.Hs.egMsigdbC3TFT.html">org.Hs.egMsigdbC3TFT</a></code><br />Annotations of Human Entrez Genes (EG) by C3:TFT collections.</li>
            
      <li>
        <code><a href="org.Hs.egMsigdbC3MIR.html">org.Hs.egMsigdbC3MIR</a></code><br />Annotations of Human Entrez Genes (EG) by C3:MIR collections.</li>
            
      <li>
        <code><a href="org.Hs.egMsigdbC4CGN.html">org.Hs.egMsigdbC4CGN</a></code><br />Annotations of Human Entrez Genes (EG) by C4:CGN collections.</li>
            
      <li>
        <code><a href="org.Hs.egMsigdbC4CM.html">org.Hs.egMsigdbC4CM</a></code><br />Annotations of Human Entrez Genes (EG) by C4:CM collections.</li>
            
      <li>
        <code><a href="org.Hs.egMsigdbC5BP.html">org.Hs.egMsigdbC5BP</a></code><br />Annotations of Human Entrez Genes (EG) by C5:BP collections.</li>
            
      <li>
        <code><a href="org.Hs.egMsigdbC5MF.html">org.Hs.egMsigdbC5MF</a></code><br />Annotations of Human Entrez Genes (EG) by C5:MF collections.</li>
            
      <li>
        <code><a href="org.Hs.egMsigdbC5CC.html">org.Hs.egMsigdbC5CC</a></code><br />Annotations of Human Entrez Genes (EG) by C5:CC collections.</li>
            
      <li>
        <code><a href="org.Hs.egMsigdbC6.html">org.Hs.egMsigdbC6</a></code><br />Annotations of Human Entrez Genes (EG) by C6 collections.</li>
            
      <li>
        <code><a href="org.Hs.egMsigdbC7.html">org.Hs.egMsigdbC7</a></code><br />Annotations of Human Entrez Genes (EG) by C7 collections.</li>
    
    </ul>
    <h3>Built-in databases in mouse</h3>
    <p>These databases are used for analysis (i.e. enrichment, evolution and network analysis) in mouse (Mm; tax_id=10090).</p>

    
    <ul class="index">
        
      <li>
        <code><a href="org.Mm.eg.html">org.Mm.eg</a></code><br />Mouse Entrez Genes (EG).</li>
            
      <li>
        <code><a href="org.Mm.egDO.html">org.Mm.egDO</a></code><br />Annotations of Mouse Entrez Genes (EG) by Disease Ontology (DO).</li>
            
      <li>
        <code><a href="org.Mm.egGOBP.html">org.Mm.egGOBP</a></code><br />Annotations of Mouse Entrez Genes (EG) by Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="org.Mm.egGOCC.html">org.Mm.egGOCC</a></code><br />Annotations of Mouse Entrez Genes (EG) by Gene Ontology Cellular Component (GOCC).</li>
            
      <li>
        <code><a href="org.Mm.egGOMF.html">org.Mm.egGOMF</a></code><br />Annotations of Mouse Entrez Genes (EG) by Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="org.Mm.egHPMI.html">org.Mm.egHPMI</a></code><br />Annotations of Mouse Entrez Genes (EG) by Human Phenotype Mode of Inheritance (HPMI).</li>
            
      <li>
        <code><a href="org.Mm.egHPON.html">org.Mm.egHPON</a></code><br />Annotations of Mouse Entrez Genes (EG) by Human Phenotype ONset and clinical course (HPON).</li>
            
      <li>
        <code><a href="org.Mm.egHPPA.html">org.Mm.egHPPA</a></code><br />Annotations of Mouse Entrez Genes (EG) by Human Phenotype Phenotypic Abnormality (HPPA).</li>
            
      <li>
        <code><a href="org.Mm.egMP.html">org.Mm.egMP</a></code><br />Annotations of Mouse Entrez Genes (EG) by Mammalian Phenotype (MP).</li>
            
      <li>
        <code><a href="org.Mm.egPS.html">org.Mm.egPS</a></code><br />Annotations of Mouse Entrez Genes (EG) by phylostratific age (PS).</li>
            
      <li>
        <code><a href="org.Mm.egSF.html">org.Mm.egSF</a></code><br />Annotations of Mouse Entrez Genes (EG) by domain superfamilies (SF).</li>
            
      <li>
        <code><a href="org.Mm.string.html">org.Mm.string</a></code><br />Mouse functional protein association network from STRING (version 9.1).</li>
    
    </ul>
    <h3>Built-in databases in arabidopsis</h3>
    <p>These databases are used for analysis (i.e. enrichment, evolution and network analysis) in arabidopsis (At; tax_id=3702).</p>

    
    <ul class="index">
        
      <li>
        <code><a href="org.At.eg.html">org.At.eg</a></code><br />Arabidopsis Entrez Genes (EG).</li>
            
      <li>
        <code><a href="org.At.egGOBP.html">org.At.egGOBP</a></code><br />Annotations of Arabidopsis Entrez Genes (EG) by Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="org.At.egGOCC.html">org.At.egGOCC</a></code><br />Annotations of Arabidopsis Entrez Genes (EG) by Gene Ontology Cellular Component (GOCC).</li>
            
      <li>
        <code><a href="org.At.egGOMF.html">org.At.egGOMF</a></code><br />Annotations of Arabidopsis Entrez Genes (EG) by Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="org.At.egPS.html">org.At.egPS</a></code><br />Annotations of Arabidopsis Entrez Genes (EG) by phylostratific age (PS).</li>
            
      <li>
        <code><a href="org.At.egSF.html">org.At.egSF</a></code><br />Annotations of Arabidopsis Entrez Genes (EG) by domain superfamilies (SF).</li>
            
      <li>
        <code><a href="org.At.string.html">org.At.string</a></code><br />Arabidopsis functional protein association network from STRING (version 9.1).</li>
    
    </ul>
    <h3>Built-in databases in c.elegans</h3>
    <p>These databases are used for analysis (i.e. enrichment, evolution and network analysis) in c.elegans (Ce; tax_id=6239).</p>

    
    <ul class="index">
        
      <li>
        <code><a href="org.Ce.eg.html">org.Ce.eg</a></code><br />C.elegans Entrez Genes (EG).</li>
            
      <li>
        <code><a href="org.Ce.egGOBP.html">org.Ce.egGOBP</a></code><br />Annotations of C.elegans Entrez Genes (EG) by Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="org.Ce.egGOCC.html">org.Ce.egGOCC</a></code><br />Annotations of C.elegans Entrez Genes (EG) by Gene Ontology Cellular Component (GOCC).</li>
            
      <li>
        <code><a href="org.Ce.egGOMF.html">org.Ce.egGOMF</a></code><br />Annotations of C.elegans Entrez Genes (EG) by Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="org.Ce.egPS.html">org.Ce.egPS</a></code><br />Annotations of C.elegans Entrez Genes (EG) by phylostratific age (PS).</li>
            
      <li>
        <code><a href="org.Ce.egSF.html">org.Ce.egSF</a></code><br />Annotations of C.elegans Entrez Genes (EG) by domain superfamilies (SF).</li>
            
      <li>
        <code><a href="org.Ce.string.html">org.Ce.string</a></code><br />C.elegans functional protein association network from STRING (version 9.1).</li>
    
    </ul>
    <h3>Built-in databases in fruitfly</h3>
    <p>These databases are used for analysis (i.e. enrichment, evolution and network analysis) in fruitfly (Dm; tax_id=7227).</p>

    
    <ul class="index">
        
      <li>
        <code><a href="org.Dm.eg.html">org.Dm.eg</a></code><br />Fruitfly Entrez Genes (EG).</li>
            
      <li>
        <code><a href="org.Dm.egGOBP.html">org.Dm.egGOBP</a></code><br />Annotations of Fruitfly Entrez Genes (EG) by Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="org.Dm.egGOCC.html">org.Dm.egGOCC</a></code><br />Annotations of Fruitfly Entrez Genes (EG) by Gene Ontology Cellular Component (GOCC).</li>
            
      <li>
        <code><a href="org.Dm.egGOMF.html">org.Dm.egGOMF</a></code><br />Annotations of Fruitfly Entrez Genes (EG) by Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="org.Dm.egPS.html">org.Dm.egPS</a></code><br />Annotations of Fruitfly Entrez Genes (EG) by phylostratific age (PS).</li>
            
      <li>
        <code><a href="org.Dm.egSF.html">org.Dm.egSF</a></code><br />Annotations of Fruitfly Entrez Genes (EG) by domain superfamilies (SF).</li>
            
      <li>
        <code><a href="org.Dm.string.html">org.Dm.string</a></code><br />Fruitfly functional protein association network from STRING (version 9.1).</li>
    
    </ul>
    <h3>Built-in databases in zebrafish</h3>
    <p>These databases are used for analysis (i.e. enrichment, evolution and network analysis) in zebrafish (Da; tax_id=7955).</p>

    
    <ul class="index">
        
      <li>
        <code><a href="org.Da.eg.html">org.Da.eg</a></code><br />Zebrafish Entrez Genes (EG).</li>
            
      <li>
        <code><a href="org.Da.egGOBP.html">org.Da.egGOBP</a></code><br />Annotations of Zebrafish Entrez Genes (EG) by Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="org.Da.egGOCC.html">org.Da.egGOCC</a></code><br />Annotations of Zebrafish Entrez Genes (EG) by Gene Ontology Cellular Component (GOCC).</li>
            
      <li>
        <code><a href="org.Da.egGOMF.html">org.Da.egGOMF</a></code><br />Annotations of Zebrafish Entrez Genes (EG) by Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="org.Da.egPS.html">org.Da.egPS</a></code><br />Annotations of Zebrafish Entrez Genes (EG) by phylostratific age (PS).</li>
            
      <li>
        <code><a href="org.Da.egSF.html">org.Da.egSF</a></code><br />Annotations of Zebrafish Entrez Genes (EG) by domain superfamilies (SF).</li>
            
      <li>
        <code><a href="org.Da.string.html">org.Da.string</a></code><br />Zebrafish functional protein association network from STRING (version 9.1).</li>
    
    </ul>
    <h3>Built-in databases in rat</h3>
    <p>These databases are used for analysis (i.e. enrichment, evolution and network analysis) in rat (Rn; tax_id=10116).</p>

    
    <ul class="index">
        
      <li>
        <code><a href="org.Rn.eg.html">org.Rn.eg</a></code><br />Rat Entrez Genes (EG).</li>
            
      <li>
        <code><a href="org.Rn.egGOBP.html">org.Rn.egGOBP</a></code><br />Annotations of Rat Entrez Genes (EG) by Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="org.Rn.egGOCC.html">org.Rn.egGOCC</a></code><br />Annotations of Rat Entrez Genes (EG) by Gene Ontology Cellular Component (GOCC).</li>
            
      <li>
        <code><a href="org.Rn.egGOMF.html">org.Rn.egGOMF</a></code><br />Annotations of Rat Entrez Genes (EG) by Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="org.Rn.egPS.html">org.Rn.egPS</a></code><br />Annotations of Rat Entrez Genes (EG) by phylostratific age (PS).</li>
            
      <li>
        <code><a href="org.Rn.egSF.html">org.Rn.egSF</a></code><br />Annotations of Rat Entrez Genes (EG) by domain superfamilies (SF).</li>
            
      <li>
        <code><a href="org.Rn.string.html">org.Rn.string</a></code><br />Rat functional protein association network from STRING (version 9.1).</li>
    
    </ul>
    <h3>Built-in databases in chicken</h3>
    <p>These databases are used for analysis (i.e. enrichment, evolution and network analysis) in chicken (Gg; tax_id=9031).</p>

    
    <ul class="index">
        
      <li>
        <code><a href="org.Gg.eg.html">org.Gg.eg</a></code><br />Chicken Entrez Genes (EG).</li>
            
      <li>
        <code><a href="org.Gg.egGOBP.html">org.Gg.egGOBP</a></code><br />Annotations of Chicken Entrez Genes (EG) by Gene Ontology Biological Process (GOBP).</li>
            
      <li>
        <code><a href="org.Gg.egGOCC.html">org.Gg.egGOCC</a></code><br />Annotations of Chicken Entrez Genes (EG) by Gene Ontology Cellular Component (GOCC).</li>
            
      <li>
        <code><a href="org.Gg.egGOMF.html">org.Gg.egGOMF</a></code><br />Annotations of Chicken Entrez Genes (EG) by Gene Ontology Molecular Function (GOMF).</li>
            
      <li>
        <code><a href="org.Gg.egPS.html">org.Gg.egPS</a></code><br />Annotations of Chicken Entrez Genes (EG) by phylostratific age (PS).</li>
            
      <li>
        <code><a href="org.Gg.egSF.html">org.Gg.egSF</a></code><br />Annotations of Chicken Entrez Genes (EG) by domain superfamilies (SF).</li>
            
      <li>
        <code><a href="org.Gg.string.html">org.Gg.string</a></code><br />Chicken functional protein association network from STRING (version 9.1).</li>
    
    </ul>
    <h3>Built-in datasets</h3>
    <p>These datasets are used for the demos.</p>

    
    <ul class="index">
        
      <li>
        <code><a href="Hiratani_TableS1.html">Hiratani_TableS1</a></code>(RT, CpG, EX)<br />Mouse multilayer omics dataset from Hiratani et al. (2010)</li>
            
      <li>
        <code><a href="CLL.html">CLL</a></code><br />Transcriptome dataset for Chronic Lymphocytic Leukemia patients from Chuang et al. (2012)</li>
            
      <li>
        <code><a href="TCGA_mutations.html">TCGA_mutations</a></code><br />TCGA mutational profiles across 12 major cancer types from Kandoth et al. (2013)</li>
    
    </ul>
  </div>
  
  
  <div class="span4">
    
    <!--
    
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></li>
    </ul>
    
    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) DNET: dynamic networks via integrative analysis of digitised data in terms of network, ontology and evolution. R package version 1.0.1. <a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></p>

    
    <h2>Author</h2>
    Hai Fang and Julian Gough
    
    -->
    
  </div>
  
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>