<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>dRDataLoader @ dnet 1.0.7</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 1.0.7</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="rdata.html"><i class="icon-file icon-white"></i> RData</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="faqs.html"><i class="icon-question-sign icon-white"></i> FAQs</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dnet R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:djnz9oat1hu';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to load dnet built-in RData</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>dRDataLoader</code> is supposed to load the package built-in RData.</p>
  
    <h3>Usage</h3>
    <pre>dRDataLoader(RData = c(NA, "TCGA_mutations", "ig.DO", "ig.GOBP", "ig.GOCC", "ig.GOMF", 
  "ig.HPCM", "ig.HPMA", "ig.HPMI", "ig.HPPA", "ig.MP", "org.At.eg", "org.At.egGOBP", 
      "org.At.egGOCC", "org.At.egGOMF", "org.At.egPS", "org.At.egSF", "org.At.string", 
      "org.Ce.eg", "org.Ce.egGOBP", "org.Ce.egGOCC", "org.Ce.egGOMF", "org.Ce.egPS", 
      "org.Ce.egSF", "org.Ce.string", "org.Da.eg", "org.Da.egGOBP", "org.Da.egGOCC", 
      "org.Da.egGOMF", "org.Da.egPS", "org.Da.egSF", "org.Da.string", "org.Dm.eg", 
      "org.Dm.egGOBP", "org.Dm.egGOCC", "org.Dm.egGOMF", "org.Dm.egPS", "org.Dm.egSF", 
      "org.Dm.string", "org.Gg.eg", "org.Gg.egGOBP", "org.Gg.egGOCC", "org.Gg.egGOMF", 
      "org.Gg.egPS", "org.Gg.egSF", "org.Gg.string", "org.Hs.eg", "org.Hs.egDGIdb", 
      "org.Hs.egDO", "org.Hs.egGOBP", "org.Hs.egGOCC", "org.Hs.egGOMF", "org.Hs.egHPCM", 
      "org.Hs.egHPMA", "org.Hs.egHPMI", "org.Hs.egHPPA", "org.Hs.egMP", "org.Hs.egMsigdbC1", 
      "org.Hs.egMsigdbC2BIOCARTA", "org.Hs.egMsigdbC2CGP", "org.Hs.egMsigdbC2CP", "org.Hs.egMsigdbC2KEGG", 
      "org.Hs.egMsigdbC2REACTOME", "org.Hs.egMsigdbC3MIR", "org.Hs.egMsigdbC3TFT", 
      "org.Hs.egMsigdbC4CGN", "org.Hs.egMsigdbC4CM", "org.Hs.egMsigdbC5BP", "org.Hs.egMsigdbC5CC", 
      "org.Hs.egMsigdbC5MF", "org.Hs.egMsigdbC6", "org.Hs.egMsigdbC7", "org.Hs.egMsigdbH", 
      "org.Hs.egPS", "org.Hs.egSF", "org.Hs.string", "org.Mm.eg", "org.Mm.egDO", "org.Mm.egGOBP", 
      "org.Mm.egGOCC", "org.Mm.egGOMF", "org.Mm.egHPCM", "org.Mm.egHPMA", "org.Mm.egHPMI", 
      "org.Mm.egHPPA", "org.Mm.egMP", "org.Mm.egPS", "org.Mm.egSF", "org.Mm.string", 
      "org.Rn.eg", "org.Rn.egGOBP", "org.Rn.egGOCC", "org.Rn.egGOMF", "org.Rn.egPS", 
      "org.Rn.egSF", "CLL", "org.Rn.string"), genome = c(NA, "Hs", "Mm", "Rn", "Gg", 
      "Ce", "Dm", "Da", "At"), ontology = c(NA, "GOBP", "GOMF", "GOCC", "PS", "PS2", 
      "SF", "DO", "HPPA", "HPMI", "HPCM", "HPMA", "MP", "MsigdbH", "MsigdbC1", "MsigdbC2CGP", 
      "MsigdbC2CP", "MsigdbC2KEGG", "MsigdbC2REACTOME", "MsigdbC2BIOCARTA", "MsigdbC3TFT", 
      "MsigdbC3MIR", "MsigdbC4CGN", "MsigdbC4CM", "MsigdbC5BP", "MsigdbC5MF", "MsigdbC5CC", 
      "MsigdbC6", "MsigdbC7", "DGIdb"), verbose = T, RData.location = "https://github.com/hfang-bristol/RDataCentre/blob/master/dnet/1.0.7")</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>RData</dt>
      <dd>which built-in RData to load. It can be one of
"TCGA_mutations", "ig.DO", "ig.GOBP", "ig.GOCC", "ig.GOMF", "ig.HPCM",
"ig.HPMA", "ig.HPMI", "ig.HPPA", "ig.MP", "org.At.eg", "org.At.egGOBP",
"org.At.egGOCC", "org.At.egGOMF", "org.At.egPS", "org.At.egSF",
"org.At.string", "org.Ce.eg", "org.Ce.egGOBP", "org.Ce.egGOCC",
"org.Ce.egGOMF", "org.Ce.egPS", "org.Ce.egSF", "org.Ce.string",
"org.Da.eg", "org.Da.egGOBP", "org.Da.egGOCC", "org.Da.egGOMF",
"org.Da.egPS", "org.Da.egSF", "org.Da.string", "org.Dm.eg",
"org.Dm.egGOBP", "org.Dm.egGOCC", "org.Dm.egGOMF", "org.Dm.egPS",
"org.Dm.egSF", "org.Dm.string", "org.Gg.eg", "org.Gg.egGOBP",
"org.Gg.egGOCC", "org.Gg.egGOMF", "org.Gg.egPS", "org.Gg.egSF",
"org.Gg.string", "org.Hs.eg", "org.Hs.egDGIdb", "org.Hs.egDO",
"org.Hs.egGOBP", "org.Hs.egGOCC", "org.Hs.egGOMF", "org.Hs.egHPCM",
"org.Hs.egHPMA", "org.Hs.egHPMI", "org.Hs.egHPPA", "org.Hs.egMP",
"org.Hs.egMsigdbC1", "org.Hs.egMsigdbC2BIOCARTA",
"org.Hs.egMsigdbC2CGP", "org.Hs.egMsigdbC2CP", "org.Hs.egMsigdbC2KEGG",
"org.Hs.egMsigdbC2REACTOME", "org.Hs.egMsigdbC3MIR",
"org.Hs.egMsigdbC3TFT", "org.Hs.egMsigdbC4CGN", "org.Hs.egMsigdbC4CM",
"org.Hs.egMsigdbC5BP", "org.Hs.egMsigdbC5CC", "org.Hs.egMsigdbC5MF",
"org.Hs.egMsigdbC6", "org.Hs.egMsigdbC7", "org.Hs.egMsigdbH",
"org.Hs.egPS", "org.Hs.egSF", "org.Hs.string", "org.Mm.eg",
"org.Mm.egDO", "org.Mm.egGOBP", "org.Mm.egGOCC", "org.Mm.egGOMF",
"org.Mm.egHPCM", "org.Mm.egHPMA", "org.Mm.egHPMI", "org.Mm.egHPPA",
"org.Mm.egMP", "org.Mm.egPS", "org.Mm.egSF", "org.Mm.string",
"org.Rn.eg", "org.Rn.egGOBP", "org.Rn.egGOCC", "org.Rn.egGOMF",
"org.Rn.egPS", "org.Rn.egSF", "CLL", "org.Rn.string". On the meanings,
please refer to the Documentations at
<a href = 'http://supfam.org/dnet/docs.html'>http://supfam.org/dnet/docs.html</a></dd>
      <dt>genome</dt>
      <dd>the genome identity. It can be one of "Hs" for human,
"Mm" for mouse, "Rn" for rat, "Gg" for chicken, "Ce" for c.elegans,
"Dm" for fruitfly, "Da" for zebrafish, and "At" for arabidopsis</dd>
      <dt>ontology</dt>
      <dd>the ontology supported currently. It can be "GOBP" for
Gene Ontology Biological Process, "GOMF" for Gene Ontology Molecular
Function, "GOCC" for Gene Ontology Cellular Component, "PS" for
phylostratific age information, "PS2" for the collapsed PS version
(inferred ancestors being collapsed into one with the known taxonomy
information), "SF" for domain superfamily assignments, "DO" for Disease
Ontology, "HPPA" for Human Phenotype Phenotypic Abnormality, "HPMI" for
Human Phenotype Mode of Inheritance, "HPCM" for Human Phenotype
Clinical Modifier, "HPMA" for Human Phenotype Mortality Aging, "MP" for
Mammalian Phenotype, and Drug-Gene Interaction database (DGIdb) and the
molecular signatures database (Msigdb) only in human (including
"MsigdbH", "MsigdbC1", "MsigdbC2CGP", "MsigdbC2CP", "MsigdbC2KEGG",
"MsigdbC2REACTOME", "MsigdbC2BIOCARTA", "MsigdbC3TFT", "MsigdbC3MIR",
"MsigdbC4CGN", "MsigdbC4CM", "MsigdbC5BP", "MsigdbC5MF", "MsigdbC5CC",
"MsigdbC6", "MsigdbC7"). Note: These four ("GOBP", "GOMF", "GOCC" and
"PS") are availble for all genomes/species; for "Hs" and "Mm", these
six ("DO", "HPPA", "HPMI", "HPCM", "HPMA" and "MP") are also supported;
all "Msigdb" are only supported in "Hs". For details on the eligibility
for pairs of input genome and ontology, please refer to the online
Documentations at <a href = 'http://supfam.org/dnet/docs.html'>http://supfam.org/dnet/docs.html</a></dd>
      <dt>verbose</dt>
      <dd>logical to indicate whether the messages will be
displayed in the screen. By default, it sets to TRUE for display</dd>
      <dt>RData.location</dt>
      <dd>the characters to tell the location of built-in
RData files. By default, it remotely locates at
<a href = 'https://github.com/hfang-bristol/RDataCentre/blob/master/dnet'>https://github.com/hfang-bristol/RDataCentre/blob/master/dnet</a> and
<a href = 'http://dnet.r-forge.r-project.org/RData'>http://dnet.r-forge.r-project.org/RData</a>. Be aware of several
versions and the latest one is matched to the current package version.
For the user equipped with fast internet connection, this option can be
just left as default. But it is always advisable to download these
files locally. Especially when the user needs to run this function many
times, there is no need to ask the function to remotely download every
time (also it will unnecessarily increase the runtime). For examples,
these files (as a whole or part of them) can be first downloaded into
your current working directory, and then set this option as:
<code class = 'eq'>RData.location="."</code>. Surely, the location can be anywhere as long
as the user provides the correct path pointing to (otherwise, the
script will have to remotely download each time). Here is the UNIX
command for downloading all RData files (preserving the directory
structure): <code class = 'eq'>wget -r -l2 -A "*.RData" -np -nH --cut-dirs=0
"http://dnet.r-forge.r-project.org/RData"</code></dd>
    </dl>
    <!-- 
    -->
    
    <!-- 
    <div class="Value">
    -->
      <h3>Value</h3>
      
      <p>any use-specified variable that is given on the right side of the
assigement sign '<-', which contains the loaded RData.</p>
    <!-- 
    </div>
    -->
    <!-- 
    <div class="Note">
    -->
      <h3>Note</h3>
      
      <p>If there are no use-specified variable that is given on the right side
of the assigement sign '<-', then no RData will be loaded onto the
working environment.</p>
    <!-- 
    </div>
    -->
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'>org.Hs.egSF <- dRDataLoader(RData='org.Hs.egSF')
</div>
<strong class='message'>'org.Hs.egSF' (from https://github.com/hfang-bristol/RDataCentre/blob/master/dnet/1.0.7/org.Hs.egSF.RData?raw=true) has been loaded into the working environment (at 2015-07-22 23:31:42)</strong>
<div class='input'>org.Hs.eg <- dRDataLoader(RData='org.Hs.eg')
</div>
<strong class='message'>'org.Hs.eg' (from https://github.com/hfang-bristol/RDataCentre/blob/master/dnet/1.0.7/org.Hs.eg.RData?raw=true) has been loaded into the working environment (at 2015-07-22 23:31:45)</strong>
<div class='input'>org.Hs.egDGIdb <- dRDataLoader(RData='org.Hs.egDGIdb')
</div>
<strong class='message'>'org.Hs.egDGIdb' (from https://github.com/hfang-bristol/RDataCentre/blob/master/dnet/1.0.7/org.Hs.egDGIdb.RData?raw=true) has been loaded into the working environment (at 2015-07-22 23:31:46)</strong>
<div class='input'>org.Hs.egMsigdbC2KEGG <- dRDataLoader(RData='org.Hs.egMsigdbC2KEGG')
</div>
<strong class='message'>'org.Hs.egMsigdbC2KEGG' (from https://github.com/hfang-bristol/RDataCentre/blob/master/dnet/1.0.7/org.Hs.egMsigdbC2KEGG.RData?raw=true) has been loaded into the working environment (at 2015-07-22 23:31:47)</strong>
<div class='input'>org.Hs.egHPPA <- dRDataLoader(genome='Hs', ontology='HPPA')
</div>
<strong class='message'>'org.Hs.egHPPA' (from package 'dnet' version 1.0.7) has been loaded into the working environment (at 2015-07-22 23:31:47)</strong>
<div class='input'>ig.MP <- dRDataLoader(RData='ig.MP')
</div>
<strong class='message'>'ig.MP' (from https://github.com/hfang-bristol/RDataCentre/blob/master/dnet/1.0.7/ig.MP.RData?raw=true) has been loaded into the working environment (at 2015-07-22 23:31:50)</strong></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>dRDataLoader</li>
    </ul>
    <ul>
      
    </ul> 
    -->
    
    <h2>Source code</h2>
    <a href="javascript:newWin('R/dRDataLoader.r', 'R/dRDataLoader.r', '1200', '1200')" title="Click to view"><B><code>dRDataLoader.r</code></B></a>
    
    <h2>Source man</h2>
    <a href="javascript:newWin('man/dRDataLoader.txt', 'man/dRDataLoader.txt', '1200', '1200')" title="Click to view"><B><code>dRDataLoader.Rd</code></B></a>
    
    <a href="javascript:newWin('man/dRDataLoader.pdf', 'man/dRDataLoader.pdf', '1200', '1200')" title="Click to view"><B><code>dRDataLoader.pdf</code></B></a>
    
    <h2>See also</h2>
    
<code><a href='dRDataLoader.html'>dRDataLoader</a></code>

    <!--

    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) <B>The &#39;dnet&#39; approach promotes emerging research on cancer patient survival</B>. <I>Genome Medicine</I>, 6(8):64. <a href="http://dx.doi.org/10.1186/s13073-014-0064-8">doi:10.1186/s13073-014-0064-8</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/25246945">25246945</a></p>

<p>You may also cite the package as:</p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dnet&rdquo; [<a href="http://cran.r-project.org/package=dnet">http://cran.r-project.org/package=dnet</a>]</p>

    
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dnet">http://supfam.org/dnet</a></li>
      <li><a href=" http://dnet.r-forge.r-project.org"> http://dnet.r-forge.r-project.org</a></li>
      <li><a href=" http://cran.r-
project.org/package=dnet"> http://cran.r-
project.org/package=dnet</a></li>
      <li><a href=" https://github.com/hfang-bristol/dnet"> https://github.com/hfang-bristol/dnet</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@well.ox.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>