<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>dRWRpipeline @ dnet 1.0.6</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 1.0.6</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="rdata.html"><i class="icon-file icon-white"></i> RData</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="faqs.html"><i class="icon-question-sign icon-white"></i> FAQs</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dnet R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:djnz9oat1hu';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to setup a pipeine to estimate RWR-based contact strength between samples from an input gene-sample data matrix and an input graph</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>dRWRpipeline</code> is supposed to estimate sample relationships (ie.
contact strength between samples) from an input gene-sample matrix and
an input graph. The pipeline includes: 1) random walk restart (RWR) of
the input graph using the input matrix as seeds; 2) calculation of
contact strength (inner products of RWR-smoothed columns of input
matrix); 3) estimation of the contact signficance by a randomalisation
procedure. It supports two methods how to use RWR: 'direct' for
directly applying RWR in the given seeds; 'indirectly' for first
pre-computing affinity matrix of the input graph, and then deriving the
affinity score. Parallel computing is also supported for Linux or Mac
operating systems.</p>
  
    <h3>Usage</h3>
    <pre>dRWRpipeline(data, g, method = c("direct", "indirect"), normalise = c("laplacian", 
  "row", "column", "none"), restart = 0.75, normalise.affinity.matrix = c("none", 
      "quantile"), permutation = c("random", "degree"), num.permutation = 10, p.adjust.method = c("BH", 
      "BY", "bonferroni", "holm", "hochberg", "hommel"), adjp.cutoff = 0.05, parallel = TRUE, 
      multicores = NULL, verbose = T)</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>data</dt>
      <dd>an input gene-sample data matrix used for seeds. Each value
in input gene-sample matrix does not necessarily have to be binary
(non-zeros will be used as a weight, but should be non-negative for
easy interpretation).</dd>
      <dt>g</dt>
      <dd>an object of class "igraph" or "graphNEL"</dd>
      <dt>method</dt>
      <dd>the method used to calculate RWR. It can be 'direct' for
directly applying RWR, 'indirect' for indirectly applying RWR (first
pre-compute affinity matrix and then derive the affinity score)</dd>
      <dt>normalise</dt>
      <dd>the way to normalise the adjacency matrix of the input
graph. It can be 'laplacian' for laplacian normalisation, 'row' for
row-wise normalisation, 'column' for column-wise normalisation, or
'none'</dd>
      <dt>restart</dt>
      <dd>the restart probability used for RWR. The restart
probability takes the value from 0 to 1, controlling the range from the
starting nodes/seeds that the walker will explore. The higher the
value, the more likely the walker is to visit the nodes centered on the
starting nodes. At the extreme when the restart probability is zero,
the walker moves freely to the neighbors at each step without
restarting from seeds, i.e., following a random walk (RW)</dd>
      <dt>normalise.affinity.matrix</dt>
      <dd>the way to normalise the output
affinity matrix. It can be 'none' for no normalisation, 'quantile' for
quantile normalisation to ensure that columns (if multiple) of the
output affinity matrix have the same quantiles</dd>
      <dt>permutation</dt>
      <dd>how to do permutation. It can be 'degree' for
degree-preserving permutation, 'random' for permutation in random</dd>
      <dt>num.permutation</dt>
      <dd>the number of permutations used to for
generating the distribution of contact strength under randomalisation</dd>
      <dt>p.adjust.method</dt>
      <dd>the method used to adjust p-values. It can be
one of "BH", "BY", "bonferroni", "holm", "hochberg" and "hommel". The
first two methods "BH" (widely used) and "BY" control the false
discovery rate (FDR: the expected proportion of false discoveries
amongst the rejected hypotheses); the last four methods "bonferroni",
"holm", "hochberg" and "hommel" are designed to give strong control of
the family-wise error rate (FWER). Notes: FDR is a less stringent
condition than FWER</dd>
      <dt>adjp.cutoff</dt>
      <dd>the cutoff of adjusted pvalue to construct the
contact graph</dd>
      <dt>parallel</dt>
      <dd>logical to indicate whether parallel computation with
multicores is used. By default, it sets to true, but not necessarily
does so. Partly because parallel backends available will be
system-specific (now only Linux or Mac OS). Also, it will depend on
whether these two packages "foreach" and "doMC" have been installed. It
can be installed via:
<code>source("http://bioconductor.org/biocLite.R");
biocLite(c("foreach","doMC"))</code>. If not yet installed, this option will
be disabled</dd>
      <dt>multicores</dt>
      <dd>an integer to specify how many cores will be
registered as the multicore parallel backend to the 'foreach' package.
If NULL, it will use a half of cores available in a user's computer.
This option only works when parallel computation is enabled</dd>
      <dt>verbose</dt>
      <dd>logical to indicate whether the messages will be
displayed in the screen. By default, it sets to true for display</dd>
    </dl>
    <!-- 
    -->
    
    <!-- 
    <div class="Value">
    -->
      <h3>Value</h3>
      
      <p>an object of class "dContact", a list with following components:
<ul>
<li><code>ratio</code>: a symmetric matrix storing ratio (the observed
against the expected) between pairwise samples
</li>
<li><code>zscore</code>: a symmetric matrix storing zscore between pairwise
samples
</li>
<li><code>pval</code>: a symmetric matrix storing pvalue between pairwise
samples
</li>
<li><code>adjpval</code>: a symmetric matrix storing adjusted pvalue
between pairwise samples
</li>
<li><code>cgraph</code>: the constructed contact graph (as a 'igraph'
object) under the cutoff of adjusted value
</li>
<li><code>Amatrix</code>: a pre-computated affinity matrix when using
'inderect' method; NULL otherwise
</li>
<li><code>call</code>: the call that produced this result
</li>
</ul></p>
      <p></p>
    <!-- 
    </div>
    -->
    <!-- 
    <div class="Note">
    -->
      <h3>Note</h3>
      
      <p>The choice of which method to use RWR depends on the number of seed
sets and the number of permutations for statistical test. If the total
product of both numbers are huge, it is better to use 'indrect' method
(for a single run). However, if the user wants to re-use pre-computed
affinity matrix (ie. re-use the input graph a lot), then it is highly
recommended to sequentially use <code><a href='dRWR.html'>dRWR</a></code> and
<code><a href='dRWRcontact.html'>dRWRcontact</a></code> instead.</p>
    <!-- 
    </div>
    -->
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'># 1) generate a random graph according to the ER model
g <- erdos.renyi.game(100, 1/100)

# 2) produce the induced subgraph only based on the nodes in query
subg <- dNetInduce(g, V(g), knn=0)
V(subg)$name <- 1:vcount(subg)

# 3) estimate RWR dating based sample relationships
# define sets of seeds as data
# each seed with equal weight (i.e. all non-zero entries are '1')
aSeeds <- c(1,0,1,0,1)
bSeeds <- c(0,0,1,0,1)
data <- data.frame(aSeeds,bSeeds)
rownames(data) <- 1:5
# calcualte their two contact graph
dContact <- dRWRpipeline(data=data, g=subg, parallel=FALSE)
</div>
<strong class='message'>Start at 2014-09-18 16:50:30</strong>
<strong class='message'></strong>
<strong class='message'>First, RWR on input graph (15 nodes and 14 edges) using input matrix (5 rows and 2 columns) as seeds (2014-09-18 16:50:30)...</strong>
<strong class='message'>Second, calculate contact strength (2014-09-18 16:50:31)...</strong>
<strong class='message'>Third, generate the distribution of contact strength based on 10 permutations on nodes respecting random (2014-09-18 16:50:31)...</strong>
<strong class='message'>	1 out of 10 (2014-09-18 16:50:31)</strong>
<strong class='message'>	2 out of 10 (2014-09-18 16:50:31)</strong>
<strong class='message'>	3 out of 10 (2014-09-18 16:50:31)</strong>
<strong class='message'>	4 out of 10 (2014-09-18 16:50:31)</strong>
<strong class='message'>	5 out of 10 (2014-09-18 16:50:31)</strong>
<strong class='message'>	6 out of 10 (2014-09-18 16:50:31)</strong>
<strong class='message'>	7 out of 10 (2014-09-18 16:50:31)</strong>
<strong class='message'>	8 out of 10 (2014-09-18 16:50:32)</strong>
<strong class='message'>	9 out of 10 (2014-09-18 16:50:32)</strong>
<strong class='message'>	10 out of 10 (2014-09-18 16:50:32)</strong>
<strong class='message'>Last, estimate the significance of contact strength: zscore, pvalue, and BH adjusted-pvalue (2014-09-18 16:50:32)...</strong>
<strong class='message'>Also, construct the contact graph under the cutoff 5.0e-02 of adjusted-pvalue (2014-09-18 16:50:32)...</strong>
<strong class='message'></strong>
<strong class='message'>Finish at 2014-09-18 16:50:32</strong>
<strong class='message'>Runtime in total is: 2 secs
</strong>
<div class='input'>dContact
</div>
<div class='output'>$ratio
         [,1]     [,2]
[1,] 1.181814 1.191843
[2,] 1.191843 1.141409

$zscore
         [,1]     [,2]
[1,] 2.170178 2.121734
[2,] 2.121734 1.185599

$pval
     [,1] [,2]
[1,]    0  0.0
[2,]    0  0.1

$adjpval
     [,1] [,2]
[1,]    0  0.0
[2,]    0  0.1

$cgraph
IGRAPH U-W- 2 1 -- 
+ attr: weight (e/n)

$Amatrix
NULL

$call
dRWRpipeline(data = data, g = subg, parallel = FALSE)

$method
[1] "dnet"

attr(,"class")
[1] "dContact"
</div></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>dRWRpipeline</li>
    </ul>
    <ul>
      
    </ul> 
    -->
    
    <h2>Source code</h2>
    <a href="javascript:newWin('R/dRWRpipeline.r', 'R/dRWRpipeline.r', '1200', '1200')" title="Click to view"><B><code>dRWRpipeline.r</code></B></a>
    
    <h2>See also</h2>
    
<code><a href='dRWR.html'>dRWR</a></code>, <code><a href='dRWRcontact.html'>dRWRcontact</a></code>,
<code><a href='dCheckParallel.html'>dCheckParallel</a></code>

    <!--

    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) <B>The &#39;dnet&#39; approach promotes emerging research on cancer patient survival</B>. <I>Genome Medicine</I> 6:64 (<a href="http://dx.doi.org/10.1186/s13073-014-0064-8">doi:10.1186/s13073-014-0064-8</a>)</p>

<p>You may also cite the package as:</p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dnet&rdquo; [<a href="http://cran.r-project.org/package=dnet">http://cran.r-project.org/package=dnet</a>]</p>

    
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dnet">http://supfam.org/dnet</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>