<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>dSVDsignif @ dnet 1.0.6</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 1.0.6</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="rdata.html"><i class="icon-file icon-white"></i> RData</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="faqs.html"><i class="icon-question-sign icon-white"></i> FAQs</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dnet R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:djnz9oat1hu';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to obtain SVD-based gene significance from the input gene-sample matrix</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>dSVDsignif</code> is supposed to obtain gene signficance from the given
gene-sample matrix according to singular value decomposition
(SVD)-based method. The method includes: 1) singular value
decomposition of the input matrix; 2) determination of the eigens in
consideration (if not given); 3) construction of the gene-specific
project vector based on the considered eigens; 4) calculation of the
distance statistic from the projection vector to zero point vector; and
5) based on distance statistic to obtain the gene significance.</p>
  
    <h3>Usage</h3>
    <pre>dSVDsignif(data, num.eigen = NULL, pval.eigen = 0.01, signif = c("fdr", "pval"), 
  orient.permutation = c("row", "column", "both"), num.permutation = 100, fdr.procedure = c("stepup", 
          "stepdown"), verbose = T)</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>data</dt>
      <dd>an input gene-sample data matrix used for singular value
decomposition</dd>
      <dt>num.eigen</dt>
      <dd>an integer specifying the number of eigens in
consideration. If NULL, this number will be automatically decided on
based on the observed relative eigenexpression against randomised
relative eigenexpression calculated from a list (here 100) of
permutated input matrix</dd>
      <dt>pval.eigen</dt>
      <dd>p-value used to call those eigens as dominant. This
parameter is used only when parameter 'num.eigen' is NULL. Here,
p-value is calcualted to assess how likely the observed relative
eigenexpression are more than the maximum relative eigenexpression
calculated from permutated matrix</dd>
      <dt>signif</dt>
      <dd>the singificance to return. It can be either "pval" for
using the p-value as the gene significance, or "fdr" for using the fdr
as the gene significance</dd>
      <dt>orient.permutation</dt>
      <dd>the orientation of matrix being permutated.
It can be either "row" to permutate values within each row, or "column"
to permutate values within each column, or "both" to permutate values
both within rows and columns. Notably, when using the p-value as the
gene significance, it is always to permutate values within each row.</dd>
      <dt>num.permutation</dt>
      <dd>an integer specifying how many permutations are
used</dd>
      <dt>fdr.procedure</dt>
      <dd>the procedure to adjust the fdr. To ensure that
the high distance statistic the more significance, the fdr should be
adjusted either using "stepup" for step-up procedure (from the most
significant to the least significant) or using "stepdown" for step-down
procedure (from the least significant to the most significant)</dd>
      <dt>verbose</dt>
      <dd>logical to indicate whether the messages will be
displayed in the screen. By default, it sets to true for display</dd>
    </dl>
    <!-- 
    -->
    
    <!-- 
    <div class="Value">
    -->
      <h3>Value</h3>
      
      <p>a vector storing gene significance</p>
    <!-- 
    </div>
    -->
    <!-- 
    <div class="Note">
    -->
      <h3>Note</h3>
      
      <p>none</p>
    <!-- 
    </div>
    -->
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'># 1) generate data with an iid matrix of 1000 x 9
data <- cbind(matrix(rnorm(1000*3,mean=0,sd=1), nrow=1000, ncol=3),
matrix(rnorm(1000*3,mean=0.5,sd=1), nrow=1000, ncol=3),
matrix(rnorm(1000*3,mean=-0.5,sd=1), nrow=1000, ncol=3))

# 2) calculate the significance according to SVD
# using "fdr" significance
fdr <- dSVDsignif(data, signif="fdr", num.permutation=10)
</div>
<strong class='message'>Start at 2014-09-22 19:29:38</strong>
<strong class='message'></strong>
<strong class='message'>First, singular value decomposition of the input matrix (with 1000 rows and 9 columns)...</strong>
<strong class='message'>Second, determinate the eigens...</strong>
<strong class='message'>	via automatically deciding on the number of dominant eigens under the cutoff of 1.00e-02 pvalue</strong>
<strong class='message'>	number of the eigens in consideration: 1</strong>
<strong class='message'>Third, construct the gene-specific projection vector,and calculate distance statistics...</strong>
<strong class='message'>Finally, obtain gene significance (fdr) based on 10 permutations...</strong>
<strong class='message'>	doing row-wise permutations...</strong>
<strong class='message'>	estimating fdr...</strong>
<strong class='message'>1 (out of 1000) at 2014-09-22 19:29:40</strong>
<strong class='message'>100 (out of 1000) at 2014-09-22 19:29:40</strong>
<strong class='message'>200 (out of 1000) at 2014-09-22 19:29:40</strong>
<strong class='message'>300 (out of 1000) at 2014-09-22 19:29:40</strong>
<strong class='message'>400 (out of 1000) at 2014-09-22 19:29:40</strong>
<strong class='message'>500 (out of 1000) at 2014-09-22 19:29:40</strong>
<strong class='message'>600 (out of 1000) at 2014-09-22 19:29:40</strong>
<strong class='message'>700 (out of 1000) at 2014-09-22 19:29:40</strong>
<strong class='message'>800 (out of 1000) at 2014-09-22 19:29:40</strong>
<strong class='message'>900 (out of 1000) at 2014-09-22 19:29:40</strong>
<strong class='message'>1000 (out of 1000) at 2014-09-22 19:29:40</strong>
<strong class='message'>	using stepup procedure...</strong>
<strong class='message'></strong>
<strong class='message'>Finish at 2014-09-22 19:29:40</strong>
<strong class='message'>Runtime in total is: 2 secs
</strong>
<div class='input'># using "pval" significance
pval <- dSVDsignif(data, signif="pval", num.permutation=10)
</div>
<strong class='message'>Start at 2014-09-22 19:29:40</strong>
<strong class='message'></strong>
<strong class='message'>First, singular value decomposition of the input matrix (with 1000 rows and 9 columns)...</strong>
<strong class='message'>Second, determinate the eigens...</strong>
<strong class='message'>	via automatically deciding on the number of dominant eigens under the cutoff of 1.00e-02 pvalue</strong>
<strong class='message'>	number of the eigens in consideration: 1</strong>
<strong class='message'>Third, construct the gene-specific projection vector,and calculate distance statistics...</strong>
<strong class='message'>Finally, obtain gene significance (p-value) based on 10 of row-wise permutations...</strong>
<strong class='message'></strong>
<strong class='message'>Finish at 2014-09-22 19:29:41</strong>
<strong class='message'>Runtime in total is: 1 secs
</strong></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>dSVDsignif</li>
    </ul>
    <ul>
      
    </ul> 
    -->
    
    <h2>Source code</h2>
    <a href="javascript:newWin('R/dSVDsignif.r', 'R/dSVDsignif.r', '1200', '1200')" title="Click to view"><B><code>dSVDsignif.r</code></B></a>
    
    <h2>Source man</h2>
    <a href="javascript:newWin('man/dSVDsignif.txt', 'man/dSVDsignif.txt', '1200', '1200')" title="Click to view"><B><code>dSVDsignif.Rd</code></B></a>
    
    <h2>See also</h2>
    
<code><a href='dFDRscore.html'>dFDRscore</a></code>

    <!--

    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) <B>The &#39;dnet&#39; approach promotes emerging research on cancer patient survival</B>. <I>Genome Medicine</I> 6:64 (<a href="http://dx.doi.org/10.1186/s13073-014-0064-8">doi:10.1186/s13073-014-0064-8</a>)</p>

<p>You may also cite the package as:</p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dnet&rdquo; [<a href="http://cran.r-project.org/package=dnet">http://cran.r-project.org/package=dnet</a>]</p>

    
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dnet">http://supfam.org/dnet</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>