<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>eTOL @ dnet 1.0.4</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 1.0.4</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="rdata.html"><i class="icon-file icon-white"></i> RData</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="faqs.html"><i class="icon-question-sign icon-white"></i> FAQs</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dnet R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:djnz9oat1hu';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>eukaryotic Tree Of Life (eTOL)</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p>A 'phylo' object that contains information about eukaryotic part of species tree of life (eTOL). It is a rooted binary tree. Tips represent extant genomes. Since its reconstruction is guided under the NCBI taxonomy, each internal node is either mapped onto a unique taxonomic identifier or left empty (assumedly a hypothetical unknown ancestral genome).</p>
  
    <h3>Usage</h3>
    <pre>eTOL <- dRDataLoader(RData = "eTOL")</pre>
        <!-- 
    -->
    
    <div class="Value">
      <h3>Value</h3>
      
      <p>an object of class "phylo" with the following components:
    <ul>
<li><code>Nnode</code>: the number of (internal) nodes
        </li>
<li><code>tip.label</code>: a vector giving the names of the tips (i.e., "left_id" to define the post-ordered binary tree structure)
        </li>
<li><code>node.label</code>: a vector giving the names of the internal nodes (i.e., "left_id" to define the post-ordered binary tree structure)
        </li>
<li><code>genome_info</code>: a matrix of all nodes (including tips and internal nodes) X 8, giving extant/ancestral genome information: "left_id" (unique and used as internal id), "right_id" (used in combination with "left_id" to define the post-ordered binary tree structure), "taxon_id" (NCBI taxonomy id, if matched), "genome" (2-letter genome identifiers used in SUPERFAMILY, if being extant), "name" (NCBI taxonomy name, if matched), "rank" (NCBI taxonomy rank, if matched), "branchlength" (branch length in relevance to the parent), and "common_name" (NCBI taxonomy common name, if matched and existed)
        </li>
<li><code>edge</code>: a two-column matrix of mode numeric where each row represents an edge of the tree; the nodes and the tips are symbolized with numbers; the tips are numbered 1, 2, ..., and the internal nodes are numbered after the tips. For each row, the first column gives the ancestor
        </li>
<li><code>edge.length</code>: a numeric vector giving the lengths of the branches given by 'edge'
        </li>
<li><code>root.length</code>: a numeric value giving the length of the branch at the root
        </li>
<li><code>connectivity</code>: a matrix of internal nodes X all nodes (including tips and internal nodes), with 1 for the presence of a ancestor-descenant path, and 0 otherwise
    </li>
</ul></p>
      <p></p>
    </div>
    <div class="References">
      <h3>References</h3>
      
      <p>Fang et al. (2013) A daily-updated tree of (sequenced) life as a reference for genome research. <em>Scientific reports</em>, 3:2015.</p>
    </div>
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'>library(ape)
eTOL <- dRDataLoader(RData='eTOL')
</div>
<strong class='message'>'eTOL' (from http://supfam.org/dnet/data/Datasets/eTOL.RData) has been loaded into the working environment</strong>
<div class='input'>eTOL
</div>
<div class='output'>
Phylogenetic tree with 438 tips and 437 internal nodes.

Tip labels:
	37, 40, 42, 46, 49, 52, ...
Node labels:
	3, 4, 5, 6, 7, 8, ...

Rooted; includes branch lengths.
</div>
<div class='input'># list all components
names(eTOL)
</div>
<div class='output'>[1] "edge"         "Nnode"        "tip.label"    "edge.length"  "node.label"  
[6] "root.edge"    "connectivity" "genome_info" 
</div>
<div class='input'># extract information about the first 5 genomes
eTOL$genome_info[1:5,]
</div>
<div class='output'>  left_id right_id taxon_id genome      name         rank branchlength
3       3     1752     2759   <NA> Eukaryota superkingdom  0.032247873
4       4     1749       NA   <NA>      <NA>         <NA>  0.022275411
5       5     1546       NA   <NA>      <NA>         <NA>  0.004497603
6       6     1499       NA   <NA>      <NA>         <NA>  0.003496342
7       7     1360       NA   <NA>      <NA>         <NA>  0.005465982
  common_name
3  eucaryotes
4        <NA>
5        <NA>
6        <NA>
7        <NA>
</div>
<div class='input'># look at the dimension of connectivity
dim(eTOL$connectivity)
</div>
<div class='output'>[1] 437 875
</div>
<div class='input'># visualise the connectivity matrix
Ntip <- length(eTOL$tip.label) # number of tips
Nnode <- eTOL$Nnode # number of internal nodes
data <- eTOL$connectivity
visHeatmapAdv(data, Rowv=F,Colv=F, zlim=c(0,1), colormap="gray-black", add.expr=abline(v=c(1,Ntip+1,(Ntip+Nnode+1))-0.5, col="white"), key=F, labRow=NA, labCol=NA)
</div>
<a href="javascript:newWin('eTOL-12.png', 'eTOL-12.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='eTOL-12.png', width="400" /></a></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>eTOL</li>
    </ul>
    <ul>
      <li>datasets</li>
    </ul> 
    -->
    
    
    <!--

    <h2>Citation</h2>
    <ul>
<li>Cite as <code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dnet&rdquo; [<a href="http://cran.r-project.org/package=dnet">http://cran.r-project.org/package=dnet</a>]</p>

<ul>
<li>Cite as <code>standard reference</code></li>
</ul>

<p>Fang H, Gough J. (2014) dnet: an open-source R package for omics data integrative analysis in terms of network, evolution and ontology. <a href="http://cran.r-project.org/package=dnet">http://cran.r-project.org/package=dnet</a></p>

<ul>
<li>Cite as <code>an application</code></li>
</ul>

<p>Fang H, Gough J. (2014) The &#39;dnet&#39; approach promotes emerging research on cancer patient survival. <I>Genome Medicine</I> (in revision)</p>

    
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dnet">http://supfam.org/dnet</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>