<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>How many ways are supported to visualise gene-active networks? @ dnet 1.0.3</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="Hai Fang and Julian Gough">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 1.0.3</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="rdata.html"><i class="icon-file icon-white"></i> RData</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="faqs.html"><i class="icon-question-sign icon-white"></i> FAQs</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>How many ways are supported to visualise gene-active networks?</h2>

Notes:
<li> All results are based on dnet (version 1.0.3). </li>
<li> R scripts (i.e. R expressions) plus necessary comments are highlighted in light-cyan background, and the rest are outputs in the screen. </li>
<li> Images displayed below may be distorted, but should be normal in your screen. </li>
<li> Functions contained in dnet 1.0.3 are hyperlinked in-place and also listed on the right side. </li>
<li> Key texts are underlined, in bold and in pumpkin-orange color. </li>

<div class="row">
    <div class="span8">
      <pre>
      <div class='input'># <span style="font-weight:bold; color:#F87217; text-decoration:underline">The dnet package supports two ways to visualise the identified networks/graphs: 1) the network itself as a single display, 2) the same network but with the multiple colorings/displays according to multiple samples.</span>

# To demonstrate the visuals supported, we use a random graph according to the ER model, and only keep the largest component
g <- erdos.renyi.game(100, 1/100)
g <- <a href='dNetInduce.html' target=dNetInduce>dNetInduce</a>(g, nodes_query=V(g))

# <span style="font-weight:bold; color:#F87217; text-decoration:underline">To display the network itself, the key setting is the layout.</span> Since the network is represented as an object of class 'igraph', the layouts supported in the 'igraph' package can be found in the <a href="http://igraph.org/r/doc/layout.html">layout</a> page. 
## using a force-based algorithm proposed by Fruchterman and Reingold
<a href='visNet.html' target=visNet>visNet</a>(g, layout=layout.fruchterman.reingold)
</div>
<a href="javascript:newWin('FAQ4-2.png', 'FAQ4-2.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ4-2.png', width="400" /></a>
<div class='input'>## using a spring embedder algorithm
<a href='visNet.html' target=visNet>visNet</a>(g, layout=layout.spring)
</div>
<a href="javascript:newWin('FAQ4-4.png', 'FAQ4-4.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ4-4.png', width="400" /></a>
<div class='input'>
# Additionally, the dnet package also uses two other layouts/diagrams: 1) arc diagram in one-dimensional layout, and 2) circle diagram. 

# <span style="font-weight:bold; color:#F87217; text-decoration:underline">To better display the network, it is advisable to incorporate community information.</span> In doing so, we first identify communities based on a spin-glass model and simulated annealing.
com <- igraph::spinglass.community(g, spins=25)
vgroups <- com$membership
## color nodes: according to communities
mcolors <- visColormap(colormap="rainbow")(length(com))
vcolors <- mcolors[vgroups]
## size nodes: according to degrees
vdegrees <- igraph::degree(g)
## sort nodes: first by communities and then degrees
df <- data.frame(ind=1:vcount(g), vgroups, vdegrees)
ordering <- df[order(vgroups,vdegrees),]$ind

# Now, make comparsions between different layouts
## using 1-dimensional arc diagram
<a href='visNetArc.html' target=visNetArc>visNetArc</a>(g, ordering=ordering, vertex.label.color=vcolors, vertex.color=vcolors, vertex.frame.color=vcolors, vertex.size=log(vdegrees)+0.1)
</div>
<a href="javascript:newWin('FAQ4-6.png', 'FAQ4-6.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ4-6.png', width="400" /></a>
<div class='input'>## using circle diagram (drawn into a single circle)
<a href='visNetCircle.html' target=visNetCircle>visNetCircle</a>(g, colormap="rainbow", com=com, ordering=ordering)
</div>
<a href="javascript:newWin('FAQ4-8.png', 'FAQ4-8.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ4-8.png', width="400" /></a>
<div class='input'>## using circle diagram (drawn into multlpe circles: one circle per community)
<a href='visNetCircle.html' target=visNetCircle>visNetCircle</a>(g, colormap="rainbow", com=com, circles="multiple", ordering=ordering)
</div>
<a href="javascript:newWin('FAQ4-10.png', 'FAQ4-10.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ4-10.png', width="400" /></a>
<div class='input'>## using a force-based algorithm proposed by Fruchterman and Reingold
<a href='visNet.html' target=visNet>visNet</a>(g, colormap="rainbow", layout=layout.fruchterman.reingold, vertex.color=vcolors, vertex.frame.color=vcolors, vertex.shape="sphere")
</div>
<a href="javascript:newWin('FAQ4-12.png', 'FAQ4-12.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ4-12.png', width="400" /></a>
<div class='input'>## when using force-based layout, it is also useful to highlight the communities in the background, and have edges being colored differently according whether an edge lies within a community or between communities. 
mark.groups <- igraph::communities(com)
mark.col <- visColoralpha(mcolors, alpha=0.2)
mark.border <- visColoralpha(mcolors, alpha=0.2)
edge.color <- c("grey", "black")[igraph::crossing(com,g)+1]
<a href='visNet.html' target=visNet>visNet</a>(g, colormap="rainbow", glayout=layout.fruchterman.reingold, vertex.color=vcolors, vertex.frame.color=vcolors, vertex.shape="sphere", mark.groups=mark.groups, mark.col=mark.col, mark.border=mark.border, mark.shape=1, mark.expand=10, edge.color=edge.color)</div>
<a href="javascript:newWin('FAQ4-14.png', 'FAQ4-14.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ4-14.png', width="400" /></a>
      </pre>
    </div>

  <div class="span4">
    
    <h2>Functions used in this FAQ</h2>
    <ul>
      <li><a href="dNetInduce.html">dNetInduce</a></li>
      <li><a href="visNet.html">visNet</a></li>
      <li><a href="visNetArc.html">visNetArc</a></li>
      <li><a href="visNetCircle.html">visNetCircle</a></li>
    </ul>
    
    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) dnet: an open-source R package for omics data integrative analysis in terms of network, evolution and ontology. <a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></p>

    
    <!--
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></li>
    </ul>
    
    <h2>Author</h2>
    Hai Fang and Julian Gough
    
    -->
    
  </div>

</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>