<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>How to extract a subnetwork that revolves around a specific gene of interest? @ dnet 1.0.7</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="Hai Fang and Julian Gough">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 1.0.7</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="rdata.html"><i class="icon-file icon-white"></i> RData</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="faqs.html"><i class="icon-question-sign icon-white"></i> FAQs</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dnet R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:djnz9oat1hu';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>How to extract a subnetwork that revolves around a specific gene of interest?</h2>

Notes:
<li> All results are based on dnet (version 1.0.7). </li>
<li> R scripts (i.e. R expressions) plus necessary comments are highlighted in light-cyan background, and the rest are outputs in the screen. </li>
<li> Images displayed below may be distorted, but should be normal in your screen. </li>
<li> Functions contained in dnet 1.0.7 are hyperlinked in-place and also listed on the right side. </li>
<li> Key texts are underlined, in bold and in pumpkin-orange color. </li>

<div class="row">
    <div class="span8">
      <pre>
      <div class='input'># <span style="font-weight:bold; color:#F87217; text-decoration:underline">To answer this question, without loss of generality we formulate it as an often-encountered problem: from the whole human gene network, how to derive a subnetwork that revolves around a gene 'POU5F1' and its direct interacting neighbors.</span>

# 1) load the whole human gene network. The network is extracted from the STRING database (version 9.1). Only those associations with medium confidence (score>=400) are retained
network <- <a href='dRDataLoader.html' target=dRDataLoader>dRDataLoader</a>(RData = "org.Hs.string")
</div>
<strong class='message'>'org.Hs.string' (from http://dnet.r-forge.r-project.org/data/Hs/org.Hs.string.RData) has been loaded into the working environment</strong>
<div class='input'># If required, can only focus on those edges with confidence score >= 700
network <- subgraph.edges(network, eids=E(network)[combined_score>=700])
network
</div>
<div class='output'>IGRAPH UN-- 13026 198516 -- 
+ attr: name (v/c), seqid (v/c), geneid (v/n), symbol (v/c),
  description (v/c), neighborhood_score (e/n), fusion_score (e/n),
  cooccurence_score (e/n), coexpression_score (e/n), experimental_score
  (e/n), database_score (e/n), textmining_score (e/n), combined_score
  (e/n)
</div>
<div class='input'># 2) find the gene 'POU5F1' and its direct neighbors
ind <- match(V(network)$symbol, 'POU5F1')
nodes_query <- V(network)$name[!is.na(ind)]
g <- <a href='dNetInduce.html' target=dNetInduce>dNetInduce</a>(g=network, nodes_query=nodes_query, knn=1, remove.loops=F, largest.comp=T)
g
</div>
<div class='output'>IGRAPH UN-- 14 20 -- 
+ attr: name (v/c), seqid (v/c), geneid (v/n), symbol (v/c),
  description (v/c), neighborhood_score (e/n), fusion_score (e/n),
  cooccurence_score (e/n), coexpression_score (e/n), experimental_score
  (e/n), database_score (e/n), textmining_score (e/n), combined_score
  (e/n)
</div>
<div class='input'># 3) visualise the extracted subnetwork
<a href='visNet.html' target=visNet>visNet</a>(g, vertex.label=V(g)$symbol, vertex.shape="sphere")
# 4) highlight the query gene in red, and its neighours in black
vcolors <- rep('black', vcount(g))
</div>
<a href="javascript:newWin('FAQ6-8.png', 'FAQ6-8.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ6-8.png', width="400" /></a>
<div class='input'>vcolors[match('POU5F1', V(g)$symbol)] <- 'red'
<a href='visNet.html' target=visNet>visNet</a>(g, vertex.label=V(g)$symbol, vertex.shape="sphere", vertex.color=vcolors)
# 5) output the detailed information for genes in the subnetwork onto the file called 'POU5F1.subnetwork.txt'
out <- data.frame(GeneID=V(g)$geneid, Symbol=V(g)$symbol, Description=V(g)$description)
</div>
<a href="javascript:newWin('FAQ6-10.png', 'FAQ6-10.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ6-10.png', width="400" /></a>
<div class='input'>write.table(out, file="POU5F1.subnetwork.txt", quote=F, row.names=F,col.names=T,sep="\t")
out
</div>
<div class='output'>   GeneID  Symbol                                          Description
1   27022   FOXD3                                      forkhead box D3
2    6657    SOX2                 SRY (sex determining region Y)-box 2
3  135935   NOBOX                             NOBOX oogenesis homeobox
4   57167   SALL4                    spalt-like transcription factor 4
5   79923   NANOG                                       Nanog homeobox
6   22887   FOXJ3                                      forkhead box J3
7    7178    TPT1          tumor protein, translationally-controlled 1
8   84217 ZMYND12                 zinc finger, MYND-type containing 12
9    2103   ESRRB                       estrogen-related receptor beta
10  10637  LEFTY1                    left-right determination factor 1
11  10664    CTCF           CCCTC-binding factor (zinc finger protein)
12   2981  GUCA2B         guanylate cyclase activator 2B (uroguanylin)
13  23401   FRAT2 frequently rearranged in advanced T-cell lymphomas 2
14   5460  POU5F1                               POU class 5 homeobox 1
</div>
<div class='input'>
# <span style="font-weight:bold; color:#F87217; text-decoration:underline">A more complicated question could be: the interaction neighbors are also required to be functionally related to 'stem'.</span> For this, we can further use terms from Gene Ontology (and its subontology: Biological Process).
keyword <- 'stem'
org.Hs.egGOBP <- <a href='dRDataLoader.html' target=dRDataLoader>dRDataLoader</a>(RData = "org.Hs.egGOBP")
</div>
<strong class='message'>'org.Hs.egGOBP' (from http://dnet.r-forge.r-project.org/data/Hs/org.Hs.egGOBP.RData) has been loaded into the working environment</strong>
<div class='input'>ind <- grep(keyword, org.Hs.egGOBP$set_info$name, ignore.case=T, perl=T)
genes <- unique(unlist(org.Hs.egGOBP$gs[ind]))

# Next, identify genes in the subnetwork that are functionally related to 'stem'
ind <- match(V(g)$geneid, genes)
intersect_genes <- V(g)$symbol[!is.na(ind)]
intersect_genes
</div>
<div class='output'>[1] "FOXD3"  "SOX2"   "SALL4"  "NANOG"  "TPT1"   "ESRRB"  "POU5F1"
</div>
<div class='input'>
# Now, highlight the query gene in red, its 'stem'-related neighours in blue, and the rest in black
vcolors <- rep('black', vcount(g))
vcolors[match(intersect_genes, V(g)$symbol)] <- 'blue'
vcolors[match('POU5F1', V(g)$symbol)] <- 'red'
<a href='visNet.html' target=visNet>visNet</a>(g, vertex.label=V(g)$symbol, vertex.shape="sphere", vertex.color=vcolors)

# <span style="font-weight:bold; color:#F87217; text-decoration:underline">Alternatively, remain only genes related to 'stem' and the query gene 'POU5F1'.</span>
both_genes <- c(intersect_genes, 'POU5F1')
</div>
<a href="javascript:newWin('FAQ6-18.png', 'FAQ6-18.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ6-18.png', width="400" /></a>
<div class='input'>both_genes
</div>
<div class='output'>[1] "FOXD3"  "SOX2"   "SALL4"  "NANOG"  "TPT1"   "ESRRB"  "POU5F1" "POU5F1"
</div>
<div class='input'>
# extract the subnetwork containing the gene 'CDKN2D' and its differentiation-related neighbors
ind <- match(V(g)$symbol, both_genes)
nodes_query <- V(g)$name[!is.na(ind)]
g1 <- <a href='dNetInduce.html' target=dNetInduce>dNetInduce</a>(g, nodes_query=nodes_query, knn=0, remove.loops=F, largest.comp=T)
vcolors <- rep('blue', vcount(g1))
vcolors[match('POU5F1', V(g1)$symbol)] <- 'red'
<a href='visNet.html' target=visNet>visNet</a>(g1, vertex.label=V(g1)$symbol, vertex.shape="sphere", vertex.color=vcolors)

# <span style="font-weight:bold; color:#F87217; text-decoration:underline">Also, only edges linked to the query gene 'POU5F1' are remained.</span>
ind <- match('POU5F1', V(g1)$symbol)
</div>
<a href="javascript:newWin('FAQ6-22.png', 'FAQ6-22.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ6-22.png', width="400" /></a>
<div class='input'>eids <- E(g1)[from(ind)]
g2 <- subgraph.edges(g1, eids=eids)
vcolors <- rep('blue', vcount(g2))
vcolors[match('POU5F1', V(g2)$symbol)] <- 'red'
<a href='visNet.html' target=visNet>visNet</a>(g2, vertex.label=V(g2)$symbol, vertex.shape="sphere", vertex.color=vcolors)

# <span style="font-weight:bold; color:#F87217; text-decoration:underline">In addition to the direct interacting neighbors, the indirect neighbors can be specifed via the parameter 'knn' in the function <a href='dNetInduce.html' target=dNetInduce>dNetInduce</a>.</span></div>
<a href="javascript:newWin('FAQ6-24.png', 'FAQ6-24.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ6-24.png', width="400" /></a>
      </pre>
    </div>

  <div class="span4">
    
    <h2>Source faq</h2>
    <a href="javascript:newWin('faq/FAQ6.r', 'faq/FAQ6.r', '1200', '1200')" title="Click to view"><B><code>FAQ6.r</code></B></a>
    
    <h2>Functions used in this FAQ</h2>
    <ul>
      <li><a href="dRDataLoader.html">dRDataLoader</a></li>
      <li><a href="dNetInduce.html">dNetInduce</a></li>
      <li><a href="visNet.html">visNet</a></li>
    </ul>
    <!--    
    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) <B>The &#39;dnet&#39; approach promotes emerging research on cancer patient survival</B>. <I>Genome Medicine</I>, 6(8):64. <a href="http://dx.doi.org/10.1186/s13073-014-0064-8">doi:10.1186/s13073-014-0064-8</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/25246945">25246945</a></p>

<p>You may also cite the package as:</p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dnet&rdquo; [<a href="http://cran.r-project.org/package=dnet">http://cran.r-project.org/package=dnet</a>]</p>

    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dnet">http://supfam.org/dnet</a></li>
      <li><a href=" http://dnet.r-forge.r-project.org"> http://dnet.r-forge.r-project.org</a></li>
      <li><a href=" http://cran.r-
project.org/package=dnet"> http://cran.r-
project.org/package=dnet</a></li>
      <li><a href=" https://github.com/hfang-bristol/dnet"> https://github.com/hfang-bristol/dnet</a></li>
    </ul>
    
    <h2>Author</h2>
    Hai Fang and Julian Gough
    
    -->
    
  </div>

</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@well.ox.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>