<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>What is Random Walk with Restart (RWR)? @ dnet 1.0.7</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="Hai Fang and Julian Gough">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 1.0.7</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="rdata.html"><i class="icon-file icon-white"></i> RData</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="faqs.html"><i class="icon-question-sign icon-white"></i> FAQs</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dnet R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:djnz9oat1hu';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>What is Random Walk with Restart (RWR)?</h2>

Notes:
<li> All results are based on dnet (version 1.0.7). </li>
<li> R scripts (i.e. R expressions) plus necessary comments are highlighted in light-cyan background, and the rest are outputs in the screen. </li>
<li> Images displayed below may be distorted, but should be normal in your screen. </li>
<li> Functions contained in dnet 1.0.7 are hyperlinked in-place and also listed on the right side. </li>
<li> Key texts are underlined, in bold and in pumpkin-orange color. </li>

<div class="row">
    <div class="span8">
      <pre>
      <div class='input'># <span style="font-weight:bold; color:#F87217; text-decoration:underline">RWR is a algorithm originally proposed for image segmentation. It iteratively explores the global structure of the network to estimate the proximity (affinity score) between two nodes.</span> Starting at a node, the walker faces two choices at each step: either moving to a randomly chosen neighbor, or jumping back to the starting node. This algorithm only contains one fixed parameter r called 'the restarting probability' (with 1 âˆ’ r for the probability of moving to a neighbor). After iteratively reaching stability, the steady probability vector contains the affinity score of all nodes in the network to the starting node. This steady probability vector can be viewed as the 'influential impact' over the network imposed by the starting node.

# In addition to a single starting node, RWR can be extended to a set of starting nodes. In this context, the resulting steady probability vector can also be considered as the 'influential impact' over the network, but collectively imposed by a set of starting nodes.

# <span style="font-weight:bold; color:#F87217; text-decoration:underline">The steady probability vector for a set of starting nodes can be calculated according to the steady probability vector for single starting nodes. For this reason, it is useful to pre-compute affinity matrix for nodes in the graph with respect to each starting node as a seed (by looping over every node in the graph).</span> The motivation behind this is that this pre-computed affinity matrix can be extensively reused for obtaining affinity scores between any combinations of nodes. For instance, such a strategy can dramatically relieve the computational burden involved in sampling a large number of random node combinations. Therefore, this pre-computation does permit some flexibility in the downstream use, in particular for statistical testing.

# <span style="font-weight:bold; color:#F87217; text-decoration:underline">In the dnet package, RWR for a single node or a set of nodes are both implemented in the function <a href='dRWR.html' target=dRWR>dRWR</a>.</span>

## generate a random graph generated according to the ER model, and only keep the largest component
g <- erdos.renyi.game(100, 1/100)
g <- <a href='dNetInduce.html' target=dNetInduce>dNetInduce</a>(g, nodes_query=V(g))
V(g)$name <- 1:vcount(g)
## obtain the pre-computated affinity matrix for all nodes
PTmatrix <- <a href='dRWR.html' target=dRWR>dRWR</a>(g, normalise="laplacian", restart=0.75)
</div>
<strong class='message'>Start at 2014-12-16 17:18:19</strong>
<strong class='message'></strong>
<strong class='message'>First, get the adjacency matrix of the input graph (2014-12-16 17:18:19) ...</strong>
<strong class='message'>Then, normalise the adjacency matrix using laplacian normalisation (2014-12-16 17:18:19) ...</strong>
<strong class='message'>Third, RWR of 11 sets of seeds using 7.5e-01 restart probability (2014-12-16 17:18:19) ...</strong>
<strong class='message'>	do parallel computation using 2 cores ...</strong>
<strong class='message'>Finally, output 11 by 11 affinity matrix normalised by none (2014-12-16 17:18:24) ...</strong>
<strong class='message'>
Finish at 2014-12-16 17:18:24</strong>
<strong class='message'>Runtime in total is: 5 secs
</strong>
<div class='input'>### visualise affinity matrix
visHeatmapAdv(as.matrix(PTmatrix), Rowv=FALSE, Colv=FALSE, colormap="wyr", KeyValueName="Affinity")
</div>
<a href="javascript:newWin('FAQ5-12.png', 'FAQ5-12.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ5-12.png', width="400" /></a>
<div class='input'>## obtain affinity score for a set of nodes
setSeeds <- as.integer(runif(vcount(g))>=0.5) # generate a random binary vector as a seedSet
setSeeds <- as.matrix(setSeeds, ncol=1) # must be a matrix
rownames(setSeeds) <- 1:vcount(g) # must be having rownames
PTmatrix <- <a href='dRWR.html' target=dRWR>dRWR</a>(g, normalise="laplacian", setSeeds=setSeeds, restart=0.75)
</div>
<strong class='message'>Start at 2014-12-16 17:18:24</strong>
<strong class='message'></strong>
<strong class='message'>First, get the adjacency matrix of the input graph (2014-12-16 17:18:24) ...</strong>
<strong class='message'>Then, normalise the adjacency matrix using laplacian normalisation (2014-12-16 17:18:24) ...</strong>
<strong class='message'>Third, RWR of 1 sets of seeds using 7.5e-01 restart probability (2014-12-16 17:18:24) ...</strong>
<strong class='message'>	do parallel computation using 2 cores ...</strong>
<strong class='message'>	1 out of 1 seed sets (2014-12-16 17:18:24)</strong>
<strong class='message'>Finally, output 11 by 1 affinity matrix normalised by none (2014-12-16 17:18:24) ...</strong>
<strong class='message'>
Finish at 2014-12-16 17:18:24</strong>
<strong class='message'>Runtime in total is: 0 secs
</strong>
<div class='input'>
# <span style="font-weight:bold; color:#F87217; text-decoration:underline">As you have noticed, here high-performance parallel computing is used but only for Linux or Mac operating systems.</span> You can specify the number of cores via the parameter 'multicores'.

# <span style="font-weight:bold; color:#F87217; text-decoration:underline">One of potential utility of using RWR is to estimate sample relationships (ie contact strength between samples) from an input gene-sample matrix.</span> It should be noted: each value in input gene-sample matrix does not necessarily have to be binary (non-zeros will be used as a weight, but should be non-negative for easy interpretation). 

# The pipeline includes: 1) RWR of the input graph using the input matrix as seeds; 2) calculation of contact strength (inner products of RWR-smoothed columns of input matrix); 3) estimation of the contact signficance by a randomalisation procedure.

## Prepare for input gene-sample matrix: vcount(g) X 10
data <- matrix(runif(vcount(g)*10), ncol=10)
rownames(data) <- 1:vcount(g)
colnames(data) <- paste('S', 1:10, sep='')
## Run using the pipeline
dContact <- <a href='dRWRpipeline.html' target=dRWRpipeline>dRWRpipeline</a>(data, g, restart=0.75)
</div>
<strong class='message'>Start at 2014-12-16 17:18:24</strong>
<strong class='message'></strong>
<strong class='message'>First, RWR on input graph (11 nodes and 10 edges) using input matrix (11 rows and 10 columns) as seeds (2014-12-16 17:18:24)...</strong>
<strong class='message'>Second, calculate contact strength (2014-12-16 17:18:25)...</strong>
<strong class='message'>Third, generate the distribution of contact strength based on 10 permutations on nodes respecting random (2014-12-16 17:18:25)...</strong>
<strong class='message'>	do parallel computation using 2 cores ...</strong>
<strong class='message'>Last, estimate the significance of contact strength: zscore, pvalue, and BH adjusted-pvalue (2014-12-16 17:18:26)...</strong>
<strong class='message'>Also, construct the contact graph under the cutoff 5.0e-02 of adjusted-pvalue (2014-12-16 17:18:26)...</strong>
<strong class='message'></strong>
<strong class='message'>Finish at 2014-12-16 17:18:26</strong>
<strong class='message'>Runtime in total is: 2 secs
</strong>
<div class='input'>## Run using two steps instead: first pre-compute affinity matrix, and then estimate RWR-based sample relationships
Amatrix <- <a href='dRWR.html' target=dRWR>dRWR</a>(g, restart=0.75)
</div>
<strong class='message'>Start at 2014-12-16 17:18:26</strong>
<strong class='message'></strong>
<strong class='message'>First, get the adjacency matrix of the input graph (2014-12-16 17:18:26) ...</strong>
<strong class='message'>Then, normalise the adjacency matrix using laplacian normalisation (2014-12-16 17:18:26) ...</strong>
<strong class='message'>Third, RWR of 11 sets of seeds using 7.5e-01 restart probability (2014-12-16 17:18:26) ...</strong>
<strong class='message'>	do parallel computation using 2 cores ...</strong>
<strong class='message'>Finally, output 11 by 11 affinity matrix normalised by none (2014-12-16 17:18:27) ...</strong>
<strong class='message'>
Finish at 2014-12-16 17:18:27</strong>
<strong class='message'>Runtime in total is: 1 secs
</strong>
<div class='input'>dContact <- <a href='dRWRcontact.html' target=dRWRcontact>dRWRcontact</a>(data, g, Amatrix)</div>
<strong class='message'>Start at 2014-12-16 17:18:27</strong>
<strong class='message'></strong>
<strong class='message'>First, RWR on input graph (11 nodes and 10 edges) using input matrix (11 rows and 10 columns) as seeds and pre-computed affinity matrix (11 rows and 11 columns) (2014-12-16 17:18:27)...</strong>
<strong class='message'>Second, calculate contact strength (2014-12-16 17:18:27)...</strong>
<strong class='message'>Third, generate the distribution of contact strength based on 10 permutations on nodes respecting random (2014-12-16 17:18:27)...</strong>
<strong class='message'>	do parallel computation using 2 cores ...</strong>
<strong class='message'>Last, estimate the significance of contact strength: zscore, pvalue, and BH adjusted-pvalue (2014-12-16 17:18:27)...</strong>
<strong class='message'>Also, construct the contact graph under the cutoff 5.0e-02 of adjusted-pvalue (2014-12-16 17:18:27)...</strong>
<strong class='message'></strong>
<strong class='message'>Finish at 2014-12-16 17:18:27</strong>
<strong class='message'>Runtime in total is: 0 secs
</strong>
      </pre>
    </div>

  <div class="span4">
    
    <h2>Source faq</h2>
    <a href="javascript:newWin('faq/FAQ5.r', 'faq/FAQ5.r', '1200', '1200')" title="Click to view"><B><code>FAQ5.r</code></B></a>
    
    <h2>Functions used in this FAQ</h2>
    <ul>
      <li><a href="dRWR.html">dRWR</a></li>
      <li><a href="dNetInduce.html">dNetInduce</a></li>
      <li><a href="dRWRpipeline.html">dRWRpipeline</a></li>
      <li><a href="dRWRcontact.html">dRWRcontact</a></li>
    </ul>
    <!--    
    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) <B>The &#39;dnet&#39; approach promotes emerging research on cancer patient survival</B>. <I>Genome Medicine</I>, 6(8):64. <a href="http://dx.doi.org/10.1186/s13073-014-0064-8">doi:10.1186/s13073-014-0064-8</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/25246945">25246945</a></p>

<p>You may also cite the package as:</p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dnet&rdquo; [<a href="http://cran.r-project.org/package=dnet">http://cran.r-project.org/package=dnet</a>]</p>

    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dnet">http://supfam.org/dnet</a></li>
      <li><a href=" http://dnet.r-forge.r-project.org"> http://dnet.r-forge.r-project.org</a></li>
      <li><a href=" http://cran.r-project.org/package=dnet"> http://cran.r-project.org/package=dnet</a></li>
      <li><a href=" https://github.com/hfang-bristol/dnet"> https://github.com/hfang-bristol/dnet</a></li>
    </ul>
    
    <h2>Author</h2>
    Hai Fang and Julian Gough
    
    -->
    
  </div>

</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>