<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>How many ways are supported to visualise gene-active networks? @ dnet 1.0.7</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="Hai Fang and Julian Gough">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 1.0.7</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="rdata.html"><i class="icon-file icon-white"></i> RData</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="faqs.html"><i class="icon-question-sign icon-white"></i> FAQs</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dnet R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:djnz9oat1hu';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>How many ways are supported to visualise gene-active networks?</h2>

Notes:
<li> All results are based on dnet (version 1.0.7). </li>
<li> R scripts (i.e. R expressions) plus necessary comments are highlighted in light-cyan background, and the rest are outputs in the screen. </li>
<li> Images displayed below may be distorted, but should be normal in your screen. </li>
<li> Functions contained in dnet 1.0.7 are hyperlinked in-place and also listed on the right side. </li>
<li> Key texts are underlined, in bold and in pumpkin-orange color. </li>

<div class="row">
    <div class="span8">
      <pre>
      <div class='input'># <span style="font-weight:bold; color:#F87217; text-decoration:underline">The dnet package supports two ways to visualise the identified networks/graphs: 1) the network itself as a single display, 2) the same network but with multiple colorings/displays according to samples.</span>

# To demonstrate the visuals supported, we use a random graph generated according to the ER model, and only keep the largest component.
g <- erdos.renyi.game(100, 1/100)
g <- <a href='dNetInduce.html' target=dNetInduce>dNetInduce</a>(g, nodes_query=V(g))

# <span style="font-weight:bold; color:#F87217; text-decoration:underline">To display the network itself, the key setting is the layout.</span> Since the network is represented as an object of class 'igraph', the layouts supported in the 'igraph' package can be found in the <a href="http://igraph.org/r/doc/layout.html">layout</a> page. Below is an example using a force-based algorithm proposed by Fruchterman and Reingold:
<a href='visNet.html' target=visNet>visNet</a>(g, layout=layout.fruchterman.reingold)
</div>
<a href="javascript:newWin('FAQ4-2.png', 'FAQ4-2.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ4-2.png', width="400" /></a>
<div class='input'># Additionally, the dnet package also uses two other layouts/diagrams: 1) arc diagram in one-dimensional layout, and 2) circle diagram. 

# <span style="font-weight:bold; color:#F87217; text-decoration:underline">To better display the network, it is advisable to incorporate community information.</span> In doing so, we first identify communities based on a spin-glass model and simulated annealing.
com <- igraph::spinglass.community(g, spins=25)
vgroups <- com$membership
## color nodes: according to communities
mcolors <- visColormap(colormap="rainbow")(length(com))
vcolors <- mcolors[vgroups]
## size nodes: according to degrees
vdegrees <- igraph::degree(g)
## sort nodes: first by communities and then degrees
df <- data.frame(ind=1:vcount(g), vgroups, vdegrees)
ordering <- df[order(vgroups,vdegrees),]$ind

# Now, make comparsions between different layouts
## using 1-dimensional arc diagram
<a href='visNetArc.html' target=visNetArc>visNetArc</a>(g, ordering=ordering, vertex.label.color=vcolors, vertex.color=vcolors, vertex.frame.color=vcolors, vertex.size=log(vdegrees)+0.1)
</div>
<a href="javascript:newWin('FAQ4-4.png', 'FAQ4-4.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ4-4.png', width="400" /></a>
<div class='input'>## using circle diagram (drawn into a single circle)
<a href='visNetCircle.html' target=visNetCircle>visNetCircle</a>(g, colormap="rainbow", com=com, ordering=ordering)
</div>
<a href="javascript:newWin('FAQ4-6.png', 'FAQ4-6.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ4-6.png', width="400" /></a>
<div class='input'>## using circle diagram (drawn into multlpe circles: one circle per community)
<a href='visNetCircle.html' target=visNetCircle>visNetCircle</a>(g, colormap="rainbow", com=com, circles="multiple", ordering=ordering)
</div>
<a href="javascript:newWin('FAQ4-8.png', 'FAQ4-8.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ4-8.png', width="400" /></a>
<div class='input'>## using a force-based algorithm proposed by Fruchterman and Reingold
<a href='visNet.html' target=visNet>visNet</a>(g, colormap="rainbow", layout=layout.fruchterman.reingold, vertex.color=vcolors, vertex.frame.color=vcolors, vertex.shape="sphere")
</div>
<a href="javascript:newWin('FAQ4-10.png', 'FAQ4-10.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ4-10.png', width="400" /></a>
<div class='input'>## when using force-based layout, it is also useful to highlight the communities in the background, and have edges being colored differently according whether an edge lies within a community or between communities. 
mark.groups <- igraph::communities(com)
mark.col <- visColoralpha(mcolors, alpha=0.2)
mark.border <- visColoralpha(mcolors, alpha=0.2)
edge.color <- c("grey", "black")[igraph::crossing(com,g)+1]
<a href='visNet.html' target=visNet>visNet</a>(g, colormap="rainbow", glayout=layout.fruchterman.reingold, vertex.color=vcolors, vertex.frame.color=vcolors, vertex.shape="sphere", mark.groups=mark.groups, mark.col=mark.col, mark.border=mark.border, mark.shape=1, mark.expand=10, edge.color=edge.color)
</div>
<a href="javascript:newWin('FAQ4-12.png', 'FAQ4-12.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ4-12.png', width="400" /></a>
<div class='input'>
# Now, let us look at <span style="font-weight:bold; color:#F87217; text-decoration:underline"> visualising the same network but with the multiple colorings/displays according to samples.</span>

# Assume we have 10 samples, each containing numeric information about nodes in the graph:
nnodes <- vcount(g)
nsamples <- 10
data <- matrix(runif(nnodes*nsamples), nrow=nnodes, ncol=nsamples)
rownames(data) <- V(g)$name

# <span style="font-weight:bold; color:#F87217; text-decoration:underline">In the dnet package, these sample-specific network visuals can be simply as they are provided, or samples being self-organised onto 2D landscape.</span>

## simply as they are provided
<a href='visNetMul.html' target=visNetMul>visNetMul</a>(g, colormap="rainbow", data=data, glayout=layout.fruchterman.reingold)
</div>
<a href="javascript:newWin('FAQ4-14.png', 'FAQ4-14.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ4-14.png', width="400" /></a>
<div class='input'>
## being self-organised onto 2D sample landscape (ie a sheet-shape rectangle grid)
sReorder <- <a href='dNetReorder.html' target=dNetReorder>dNetReorder</a>(g, data, feature="node", node.normalise="none")
</div>
<strong class='message'>Start at 2015-07-21 17:52:08</strong>
<strong class='message'></strong>
<strong class='message'>First, define topology of a map grid (2015-07-21 17:52:08)...</strong>
<strong class='message'>Second, initialise the codebook matrix (36 X 20) using 'linear' initialisation, given a topology and input data (2015-07-21 17:52:08)...</strong>
<strong class='message'>Third, get training at the rough stage (2015-07-21 17:52:08)...</strong>
<strong class='message'>	1 out of 360 (2015-07-21 17:52:08)</strong>
<strong class='message'>	36 out of 360 (2015-07-21 17:52:09)</strong>
<strong class='message'>	72 out of 360 (2015-07-21 17:52:09)</strong>
<strong class='message'>	108 out of 360 (2015-07-21 17:52:09)</strong>
<strong class='message'>	144 out of 360 (2015-07-21 17:52:09)</strong>
<strong class='message'>	180 out of 360 (2015-07-21 17:52:09)</strong>
<strong class='message'>	216 out of 360 (2015-07-21 17:52:09)</strong>
<strong class='message'>	252 out of 360 (2015-07-21 17:52:09)</strong>
<strong class='message'>	288 out of 360 (2015-07-21 17:52:09)</strong>
<strong class='message'>	324 out of 360 (2015-07-21 17:52:09)</strong>
<strong class='message'>	360 out of 360 (2015-07-21 17:52:09)</strong>
<strong class='message'>Fourth, get training at the finetune stage (2015-07-21 17:52:09)...</strong>
<strong class='message'>	1 out of 1440 (2015-07-21 17:52:09)</strong>
<strong class='message'>	144 out of 1440 (2015-07-21 17:52:09)</strong>
<strong class='message'>	288 out of 1440 (2015-07-21 17:52:09)</strong>
<strong class='message'>	432 out of 1440 (2015-07-21 17:52:09)</strong>
<strong class='message'>	576 out of 1440 (2015-07-21 17:52:09)</strong>
<strong class='message'>	720 out of 1440 (2015-07-21 17:52:09)</strong>
<strong class='message'>	864 out of 1440 (2015-07-21 17:52:09)</strong>
<strong class='message'>	1008 out of 1440 (2015-07-21 17:52:09)</strong>
<strong class='message'>	1152 out of 1440 (2015-07-21 17:52:09)</strong>
<strong class='message'>	1296 out of 1440 (2015-07-21 17:52:09)</strong>
<strong class='message'>	1440 out of 1440 (2015-07-21 17:52:09)</strong>
<strong class='message'>Next, identify the best-matching hexagon/rectangle for the input data (2015-07-21 17:52:09)...</strong>
<strong class='message'>Finally, append the response data (hits and mqe) into the sMap object (2015-07-21 17:52:09)...</strong>
<strong class='message'></strong>
<strong class='message'>Below are the summaries of the training results:</strong>
<strong class='message'>   dimension of input data: 10x20
   xy-dimension of map grid: xdim=6, ydim=6
   grid lattice: rect
   grid shape: sheet
   dimension of grid coord: 36x2
   initialisation method: linear
   dimension of codebook matrix: 36x20
   mean quantization error: 0.506666770524648
</strong>
<strong class='message'>Below are the details of trainology:</strong>
<strong class='message'>   training algorithm: sequential
   alpha type: invert
   training neighborhood kernel: gaussian
   trainlength (x input data length): 36 at rough stage; 144 at finetune stage
   radius (at rough stage): from 1 to 1
   radius (at finetune stage): from 1 to 1
</strong>
<strong class='message'>End at 2015-07-21 17:52:09</strong>
<strong class='message'>Runtime in total is: 1 secs
</strong>
<div class='input'><a href='visNetReorder.html' target=visNetReorder>visNetReorder</a>(g, colormap="rainbow", data=data, sReorder)
</div>
<a href="javascript:newWin('FAQ4-54.png', 'FAQ4-54.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ4-54.png', width="400" /></a>
<div class='input'>
## By default, 2D sample landscape is built based on node features without considering node degrees. To take into account the connectivity in the network, the information used can be on edges which are transformed from information on nodes: input data and degree. In doing so, <span style="font-weight:bold; color:#F87217; text-decoration:underline">the transformed matrix of network edges Ã— samples are used for self-organising samples onto 2D landscape (implemented in the 'supraHex' package).</span>
sReorder <- <a href='dNetReorder.html' target=dNetReorder>dNetReorder</a>(g, data, feature="edge", node.normalise="degree")
</div>
<strong class='message'>Start at 2015-07-21 17:52:18</strong>
<strong class='message'></strong>
<strong class='message'>First, define topology of a map grid (2015-07-21 17:52:18)...</strong>
<strong class='message'>Second, initialise the codebook matrix (36 X 19) using 'linear' initialisation, given a topology and input data (2015-07-21 17:52:18)...</strong>
<strong class='message'>Third, get training at the rough stage (2015-07-21 17:52:18)...</strong>
<strong class='message'>	1 out of 360 (2015-07-21 17:52:18)</strong>
<strong class='message'>	36 out of 360 (2015-07-21 17:52:18)</strong>
<strong class='message'>	72 out of 360 (2015-07-21 17:52:18)</strong>
<strong class='message'>	108 out of 360 (2015-07-21 17:52:18)</strong>
<strong class='message'>	144 out of 360 (2015-07-21 17:52:18)</strong>
<strong class='message'>	180 out of 360 (2015-07-21 17:52:18)</strong>
<strong class='message'>	216 out of 360 (2015-07-21 17:52:18)</strong>
<strong class='message'>	252 out of 360 (2015-07-21 17:52:18)</strong>
<strong class='message'>	288 out of 360 (2015-07-21 17:52:18)</strong>
<strong class='message'>	324 out of 360 (2015-07-21 17:52:18)</strong>
<strong class='message'>	360 out of 360 (2015-07-21 17:52:18)</strong>
<strong class='message'>Fourth, get training at the finetune stage (2015-07-21 17:52:18)...</strong>
<strong class='message'>	1 out of 1440 (2015-07-21 17:52:18)</strong>
<strong class='message'>	144 out of 1440 (2015-07-21 17:52:18)</strong>
<strong class='message'>	288 out of 1440 (2015-07-21 17:52:18)</strong>
<strong class='message'>	432 out of 1440 (2015-07-21 17:52:18)</strong>
<strong class='message'>	576 out of 1440 (2015-07-21 17:52:18)</strong>
<strong class='message'>	720 out of 1440 (2015-07-21 17:52:18)</strong>
<strong class='message'>	864 out of 1440 (2015-07-21 17:52:18)</strong>
<strong class='message'>	1008 out of 1440 (2015-07-21 17:52:18)</strong>
<strong class='message'>	1152 out of 1440 (2015-07-21 17:52:18)</strong>
<strong class='message'>	1296 out of 1440 (2015-07-21 17:52:18)</strong>
<strong class='message'>	1440 out of 1440 (2015-07-21 17:52:18)</strong>
<strong class='message'>Next, identify the best-matching hexagon/rectangle for the input data (2015-07-21 17:52:18)...</strong>
<strong class='message'>Finally, append the response data (hits and mqe) into the sMap object (2015-07-21 17:52:18)...</strong>
<strong class='message'></strong>
<strong class='message'>Below are the summaries of the training results:</strong>
<strong class='message'>   dimension of input data: 10x19
   xy-dimension of map grid: xdim=6, ydim=6
   grid lattice: rect
   grid shape: sheet
   dimension of grid coord: 36x2
   initialisation method: linear
   dimension of codebook matrix: 36x19
   mean quantization error: 0.148451719676308
</strong>
<strong class='message'>Below are the details of trainology:</strong>
<strong class='message'>   training algorithm: sequential
   alpha type: invert
   training neighborhood kernel: gaussian
   trainlength (x input data length): 36 at rough stage; 144 at finetune stage
   radius (at rough stage): from 1 to 1
   radius (at finetune stage): from 1 to 1
</strong>
<strong class='message'>End at 2015-07-21 17:52:18</strong>
<strong class='message'>Runtime in total is: 0 secs
</strong>
<div class='input'><a href='visNetReorder.html' target=visNetReorder>visNetReorder</a>(g, colormap="rainbow", data=data, sReorder)</div>
<a href="javascript:newWin('FAQ4-94.png', 'FAQ4-94.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='FAQ4-94.png', width="400" /></a>
      </pre>
    </div>

  <div class="span4">
    
    <h2>Source faq</h2>
    <a href="javascript:newWin('faq/FAQ4.r', 'faq/FAQ4.r', '1200', '1200')" title="Click to view"><B><code>FAQ4.r</code></B></a>
    
    <h2>Functions used in this FAQ</h2>
    <ul>
      <li><a href="dNetInduce.html">dNetInduce</a></li>
      <li><a href="visNet.html">visNet</a></li>
      <li><a href="visNetArc.html">visNetArc</a></li>
      <li><a href="visNetCircle.html">visNetCircle</a></li>
      <li><a href="visNetMul.html">visNetMul</a></li>
      <li><a href="dNetReorder.html">dNetReorder</a></li>
      <li><a href="visNetReorder.html">visNetReorder</a></li>
    </ul>
    <!--    
    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) <B>The &#39;dnet&#39; approach promotes emerging research on cancer patient survival</B>. <I>Genome Medicine</I>, 6(8):64. <a href="http://dx.doi.org/10.1186/s13073-014-0064-8">doi:10.1186/s13073-014-0064-8</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/25246945">25246945</a></p>

<p>You may also cite the package as:</p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dnet&rdquo; [<a href="http://cran.r-project.org/package=dnet">http://cran.r-project.org/package=dnet</a>]</p>

    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/dnet">http://supfam.org/dnet</a></li>
      <li><a href=" http://dnet.r-forge.r-project.org"> http://dnet.r-forge.r-project.org</a></li>
      <li><a href=" http://cran.r-
project.org/package=dnet"> http://cran.r-
project.org/package=dnet</a></li>
      <li><a href=" https://github.com/hfang-bristol/dnet"> https://github.com/hfang-bristol/dnet</a></li>
    </ul>
    
    <h2>Author</h2>
    Hai Fang and Julian Gough
    
    -->
    
  </div>

</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@well.ox.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>