<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>Home @ dnet 1.0.0</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="Hai Fang and Julian Gough">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 1.0.0</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <div class="row">
  <div class="span8">
    <p><a href="index.html"><IMG src="dnet_logo1.png" height="100px" id="logo"></a></p>

<h2>Introduction</h2>

<p><code>dnet</code> is an open-source R package for analysing biological networks and high-throughput biological data in an integrative manner.</p>

<p><code>dnet</code> intends to analyse the biological network whose nodes/genes are associated with dynamic information such as expression levels across samples.</p>

<p><code>dnet</code> aims to deliver an eye-intuitive tool for network-based sample stratifications.</p>

<h2>Features</h2>

<ul>
<li>An integrated framework for integration of network and expression information for classifying/stratifying samples</li>
<li>Identification of expression-active dynamic subnetworks</li>
<li>Network-based sample stratifications </li>
<li>Visualisations of network-based sample stratifications on 2D landscape</li>
<li>This package can run on <code>Windows</code>, <code>Mac</code> and <code>Linux</code></li>
</ul>

<h2>See also</h2>

<ul>
<li><a href="http://igraph.sourceforge.net">igraph</a></li>
<li><a href="http://supfam.org/supraHex">supraHex</a></li>
</ul>


  </div>
  
  <div class="span3 offset1">
  

    <h2>Manual</h2>
    <ul>

    
    
        <li>dnet Reference Manual (<a href="vignettes/dnet.pdf">PDF</a>, <a href="docs.html">HTML</a>)</li>
    
    </ul>

    <h2>Demos</h2>
    <ul>
      <li><a href="demo-CLL.html">Demo for patient omics dataset</a></li>
      <li><a href="demo-Hiratani.html">Demo for multilayer omics dataset</a></li>
      <li><a href="demo-Fang.html">Demo for time-course omics dataset</a></li>
    </ul>
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></li>
    </ul>
    
    <h2>Dependency</h2>
    <ul>
      <li><strong>Depends</strong>: igraph, supraHex</li>
      <li><strong>Imports</strong>: ape</li>
      <li><strong>Suggests</strong>: affy, limma</li>
      <li><strong>Extends</strong>: </li>
    </ul>
    
    <h2>License</h2>
    GPL-2
    
    <h2>Author</h2>
    Hai Fang and Julian Gough
    
    <!--
    
    
    <h2>Manual</h2>
    <ul>
        <li><a href="vignettes/dnet.pdf">dnet Reference Manual</a></li>
    </ul>
    
    <h2>Maintainer</h2>
    Hai Fang &lt;hfang@cs.bris.ac.uk&gt;
    
    -->
    
  </div>
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>