<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>org.Mm.egHPON @ dnet 0.99.0</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 0.99.0</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <!--
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
        -->
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Annotations of Mouse Entrez Genes (EG) by Human Phenotype ONset and clinical course (HPON).</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p>An R object that contains associations between HPON terms and Mouse Entrez Genes. This data is first prepared based on <a href = 'http://compbio.charite.de/svn/hpo/trunk/src/ontology/human-phenotype-ontology.obo'>http://compbio.charite.de/svn/hpo/trunk/src/ontology/human-phenotype-ontology.obo</a> and  <a href = 'http://compbio.charite.de/hudson/job/hpo.annotations.monthly/lastStableBuild/artifact/annotation/ALL_SOURCES_ALL_FREQUENCIES_genes_to_phenotype.txt'>http://compbio.charite.de/hudson/job/hpo.annotations.monthly/lastStableBuild/artifact/annotation/ALL_SOURCES_ALL_FREQUENCIES_genes_to_phenotype.txt</a>, which results in annotations of Human Entrez Genes. Then, these annotations are transferred to Mouse Entrez Genes based on <a href = 'ftp://anonymous@ftp.ncbi.nih.gov/pub/HomoloGene/build67/homologene.data'>ftp://anonymous@ftp.ncbi.nih.gov/pub/HomoloGene/build67/homologene.data</a>.</p>
  
    <h3>Usage</h3>
    <pre>data(org.Mm.egHPON)</pre>
        <!-- 
    -->
    
    <div class="Value">
      <h3>Value</h3>
      
      <p>an object of class "GS", a list with following components:
    <ul>
<li><code>set_info</code>: a matrix of nSet X 4 containing gene set information, where nSet is the number of gene sets (i.e. HPON terms), and the 4 columns are "setID" (i.e. "Term ID"), "name" (i.e. "Term Name"), "namespace" and "distance"
        </li>
<li><code>gs</code>: a list of gene sets, each storing gene members thereof. Always, gene sets are identified by "setID" and gene members identified by "Entrez ID"
    </li>
</ul></p>
      <p></p>
    </div>
    <div class="References">
      <h3>References</h3>
      
      <p>Robinson et al. (2012) The Human Phenotype Ontology: a tool for annotating and analyzing human hereditary disease. <em>Am J Hum Genet</em>, 83:610-615.</p>
      <p>Sayers et al. (2011) Database resources of the National Center for Biotechnology Information. <em>Nucleic Acids Res</em>, 39:D38-51.</p>
    </div>
      </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>org.Mm.egHPON</li>
    </ul>
    <ul>
      <li>datasets</li>
    </ul> 
    -->
      
        
    <!--
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>