<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>org.Hs.PCommons_DN.source @ dnet 1.0.11</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">dnet 1.0.11</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="rdata.html"><i class="icon-file icon-white"></i> RData</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="faqs.html"><i class="icon-question-sign icon-white"></i> FAQs</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
       
        <!-- HTML for GOOLE SEARCH BAR 
        <ul class="nav">
            <div id="search-wrapper">
                <form id="search" method="get" action="http://www.google.com/custom">
                    <div class="search-box">
                        <i class="icon-search icon-white"></i>
                        <input type="text" id="tfq" class="textinput" name="q" size="21" maxlength="120" value="Google Search"/>
                    </div>
                </form>
            </div>
        </ul>
        -->
        
      </div>
    </div>
  </div>
</div>

<!-- CSS styles for standard search box with placeholder text-->
<style type="text/css">
    #search-wrapper {
        width:300px;
        height:40px;
        background-color:transparent;
        margin:0px auto 0;
        border:0px solid #ff0000;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

    #search {
        background-color:transparent;
        padding:0px;
        height:40px;
        overflow:hidden;
    }

    .search-box {
        width:250px;
        height:32px;
        background-color:transparent;
        margin:2px 7px;
        border:0px solid #cfcfcf;
        -moz-border-radius: 2px; /* FF1+ */
        -webkit-border-radius: 2px; /* Saf3-4 */
        border-radius: 2px; /* Opera 10.5, IE 9, Saf5, Chrome */
        position:relative;
    }

	.textinput{
		margin: 0;
		padding: 5px 15px;
		font-family: Arial, Helvetica, sans-serif;
		font-size:14px;
		color:#999999;
		border:1px solid #f00; border-right:0px;
		border-top-left-radius: 5px 5px;
		border-bottom-left-radius: 5px 5px;
	}
</style>

<!-- JAVASCRIPT to clear search text when the field is clicked -->
<script type="text/javascript">
window.onload = function(){
	//Get submit button
	var submitbutton = document.getElementById("tfq");
	//Add listener to submit button
	if(submitbutton.addEventListener){
		submitbutton.addEventListener("click", function() {
			if (submitbutton.value == 'Google Search'){//Customize this text string to whatever you want
				submitbutton.value = 'dnet R ';
			}
		});
	}
}
</script>

<script>
  (function() {
    var cx = '012094530736424854714:djnz9oat1hu';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Human protein interaction directed network for each data source from Pathway Commons (version 7).</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p>A list of igraph objects, each object storing a human protein interaction (directed) network for one of 9 data sources. These data sources are downloaded from from Pathway Commons (version 7; a comprehensive collection of public biological interactions/pathways from multiple data sources). Data sources are: <code>CTD</code> (see <a href = 'http://ctdbase.org'>http://ctdbase.org</a> and <a href = 'http://www.ncbi.nlm.nih.gov/pubmed/25326323'>http://www.ncbi.nlm.nih.gov/pubmed/25326323</a>), <code>HumanCyc</code> (see <a href = 'http://www.humancyc.org'>http://www.humancyc.org</a> and <a href = 'http://www.ncbi.nlm.nih.gov/pubmed/15642094'>http://www.ncbi.nlm.nih.gov/pubmed/15642094</a>), <code>KEGG</code> (see <a href = 'http://www.genome.jp/kegg'>http://www.genome.jp/kegg</a> and <a href = 'http://www.ncbi.nlm.nih.gov/pubmed/23433509'>http://www.ncbi.nlm.nih.gov/pubmed/23433509</a>), <code>PANTHER</code> (see <a href = 'http://www.pantherdb.org'>http://www.pantherdb.org</a> and <a href = 'http://www.ncbi.nlm.nih.gov/pubmed/23193289'>http://www.ncbi.nlm.nih.gov/pubmed/23193289</a>), <code>PhosphoSite</code> (see <a href = 'http://www.phosphosite.org'>http://www.phosphosite.org</a> and <a href = 'http://www.ncbi.nlm.nih.gov/pubmed/22135298'>http://www.ncbi.nlm.nih.gov/pubmed/22135298</a>), <code>PID</code> (see <a href = 'http://pid.nci.nih.gov'>http://pid.nci.nih.gov</a> and <a href = 'http://www.ncbi.nlm.nih.gov/pubmed/18832364'>http://www.ncbi.nlm.nih.gov/pubmed/18832364</a>), <code>Reactome</code> (see <a href = 'http://www.reactome.org'>http://www.reactome.org</a> and <a href = 'http://www.ncbi.nlm.nih.gov/pubmed/24243840'>http://www.ncbi.nlm.nih.gov/pubmed/24243840</a>), <code>ReconX</code> (see <a href = 'http://humanmetabolism.org'>http://humanmetabolism.org</a> and <a href = 'http://www.ncbi.nlm.nih.gov/pubmed/23455439'>http://www.ncbi.nlm.nih.gov/pubmed/23455439</a>), <code>TRANSFAC</code> (see <a href = 'http://www.gene-regulation.com'>http://www.gene-regulation.com</a> and <a href = 'http://www.ncbi.nlm.nih.gov/pubmed/18436575'>http://www.ncbi.nlm.nih.gov/pubmed/18436575</a>). Only those interactions with directed relations (including relation types: "catalysis_precedes", "controls_expression_of", "controls_phosphorylation_of", "controls_state_change_of" and "controls_transport_of") are retained. For the pooled version of this network, please refer to <code><a href='org.Hs.PCommons_DN.html'>org.Hs.PCommons_DN</a></code>.</p>
  
    <h3>Usage</h3>
    <pre>org.Hs.PCommons_DN.source <- dRDataLoader(RData.customised = "org.Hs.PCommons_DN.source")</pre>
        <!-- 
    -->
    
    <!-- 
    <div class="Value">
    -->
      <h3>Value</h3>
      
      <p>A list of igraph objects. For details about the object of class "igraph", please see <a href = 'http://igraph.org/r/doc/aaa-igraph-package.html'>http://igraph.org/r/doc/aaa-igraph-package.html</a>. It has attributes for both vertices and edges. 
    Below are attributes for the vertices:
    <ul>
<li><code>name</code>: gene symbol for the vertices
        </li>
<li><code>geneid</code>: Entrez geneid for the vertices
        </li>
<li><code>symbol</code>: gene symbol for the vertices
        </li>
<li><code>description</code>: gene description for the vertices        
    </li>
</ul></p>
      <p>Below are attributes for the edges:
    <ul>
<li><code>catalysis_precedes</code>: an integer code indicating support evidence for "catalysis_precedes" relation, that is, the first protein (from) controls a reaction whose output molecule is input to another reaction controled by the second protein (to)
        </li>
<li><code>controls_expression_of</code>: an integer code indicating support evidence for "controls_expression_of" relation, that is, the first protein (from) controls a conversion or a template reaction that changes expression of the second protein (to)
        </li>
<li><code>controls_phosphorylation_of</code>: an integer code indicating support evidence for "controls_phosphorylation_of" relation, that is, the first protein (from) controls a reaction that changes the phosphorylation status of the second protein (to)
        </li>
<li><code>controls_state_change_of</code>: an integer code indicating support evidence for "controls_state_change_of" relation, that is, the first protein (from) controls a reaction that changes the state of the second protein (to)
        </li>
<li><code>controls_transport_of</code>: an integer code indicating support evidence for "controls_transport_of" relation, that is, the first protein (from) controls a reaction that changes the cellular location of the second protein (to)
        </li>
<li>Notably, there are two supporting evidences: with the PubMed references (assigned with 100), and the number of different data sources. The integer code is the sum of these two. A value of 0 means no supporting evidences. For examples, a value of 105 means that the relation is supported by the PubMed references and by 5 different data sources.
    </li>
</ul></p>
      <p></p>
    <!-- 
    </div>
    -->
    <!-- 
    <div class="References">
    -->
      <h3>References</h3>
      
      <p>Cerami et al. (2011) Pathway Commons, a web resource for biological pathway data. <em>Nucleic Acids Res</em>, 39(Database issue):D685-90.</p>
    <!-- 
    </div>
    -->
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'>org.Hs.PCommons_DN.source <- dRDataLoader(RData.customised='org.Hs.PCommons_DN.source')
</div>
<strong class='error'>Error in dRDataLoader(RData.customised = "org.Hs.PCommons_DN.source"): unused argument (RData.customised = "org.Hs.PCommons_DN.source")</strong>
<div class='input'>names(org.Hs.PCommons_DN.source)
</div>
<strong class='error'>Error in eval(expr, envir, enclos): object 'org.Hs.PCommons_DN.source' not found</strong>
<div class='input'>org.Hs.PCommons_DN.source$KEGG
</div>
<strong class='error'>Error in eval(expr, envir, enclos): object 'org.Hs.PCommons_DN.source' not found</strong></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>org.Hs.PCommons_DN.source</li>
    </ul>
    <ul>
      <li>datasets</li>
    </ul> 
    -->
    
    
        
    <!--

    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) <B>The &#39;dnet&#39; approach promotes emerging research on cancer patient survival</B>. <I>Genome Medicine</I>, 6(8):64. <a href="http://dx.doi.org/10.1186/s13073-014-0064-8">doi:10.1186/s13073-014-0064-8</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/25246945">25246945</a></p>

<p>You may also cite the package as:</p>

<ul>
<li><code>web link/URL</code></li>
</ul>

<p>Package &ldquo;dnet&rdquo; [<a href="http://cran.r-project.org/package=dnet">http://cran.r-project.org/package=dnet</a>]</p>

    
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://dnet.r-forge.r-project.org">http://dnet.r-forge.r-project.org</a></li>
      <li><a href=" https://github.com/hfang-bristol/dnet"> https://github.com/hfang-bristol/dnet</a></li>
      <li><a href=" http://supfam.org/dnet"> http://supfam.org/dnet</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang.shanghai@gmail.com" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>