\name{org.Hs.egMsigdbC1}
\alias{org.Hs.egMsigdbC1}

\title{Annotations of Human Entrez Genes (EG) by C1 collections.}


\usage{
org.Hs.egMsigdbC1 <- dRDataLoader(RData='org.Hs.egMsigdbC1')
}

\description{
This data is prepared based on the molecular signatures database (Msigdb; \url{http://www.broadinstitute.org/gsea/msigdb/index.jsp}). An R object that contains associations between Msigdb C1 positional gene sets and Human Entrez Genes. C1 collections are about positional gene sets for each human chromosome and cytogenetic band, each gene set corresponding to each human chromosome and each cytogenetic band that has at least one gene. These gene sets are helpful in identifying effects related to chromosomal deletions or amplifications, dosage compensation, epigenetic silencing, and other regional effects.
}

\value{
    an object of class "GS", a list with following components:
    \itemize{
        \item{\code{set_info}: a matrix of nSet X 4 containing gene set information, where nSet is the number of gene sets, and the 4 columns are "setID" (i.e. "Geneset systematic name"), "name" (i.e. "Geneset standard description"), "namespace" (i.e. Geneset category code) and "distance" (i.e. "Geneset brief description")}
        \item{\code{gs}: a list of gene sets, each storing gene members thereof. Always, gene sets are identified by "setID" and gene members identified by "Entrez ID"}
    }
}

\references{
Subramanian et al. (2005) Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles. \emph{Proc Natl Acad Sci U S A.}, 102(43):15545-50.
}

\keyword{datasets}

\examples{
org.Hs.egMsigdbC1 <- dRDataLoader(RData='org.Hs.egMsigdbC1')
names(org.Hs.egMsigdbC1)
}
